---
title: "Script to merge data"
author: "Jaakko Koskenniemi"
output: 
  html_document:
    number_sections: true
date: '2022-24-04'
editor_options: 
  chunk_output_type: console
---

```{r}
library(tidyverse)
library(rio)
library(here)

here::i_am("analyses/Harmonize.Rmd")
```


This script merges the annotation and protein data. 

GWAS sumstat of type 1 diabetes was obtained from the study by Chiou et al. 
(Nature 2021, https://doi.org/10.1038/s41586-021-03552-w). GWAS sumstat for 
protein levels of IL2RG was obtained from the study by Ferkingstad et al. 
(Nature 2021, https://doi.org/10.1038/s41588-021-00978-w).

Variants within 1 megabase before the start site of IL2RG and 1 megabase 
after the stop site are included. See '/code/gather.R' on how the the 
variants within this area were filtered in each GWAS. The gather-script 
was run in the high performance computing cluster using the 
'run_gather.sh' script. The output of '/code/Gather.R' is saved in 
'/data/cis_sumstats/', and for each possible target gene
there are three files: 

1. *_T1D_data.rds: GWAS sumstat for the risk of T1D (Chiou et al. 2021)
2. *_prot_data.rds: GWAS sumstat of serum protein level (Ferkingstad et al. 2021)
3. *_anno_data.rds: further annotation for GWAS of serum protein level  
Where * is a gene of interest (IL2RA, IL2RG, IL2RG, IL6ST, IL6R, etc.)

except for IL6R for which 
IL6R_prot_data.rds is substituted with IL6R_crp_data.rds

```{r}
#add here a figure that shows the project workflow
```


# Exposure data

## Import exposure data

```{r}
IL2RG_anno  <- import("data/cis_sumstats/IL2RG_anno.rds")
IL2RG_prot  <- import("data/cis_sumstats/IL2RG_gwas.rds")

CXCL10_anno <- import("data/cis_sumstats/CXCL10_anno.rds")  
CXCL10_prot <- import("data/cis_sumstats/CXCL10_prot.rds")  

IL12B_anno  <- import("data/cis_sumstats/IL12B_anno.rds")  
IL12B_prot  <- import("data/cis_sumstats/IL12B_prot.rds")  

IL2RA_anno  <- import("data/cis_sumstats/IL2RA_anno.rds")  
IL2RA_prot  <- import("data/cis_sumstats/IL2RA_prot.rds")  

IL2RB_prot  <- import("data/cis_sumstats/IL2RB_prot.rds")  
IL2RB_anno  <- import("data/cis_sumstats/IL2RB_anno.rds")  

IL6R_crp   <- import("data/cis_sumstats/IL6R_crp.rds")  
IL6R_anno  <- import("data/cis_sumstats/IL6R_anno.rds")  
IL6R_prot  <- import("data/cis_sumstats/IL6R_prot.rds")

IL6ST_anno  <- import("data/cis_sumstats/IL6ST_anno.rds")  
IL6ST_prot  <- import("data/cis_sumstats/IL6ST_prot.rds")

JAK2_anno   <- import("data/cis_sumstats/JAK2_anno.rds")  
JAK2_prot   <- import("data/cis_sumstats/JAK2_prot.rds")  

TNF_anno    <- import("data/cis_sumstats/TNF_anno.rds")  
TNF_prot    <- import("data/cis_sumstats/TNF_prot.rds")  

TYK2_anno   <- import("data/cis_sumstats/TYK2_anno.rds")
TYK2_prot   <- import("data/cis_sumstats/TYK2_prot.rds")
```


## Merge protein data with annotations

Ferkingstad et al. state in their README file  

>A subset of the variants in the summary statistics files should be excluded 
due to quality issues. These variants are listed in a separate Excluded variants 
file (assocvariants.excluded.txt.gz). The Extra annotation file 
(assocvariants.annotated.txt.gz) does not include these variants.

Therefore in the join process, only the data available in the annotation (IL2RG_anno)
is used (hence, left_join)

```{r}
CXCL10_prot_anno <- left_join(CXCL10_prot, CXCL10_anno, by = c("Name", "Chrom", "Pos"))
IL12B_prot_anno  <- left_join(IL12B_prot, IL12B_anno, by = c("Name", "Chrom", "Pos"))
IL2RA_prot_anno  <- left_join(IL2RA_prot, IL2RA_anno, by = c("Name", "Chrom", "Pos"))
IL2RB_prot_anno  <- left_join(IL2RB_prot, IL2RB_anno, by = c("Name", "Chrom", "Pos"))
IL2RG_prot_anno  <- left_join(IL2RG_prot, IL2RG_anno, by = c("Name", "Chrom", "Pos"))
# IL6R_prot_anno   <- left_join(IL6R_prot,  IL6R_anno, by = c("Name", "Chrom", "Pos"))
IL6ST_prot_anno  <- left_join(IL6ST_prot, IL6ST_anno, by = c("Name", "Chrom", "Pos"))
JAK2_prot_anno   <- left_join(JAK2_prot,  JAK2_anno, by = c("Name", "Chrom", "Pos"))
TNF_prot_anno    <- left_join(TNF_prot,   TNF_anno, by = c("Name", "Chrom", "Pos"))
TYK2_prot_anno   <- left_join(TYK2_prot,  TYK2_anno, by = c("Name", "Chrom", "Pos"))

rm("CXCL10_anno", "CXCL10_prot", "IL12B_anno", "IL12B_prot", 
   "IL2RA_anno", "IL2RA_prot", "IL2RB_anno", "IL2RB_prot",  
   "IL2RG_anno", "IL2RG_prot", "IL6R_anno", "IL6R_crp", 
   "IL6ST_anno", "IL6ST_prot", "JAK2_anno", "JAK2_prot", 
   "TNF_anno", "TNF_prot", "TYK2_anno", "TYK2_prot", "IL6R_prot")
```


## Sort conflicts

Further in the README file of Ferkingstad et al.  

> Also, the summary statistics files sometimes incorrectly have effectAllele=otherAllele for multiallelic variants.
In these cases the effectAllele is correct, but the otherAllele should be '!', meaning that the effectAllele is tested against the other (two or more) alleles (using the '!' sign as shorthand for 'not effectAllele').
This has been corrected in the file Extra annotation file (assocvariants.annotated.txt.gz).

Therefore, multiallelic variants were removed.
```{r}
CXCL10_prot_anno <- CXCL10_prot_anno %>% filter(otherAllele.y != "!")
IL12B_prot_anno  <- IL12B_prot_anno  %>% filter(otherAllele.y != "!")
IL2RA_prot_anno  <- IL2RA_prot_anno  %>% filter(otherAllele.y != "!")
IL2RB_prot_anno  <- IL2RB_prot_anno  %>% filter(otherAllele.y != "!")
IL2RG_prot_anno  <- IL2RG_prot_anno  %>% filter(otherAllele.y != "!")
IL6ST_prot_anno  <- IL6ST_prot_anno  %>% filter(otherAllele.y != "!")
JAK2_prot_anno   <- JAK2_prot_anno   %>% filter(otherAllele.y != "!")
TNF_prot_anno    <- TNF_prot_anno    %>% filter(otherAllele.y != "!")
TYK2_prot_anno   <- TYK2_prot_anno   %>% filter(otherAllele.y != "!")

```

After removal of multiallelic variants there were no conflicts 
between effectAllele.x and effectAllele.y or between 
otherAllele.x and otherAllele.y (results suppressed,
please see "analyses/Harmonize.Rmd") 

```{r, results = 'hide'}

#Following function is simply to avoid repetition
check_conflicts_ea_oa <- function(data, effectAllele.x, effectAllele.y) {
  filter(data, effectAllele.x != effectAllele.y | 
           is.na(effectAllele.x) & !is.na(effectAllele.y) |
           !is.na(effectAllele.x) &  is.na(effectAllele.y) | 
           otherAllele.x != otherAllele.y | 
            is.na(otherAllele.x) & !is.na(otherAllele.y) |
           !is.na(otherAllele.x) &  is.na(otherAllele.y))
}

#List those with conflicts in effectAllele or otherAllele

CXCL10_prot_anno %>% check_conflicts_ea_oa() #0 conflicts
IL12B_prot_anno  %>% check_conflicts_ea_oa() #0 conflicts
IL2RA_prot_anno  %>% check_conflicts_ea_oa() #0 conflicts
IL2RB_prot_anno  %>% check_conflicts_ea_oa() #0 conflicts
IL2RG_prot_anno  %>% check_conflicts_ea_oa() #0 conflicts
IL6ST_prot_anno  %>% check_conflicts_ea_oa() #0 conflicts
JAK2_prot_anno   %>% check_conflicts_ea_oa() #0 conflicts
TNF_prot_anno    %>% check_conflicts_ea_oa() #0 conflicts
TYK2_prot_anno   %>% check_conflicts_ea_oa() #0 conflicts

```

After this, the remaining discrepancies between the variables
obtained from annotation and protein data files were because those
variants without an rsid were listed as "." in the annotation file
and as "NA" in the main GWAS sumstat file (please see 
'/analyses/Harmonize.Rmd')

```{r, results = 'hide'}
#List subjects which had conflicting rsid information and show
#which entries there are for these
check_conflicts_rsid <- function(data, effectAllele.x, effectAllele.y) {
  filter(data, rsids.x != rsids.y | 
           is.na(rsids.x) & !is.na(rsids.y) |
           !is.na(rsids.x) &  is.na(rsids.y)) %>%
    mutate(rsids.x = factor(rsids.x),
         rsids.y = factor(rsids.y)) %>%
  select(rsids.x, rsids.y) %>%
  summary()
}

#All the conflicts in rsid.x and rsid.y are because rsid.x is NA and rsid.y is "."
CXCL10_prot_anno %>% check_conflicts_rsid() #0 conflicts
IL12B_prot_anno  %>% check_conflicts_rsid() #0 conflicts
IL2RA_prot_anno  %>% check_conflicts_rsid() #0 conflicts
IL2RB_prot_anno  %>% check_conflicts_rsid() #0 conflicts
IL2RG_prot_anno  %>% check_conflicts_rsid() #0 conflicts
# IL6ST_prot_anno  %>% check_conflicts_rsid() #0 
JAK2_prot_anno   %>% check_conflicts_rsid() #0 conflicts
TNF_prot_anno    %>% check_conflicts_rsid() #0 conflicts
TYK2_prot_anno   %>% check_conflicts_rsid() #0 conflicts

```

Thus, only those values variables with .x in their name are preserved. 
Here, we also remove the remaining multiallelic SNPs 
(where effect allele equals the other allele.)

```{r}
#Discard all variables that end with an ".y"
IL2RG_prot_anno   <- IL2RG_prot_anno  %>% select(!ends_with(".y"))
CXCL10_prot_anno  <- CXCL10_prot_anno %>% select(!ends_with(".y"))
IL12B_prot_anno   <- IL12B_prot_anno  %>% select(!ends_with(".y"))
IL2RA_prot_anno   <- IL2RA_prot_anno  %>% select(!ends_with(".y"))
IL2RB_prot_anno   <- IL2RB_prot_anno  %>% select(!ends_with(".y"))
IL2RG_prot_anno   <- IL2RG_prot_anno  %>% select(!ends_with(".y"))
# IL6ST_prot_anno <- IL6ST_prot_anno  %>% select(!ends_with(".y"))
JAK2_prot_anno    <- JAK2_prot_anno   %>% select(!ends_with(".y"))
TNF_prot_anno     <- TNF_prot_anno    %>% select(!ends_with(".y"))
TYK2_prot_anno    <- TYK2_prot_anno   %>% select(!ends_with(".y"))

#Remove ".x" from variable names
IL2RG_prot_anno   <- IL2RG_prot_anno  %>% rename_with(~gsub(".x", "", .x, fixed = TRUE)) 
CXCL10_prot_anno  <- CXCL10_prot_anno %>% rename_with(~gsub(".x", "", .x, fixed = TRUE)) 
IL12B_prot_anno   <- IL12B_prot_anno  %>% rename_with(~gsub(".x", "", .x, fixed = TRUE)) 
IL2RA_prot_anno   <- IL2RA_prot_anno  %>% rename_with(~gsub(".x", "", .x, fixed = TRUE)) 
IL2RB_prot_anno   <- IL2RB_prot_anno  %>% rename_with(~gsub(".x", "", .x, fixed = TRUE)) 
IL2RG_prot_anno   <- IL2RG_prot_anno  %>% rename_with(~gsub(".x", "", .x, fixed = TRUE)) 
# IL6ST_prot_anno <- IL6ST_prot_anno  %>% rename_with(~gsub(".x", "", .x, fixed = TRUE)) 
JAK2_prot_anno    <- JAK2_prot_anno   %>% rename_with(~gsub(".x", "", .x, fixed = TRUE)) 
TNF_prot_anno     <- TNF_prot_anno    %>% rename_with(~gsub(".x", "", .x, fixed = TRUE)) 
TYK2_prot_anno    <- TYK2_prot_anno   %>% rename_with(~gsub(".x", "", .x, fixed = TRUE)) 
```


```{r, results='hide'}
#Sanity checks
colnames(IL2RG_prot_anno)   
colnames(CXCL10_prot_anno)  
colnames(IL12B_prot_anno)
colnames(IL2RA_prot_anno)
colnames(IL2RB_prot_anno)
colnames(IL2RG_prot_anno)
#colnames(IL6ST_prot_anno)
colnames(JAK2_prot_anno)
colnames(TNF_prot_anno)
colnames(TYK2_prot_anno)
```

All data files seem to have entries without rsids (suppressed,
please see '/analyses/Harmonize.Rmd'). They are now removed

```{r, results = 'hide'}
anyNA(IL2RG_prot_anno$rsids)   
anyNA(CXCL10_prot_anno$rsids)  
anyNA(IL12B_prot_anno$rsids)
anyNA(IL2RA_prot_anno$rsids)
anyNA(IL2RB_prot_anno$rsids)
anyNA(IL2RG_prot_anno$rsids)
#anyNA(IL6ST_prot_anno$rsids)
anyNA(JAK2_prot_anno$rsids)
anyNA(TNF_prot_anno$rsids)
anyNA(TYK2_prot_anno$rsids)

IL2RG_prot_anno   <- IL2RG_prot_anno  %>% filter(!is.na(rsids))
CXCL10_prot_anno  <- CXCL10_prot_anno %>% filter(!is.na(rsids))
IL12B_prot_anno   <- IL12B_prot_anno  %>% filter(!is.na(rsids))
IL2RA_prot_anno   <- IL2RA_prot_anno  %>% filter(!is.na(rsids))
IL2RB_prot_anno   <- IL2RB_prot_anno  %>% filter(!is.na(rsids))
IL2RG_prot_anno   <- IL2RG_prot_anno  %>% filter(!is.na(rsids))
# IL6ST_prot_anno <- IL6ST_prot_anno  %>% filter(!is.na(rsids))
JAK2_prot_anno    <- JAK2_prot_anno   %>% filter(!is.na(rsids))
TNF_prot_anno     <- TNF_prot_anno    %>% filter(!is.na(rsids))
TYK2_prot_anno    <- TYK2_prot_anno   %>% filter(!is.na(rsids))
```


There were no duplicated rsids in data after exclusion of missing 
rsids (suppressed, please see /analyses/Harmonize.Rmd)

```{r, results='hide'}
#Sanity 
anyDuplicated(IL2RG_prot_anno$rsids)   
anyDuplicated(CXCL10_prot_anno$rsids)  
anyDuplicated(IL12B_prot_anno$rsids)
anyDuplicated(IL2RA_prot_anno$rsids)
anyDuplicated(IL2RB_prot_anno$rsids)
anyDuplicated(IL2RG_prot_anno$rsids)
#anyDuplicated(IL6ST_prot_anno$rsids)
anyDuplicated(JAK2_prot_anno$rsids)
anyDuplicated(TNF_prot_anno$rsids)
anyDuplicated(TYK2_prot_anno$rsids)
```


## Import T1D data

```{r}
# Import data
IL2RG_T1D <- import("data/cis_sumstats/IL2RG_T1D.rds") %>% tibble()
```

There were no entries with missing rsids, but there were duplicated rsids. They are removed.
```{r}
anyDuplicated(IL2RG_T1D$hm_rsid) #Duplicates
anyNA(IL2RG_T1D$hm_rsid) #No missing data

duplicated_snps_IL2RG_T1D <- IL2RG_T1D %>%
  filter(duplicated(hm_rsid)) %>% 
  pull(hm_rsid)

#Remove duplicated SNPs
IL2RG_T1D <- IL2RG_T1D %>%
  filter(!hm_rsid %in% duplicated_snps_IL2RG_T1D)
```


## Write data in TwoSampleMR format

Data are written in TwoSampleMR format so that they can be easily opened
in the main analyses scripts.

Rename files
```{r}
rename_for_TwoSampleMR_e  <- function(data, rsids, Beta, SE, effectAlleleFreq,
                                    effectAllele, otherAllele,
                                    Pval, Pos, N) {
  data %>%
    rename(SNP = rsids, beta = Beta, se = SE, eaf = effectAlleleFreq,
         effect_allele = effectAllele, other_allele = otherAllele,
         pval = Pval, pos = Pos) %>%
    mutate(Phenotype = "Serum IL2RG level",
         samplesize = N) 
  }

#Rename
CXCL10_prot_anno  <- CXCL10_prot_anno %>% rename_for_TwoSampleMR_e()
IL12B_prot_anno   <- IL12B_prot_anno  %>% rename_for_TwoSampleMR_e()
IL2RA_prot_anno   <- IL2RA_prot_anno  %>% rename_for_TwoSampleMR_e()
IL2RB_prot_anno   <- IL2RB_prot_anno  %>% rename_for_TwoSampleMR_e()
IL2RG_prot_anno   <- IL2RG_prot_anno  %>% rename_for_TwoSampleMR_e()
# IL6ST_prot_anno <- IL6ST_prot_anno  %>% rename_for_TwoSampleMR_e()
JAK2_prot_anno    <- JAK2_prot_anno   %>% rename_for_TwoSampleMR_e()
TNF_prot_anno     <- TNF_prot_anno    %>% rename_for_TwoSampleMR_e()
TYK2_prot_anno    <- TYK2_prot_anno   %>% rename_for_TwoSampleMR_e()
```


Export filest to a CSV file with default parameters of 
TwoSampleMR::read_exposure_data()
```{r}
#Export
CXCL10_prot_anno %>%
  export("data/export/CXCL10_prot_anno_TwoSampleMR.csv")
IL12B_prot_anno  %>% 
  export("data/export/IL12B_prot_anno_TwoSampleMR.csv")
IL2RA_prot_anno  %>% 
  export("data/export/IL2RA_prot_anno_TwoSampleMR.csv")
IL2RB_prot_anno  %>% 
  export("data/export/IL2RB_prot_anno_TwoSampleMR.csv")
IL2RG_prot_anno  %>% 
  export("data/export/IL2RG_prot_anno_TwoSampleMR.csv")
# IL6ST_prot_anno  %>%
#   export("data/export/IL6ST_prot_anno_TwoSampleMR.csv")
JAK2_prot_anno   %>% 
  export("data/export/JAK2_prot_anno_TwoSampleMR.csv")
TNF_prot_anno    %>% 
  export("data/export/TNF_prot_anno_TwoSampleMR.csv")
TYK2_prot_anno   %>% 
  export("data/export/TYK2_prot_anno_TwoSampleMR.csv")
```


```{r, results = 'hide'}

#Sanity check: no warnings
# read_exposure_data("data/export/CXCL10_prot_anno_TwoSampleMR.csv", sep = ",")
# read_exposure_data("data/export/IL12B_prot_anno_TwoSampleMR.csv", sep = ",")
# read_exposure_data("data/export/IL2RA_prot_anno_TwoSampleMR.csv", sep = ",")
# read_exposure_data("data/export/IL2RB_prot_anno_TwoSampleMR.csv", sep = ",")
# read_exposure_data("data/export/IL2RG_prot_anno_TwoSampleMR.csv", sep = ",")
# # read_exposure_data("data/export/IL6ST_prot_anno_TwoSampleMR.csv", sep = ",")
# read_exposure_data("data/export/JAK2_prot_anno_TwoSampleMR.csv", sep = ",")
# read_exposure_data("data/export/TNF_prot_anno_TwoSampleMR.csv", sep = ",")
# read_exposure_data("data/export/TYK2_prot_anno_TwoSampleMR.csv", sep = ",")
```

# Outcome data

## Import outcome data


```{r}
IL12B_T1D   <- import("data/cis_sumstats/IL12B_T1D.rds")  
CXCL10_T1D  <- import("data/cis_sumstats/CXCL10_T1D.rds")  
TYK2_T1D    <- import("data/cis_sumstats/TYK2_T1D.rds")
TNF_T1D     <- import("data/cis_sumstats/TNF_T1D.rds")  
JAK2_T1D    <- import("data/cis_sumstats/JAK2_T1D.rds")  
IL6ST_T1D   <- import("data/cis_sumstats/IL6ST_T1D.rds") 
IL6R_T1D    <- import("data/cis_sumstats/IL6R_T1D.rds")  
IL2RB_T1D   <- import("data/cis_sumstats/IL2RB_T1D.rds")  
IL2RG_T1D   <- import("data/cis_sumstats/IL2RG_T1D.rds")  
IL2RA_T1D   <- import("data/cis_sumstats/IL2RA_T1D.rds")  
```

There were no missing rsid values (suppressed)

```{r, results = 'hide'}
#every 
IL12B_T1D  %>% pull(hm_rsid) %>% anyNA()
CXCL10_T1D %>% pull(hm_rsid) %>% anyNA()
TYK2_T1D   %>% pull(hm_rsid) %>% anyNA() 
TNF_T1D    %>% pull(hm_rsid) %>% anyNA()
JAK2_T1D   %>% pull(hm_rsid) %>% anyNA()
# IL6ST_T1D  %>% pull(hm_rsid) %>% anyNA()
IL6R_T1D   %>% pull(hm_rsid) %>% anyNA()
IL2RB_T1D  %>% pull(hm_rsid) %>% anyNA()
IL2RG_T1D  %>% pull(hm_rsid) %>% anyNA()
IL2RA_T1D  %>% pull(hm_rsid) %>% anyNA()
```

## Remove duplicated SNPs

All data frames had duplicated rsids (suppressed). They are
removed.
```{r, results = 'hide'}
IL12B_T1D  %>% pull(hm_rsid) %>% anyDuplicated()
CXCL10_T1D %>% pull(hm_rsid) %>% anyDuplicated()
TYK2_T1D   %>% pull(hm_rsid) %>% anyDuplicated() 
TNF_T1D    %>% pull(hm_rsid) %>% anyDuplicated()
JAK2_T1D   %>% pull(hm_rsid) %>% anyDuplicated()
# IL6ST_T1D  %>% pull(hm_rsid) %>% anyDuplicated()
IL6R_T1D   %>% pull(hm_rsid) %>% anyDuplicated()
IL2RB_T1D  %>% pull(hm_rsid) %>% anyDuplicated()
IL2RG_T1D  %>% pull(hm_rsid) %>% anyDuplicated()
IL2RA_T1D  %>% pull(hm_rsid) %>% anyDuplicated()
```


```{r, results = 'hide'}
duplicated_snps_IL12B  <- IL12B_T1D  %>% 
  filter(duplicated(hm_rsid)) %>% pull(hm_rsid)
duplicated_snps_CXCL10 <- CXCL10_T1D %>% 
  filter(duplicated(hm_rsid)) %>% pull(hm_rsid)
duplicated_snps_TYK2   <- TYK2_T1D   %>% 
  filter(duplicated(hm_rsid)) %>% pull(hm_rsid) 
duplicated_snps_TNF   <- TNF_T1D    %>% 
  filter(duplicated(hm_rsid)) %>% pull(hm_rsid)
duplicated_snps_JAK2   <- JAK2_T1D   %>% 
  filter(duplicated(hm_rsid)) %>% pull(hm_rsid)
#duplicated_snps_IL6ST  <- IL6ST_T1D  %>% 
# filter(duplicated(hm_rsid)) %>% pull(hm_rsid)
duplicated_snps_IL6R   <- IL6R_T1D   %>% 
  filter(duplicated(hm_rsid)) %>% pull(hm_rsid)
duplicated_snps_IL2RB  <- IL2RB_T1D  %>% 
  filter(duplicated(hm_rsid)) %>% pull(hm_rsid)
duplicated_snps_IL2RG  <- IL2RG_T1D  %>% 
  filter(duplicated(hm_rsid)) %>% pull(hm_rsid)
duplicated_snps_IL2RA  <- IL2RA_T1D  %>% 
  filter(duplicated(hm_rsid)) %>% pull(hm_rsid)


IL12B_T1D  <- IL12B_T1D   %>% filter(!hm_rsid %in% duplicated_snps_IL12B)
CXCL10_T1D <- CXCL10_T1D  %>% filter(!hm_rsid %in% duplicated_snps_CXCL10)
TYK2_T1D   <- TYK2_T1D    %>% filter(!hm_rsid %in% duplicated_snps_TYK2) 
TNF_T1D    <- TNF_T1D     %>% filter(!hm_rsid %in% duplicated_snps_TNF)
JAK2_T1D   <- JAK2_T1D    %>% filter(!hm_rsid %in% duplicated_snps_JAK2)
# IL6ST_T1D  <- IL6ST_T1D %>% filter(!hm_rsid %in% duplicated_snps_IL6ST)
IL6R_T1D   <- IL6R_T1D    %>% filter(!hm_rsid %in% duplicated_snps_IL6R)
IL2RB_T1D  <- IL2RB_T1D   %>% filter(!hm_rsid %in% duplicated_snps_IL2RB)
IL2RG_T1D  <- IL2RG_T1D   %>% filter(!hm_rsid %in% duplicated_snps_IL2RG)
IL2RA_T1D  <- IL2RA_T1D   %>% filter(!hm_rsid %in% duplicated_snps_IL2RA)
```


## Write data in TwoSampleMR format

```{r}

#Change to numeric
TNF_T1D <- TNF_T1D %>% mutate(p_value = as.numeric(p_value))

rename_for_TwoSampleMR_o  <- function(data, beta, other_allele, effect_allele,
                                      hm_rsid, hm_beta, standard_error, 
                                      hm_effect_allele_frequency,
                                      hm_effectAllele, hm_otherAllele,
                                      p_value, hm_pos) {
  data %>%
  select(-beta, -other_allele, -effect_allele) %>%
  rename(SNP = hm_rsid, beta = hm_beta, se = standard_error, eaf = hm_effect_allele_frequency,
         effect_allele = hm_effect_allele, other_allele = hm_other_allele,
         pval = p_value, pos = hm_pos) %>%
  mutate(Phenotype = "Risk of type 1 diabetes",
         ncase = 18942, ncontrol = 501638, samplesize = 18942+501638)
  }

CXCL10_T1D  <- CXCL10_T1D %>% rename_for_TwoSampleMR_o()
IL12B_T1D   <- IL12B_T1D  %>% rename_for_TwoSampleMR_o()
IL2RA_T1D   <- IL2RA_T1D  %>% rename_for_TwoSampleMR_o()
IL2RB_T1D   <- IL2RB_T1D  %>% rename_for_TwoSampleMR_o()
IL2RG_T1D   <- IL2RG_T1D  %>% rename_for_TwoSampleMR_o()
# IL6ST_T1D <- IL6ST_T1D  %>% rename_for_TwoSampleMR_o()
JAK2_T1D    <- JAK2_T1D   %>% rename_for_TwoSampleMR_o()
TNF_T1D     <- TNF_T1D    %>% rename_for_TwoSampleMR_o()
TYK2_T1D    <- TYK2_T1D   %>% rename_for_TwoSampleMR_o()
```

Write data in TwoSampleMR data format
```{r}
CXCL10_T1D %>%
  export("data/export/CXCL10_T1D_TwoSampleMR.csv")
IL12B_T1D  %>% 
  export("data/export/IL12B_T1D_TwoSampleMR.csv")
IL2RA_T1D  %>% 
  export("data/export/IL2RA_T1D_TwoSampleMR.csv")
IL2RB_T1D  %>% 
  export("data/export/IL2RB_T1D_TwoSampleMR.csv")
IL2RG_T1D  %>% 
  export("data/export/IL2RG_T1D_TwoSampleMR.csv")
# IL6ST_T1D  %>%
#   export("data/export/IL6ST_T1D_TwoSampleMR.csv")
JAK2_T1D   %>% 
  export("data/export/JAK2_T1D_TwoSampleMR.csv")
TNF_T1D    %>% 
  export("data/export/TNF_T1D_TwoSampleMR.csv")
TYK2_T1D   %>% 
  export("data/export/TYK2_T1D_TwoSampleMR.csv")
```

```{r, results='hide'}
#Sanity check: no warnings
# read_outcome_data("data/export/CXCL10_T1D_TwoSampleMR.csv", sep = ",")
# read_outcome_data("data/export/IL12B_T1D_TwoSampleMR.csv", sep = ",")
# read_outcome_data("data/export/IL2RA_T1D_TwoSampleMR.csv", sep = ",")
# read_outcome_data("data/export/IL2RB_T1D_TwoSampleMR.csv", sep = ",")
# read_outcome_data("data/export/IL2RG_T1D_TwoSampleMR.csv", sep = ",")
# # read_outcome_data("data/export/IL6ST_T1D_TwoSampleMR.csv", sep = ",")
# read_outcome_data("data/export/JAK2_T1D_TwoSampleMR.csv", sep = ",")
# read_outcome_data("data/export/TNF_T1D_TwoSampleMR.csv", sep = ",")
# read_outcome_data("data/export/TYK2_T1D_TwoSampleMR.csv", sep = ",")
```

