---
title: "IL2RA"
author: "Tea Heikkil√§, Emilia Kaiser & Jaakko Koskenniemi"
output: html_document
date: '2022-05-03'
editor_options:
  chunk_output_type: console
---

Load packages and import data
```{r}
library(tidyverse)
library(data.table)
library(coloc)
library(TwoSampleMR)
library(ggpubr)
library(ieugwasr)
library(rio)
library(here)

here::i_am("analyses/IL2RA.Rmd")
```


# eQTL


```{r}
IL2RA_eqtl <- read_exposure_data("data/export/IL2RA_eqtl_TwoSampleMR.csv", sep=",")
IL2RA_T1D  <- read_outcome_data("data/export/IL2RA_T1D_TwoSampleMR.csv", sep=",")
IL2RA      <- harmonise_data(IL2RA_eqtl, IL2RA_T1D)

```



## Manhattan plots
```{r}
#create Manhattan plots with vertical lines indicating the selected area (see below 2.4)
IL2RA_eqtl.fig <-
  IL2RA %>%
  ggplot(aes(x = pos.exposure/1000, y=-log10(pval.exposure))) +
  geom_point()+
  geom_hline(yintercept=8, linetype="dashed")+
  ggtitle("GWAS of IL2RA gene expression in blood") +
  ylab("-log10(p)") +
  xlab(NULL)

IL2RA_T1D.fig <-
  IL2RA %>%
  ggplot()+
  geom_point(mapping=aes(x = pos.exposure/1000, y=-log10(pval.outcome))) +
  geom_hline(yintercept=8, linetype="dashed")+
  # geom_vline(xintercept=36.9e6/1000, linetype="dashed")+
  # geom_vline(xintercept=37.3e6/1000, linetype="dashed")+
  ggtitle("GWAS of risk of T1D") +
  ylab("-log10(p)") +
  xlab("position(kbp)")

#Obtain range for figure of gene
layer_scales(IL2RA_T1D.fig)$x$range$range

IL2RA_gene.fig <-
  ggplot(data = IL2RA) +
  geom_blank() +
  geom_segment(x=37521.878, xend=37571.094, y=1, yend=1, linewidth = 2) +
  annotate("text", label = "IL2RA",
           x= (37521.878 + 37571.094) / 2, y=1, hjust = 0.5, vjust = -1) +
  xlim(36521.878, 38571.094) +
  ylim(0.75, 2) +
  ylab(NULL) + xlab(NULL) +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank())

fig_IL2RA_manhattan <- ggarrange(IL2RA_gene.fig, IL2RA_eqtl.fig, IL2RA_T1D.fig,
                                 heights = c(1, 3, 3), nrow = 3,
                                 ncol = 1, align = "hv")
# ggsave("figures/IL2RA-manhattan-combined-fig.pdf",
#        height = 6, width = 8, units = "in")
```

## Coloc assuming 1 causal variant
```{r}
#recode MAF
IL2RA <- IL2RA %>%
  mutate(maf.exposure  = ifelse(eaf.exposure  < 0.5, eaf.exposure,  1-eaf.exposure)) %>%
  mutate(maf.outcome  = ifelse(eaf.outcome  < 0.5, eaf.outcome,  1-eaf.outcome))

colnames(IL2RA)

D1_eqtl <- list(
  type = "quant", # quantitative trait
  pvalues = IL2RA$pval.exposure, 
  N = IL2RA$samplesize.exposure,
  MAF = IL2RA$maf.exposure,
  pos = IL2RA$pos.exposure,
  snp = IL2RA$pos.exposure,
  sdY = 1)

D2_eqtl <- list(
  type = "cc", # case-control trait
  Beta = IL2RA$beta.outcome,
  varBeta = IL2RA$se.outcome,
  pvalues = IL2RA$pval.outcome,
  N = 18942+501638, # Case-control study (Chiou et al. 2021 Nature)
  s = 18942/(18942+501638), # N_case/(N_case+ N_ctrl)
  MAF = IL2RA$maf.exposure,  #eqtl used here in purpose
  pos = IL2RA$pos.outcome,
  snp = IL2RA$pos.outcome)

check_dataset(D1_eqtl)
check_dataset(D2_eqtl)

coloc_IL2RA <- coloc.abf(D1_eqtl, D2_eqtl, p1 = 1e-4, p2 = 1e-4, p12 = 1e-5)
sensitivity(coloc_IL2RA, "H4 > 0.9")
```



# pQTL

```{r}
IL2RA_prot <- read_exposure_data("data/export/IL2RA_prot_anno_TwoSampleMR.csv", sep=",")
IL2RA <- harmonise_data(IL2RA_prot, IL2RA_T1D)
```


## Manhattan plots
```{r}
#create Manhattan plots with vertical lines indicating the selected area (see below 2.4)
IL2RA_prot.fig <-
  IL2RA %>%
  ggplot(aes(x = pos.exposure/1000, y=-log10(pval.exposure))) +
  geom_point()+
  geom_hline(yintercept=8, linetype="dashed")+
  geom_vline(xintercept=5.9e3, linetype="dashed") +
  geom_vline(xintercept=6.25e3, linetype="dashed") +
  ggtitle("GWAS of IL2RA in serum") +
  ylab("-log10(p)") +
  xlab(NULL)

IL2RA_T1D.fig <-
  IL2RA %>%
  ggplot()+
  geom_point(mapping=aes(x = pos.exposure/1000, y=-log10(pval.outcome))) +
  geom_hline(yintercept=8, linetype="dashed") +
  geom_vline(xintercept=5.9e3, linetype="dashed") +
  geom_vline(xintercept=6.25e3, linetype="dashed") +
  ggtitle("GWAS of risk of T1D") +
  ylab("-log10(p)") +
  xlab("position(kbp)")

#Obtain range for figure of gene
layer_scales(IL2RA_T1D.fig)$x$range$range

IL2RA_gene.fig <-
  ggplot(data = IL2RA) +
  geom_blank() +
  geom_segment(x=37521.878, xend=37571.094, y=1, yend=1, linewidth = 2) +
  annotate("text", label = "IL2RA",
           x= (37521.878 + 37571.094) / 2, y=1, hjust = 0.5, vjust = -1) +
  xlim(36521.878, 38571.094) +
  ylim(0.75, 2) +
  ylab(NULL) + xlab(NULL) +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank())

fig_IL2RA_manhattan <- ggarrange(IL2RA_gene.fig, IL2RA_prot.fig, IL2RA_T1D.fig,
                                 heights = c(1, 3, 3), nrow = 3,
                                 ncol = 1, align = "hv")
fig_IL2RA_manhattan
# ggsave("figures/IL2RA-manhattan-combined-fig.pdf",
#        height = 6, width = 8, units = "in")
```

## Coloc assuming 1 causal variant
```{r}
#select the area containing significant SNPs
# IL2RA <- IL2RA %>%
#   filter(pos.exposure > 5.9e6 & pos.exposure < 6.25e6)


IL2RA <- IL2RA %>%
  mutate(maf.exposure  = ifelse(eaf.exposure  < 0.5, eaf.exposure,  1-eaf.exposure))

D1_pqtl <- list(
  type = "quant", # quantitative trait
  Beta = IL2RA$beta.exposure,
  varBeta = IL2RA$se.exposure^2,
  pvalues = IL2RA$pval.exposure, #What is this _origin_?
  N = IL2RA$samplesize.exposure,
  MAF = IL2RA$maf.exposure,
  pos = IL2RA$pos.exposure,
  snp = IL2RA$pos.exposure,
  sdY = 1)

D2_pqtl <- list(
  type = "cc", # case-control trait
  Beta = IL2RA$beta.outcome,
  varBeta = IL2RA$se.outcome,
  pvalues = IL2RA$pval.outcome, #why _origin_?
  N = 18942+501638, # Case-control study (Chiou et al. 2021 Nature)
  s = 18942/(18942+501638), # N_case/(N_case+ N_ctrl)
  MAF = IL2RA$maf.exposure,  #prot used here in purpose
  pos = IL2RA$pos.outcome,
  snp = IL2RA$pos.outcome)

check_dataset(D1_pqtl)
check_dataset(D2_pqtl)

coloc_IL2RA <- coloc.abf(D1_pqtl, D2_pqtl, p1 = 1e-4, p2 = 1e-4, p12 = 1e-5)
sensitivity(coloc_IL2RA, "H4 > 0.1")
```

## coloc.susie

### Get LD matrix
```{r}

# ldmat_IL2RA <- ld_matrix_local(
#     IL2RA$SNP,
#     plink_bin = genetics.binaRies::get_plink_binary(),
#     bfile = "C:/Users/jajoko/Documents/MR Projects/plinkref/EUR")
# 
# save(ldmat_IL2RA, file = "data/IL2RA_LDmat-2023-04-25.RData")
load(file = "data/IL2RA_LDmat-2023-04-25.RData")

str(ldmat_IL2RA)
```


### Harmonize correlation matrix

Change data from correlation matrix to pairwise correlation format
```{r}
#Data is changed to data.table so dtplyr can be used for better perfomance
ldmat_IL2RA <- cbind(rownames(ldmat_IL2RA), ldmat_IL2RA) %>%
  as.data.table() %>%
  pivot_longer(names_to = "variant2",
               values_to = "r2",
               cols = !matches("V1")) %>%
  rename("variant1" = "V1") %>%
  mutate(r2 = as.numeric(r2))
```

Identify effect orientation and remove effect orientation from variant names
```{r}

#Extract SNPs and get rid of duplicates
ldmat_IL2RA_snps <- ldmat_IL2RA %>%
  as_tibble() %>%
  select(variant1) %>%
  filter(duplicated(variant1) == FALSE)

#Extract EA and NEA
ldmat_IL2RA_snps <- ldmat_IL2RA_snps %>%
  mutate(EA_ld  = word(variant1, 2, sep = fixed('_'))) %>%
  mutate(NEA_ld  = word(variant1, 3, sep = fixed('_'))) %>%
  mutate(rsid = word(variant1, 1, sep=fixed('_')))
```

Correct the orientation of effect alleles in IL2RA data frame
```{r}
#remove SNPs from original (IL2RA) data that are not included in LD
#correlation matrix
IL2RA <- IL2RA %>%
  filter(SNP %in% ldmat_IL2RA_snps$rsid)

#join SNP alignment data from the LD matrix to original (IL2RA) data
#which includes betas and alignment data for the risk of T1D and protein
#levels
IL2RA <- IL2RA %>%
  full_join(ldmat_IL2RA_snps, by = c("SNP" = "rsid"))

#Look if there are unlikely cases
#variants that have different effect allele to ref allele combination
IL2RA %>% filter(EA_ld != effect_allele.exposure &
                   NEA_ld != other_allele.exposure) #no
IL2RA %>% filter(NEA_ld != other_allele.exposure &
                   EA_ld != effect_allele.exposure) #no

#If no, harmonize the alignment of Betas with correlations coefficients in LD correlation matrix
IL2RA <- IL2RA %>%
  mutate(harmonized = ifelse(EA_ld != effect_allele.exposure |
                               EA_ld != effect_allele.outcome, 1, 0))
IL2RA %>% filter(harmonized == 1) #0

```

No variants were harmonized. Remove variants that are not found in IL2RA data frame

```{r}

#Change to data.table for better performance in following tasks
ldmat_IL2RA <- ldmat_IL2RA %>% as.data.table()

#Remove orientation from rsids (this time using data.table because of computational reasons)
ldmat_IL2RA <- ldmat_IL2RA[,rsid1 := gsub('_[ATGC]*_[ATGC]*','',variant1)]
ldmat_IL2RA <- ldmat_IL2RA[,rsid2 := gsub('_[ATGC]*_[ATGC]*','',variant2)]

## Proof that the approach above works (i.e. there are no characters
## in rsids)
# ldmat_IL2RA_unique <- unique(ldmat_IL2RA, by = "rsid1")
# ldmat_IL2RA_unique <- ldmat_IL2RA_unique[,rsid1 := gsub('rs','',rsid1)]
# ldmat_IL2RA_unique <- ldmat_IL2RA_unique[,rsid1 := as.numeric(rsid1)]
# ldmat_IL2RA_unique[is.numeric(rsid1) == FALSE,]
# rm(ldmat_IL2RA_unique)

#Remove variants that are not found in IL2RA data frame
ldmat_IL2RA <- ldmat_IL2RA[rsid1 %in% IL2RA$SNP |
                             rsid2 %in% IL2RA$SNP]
length(unique(ldmat_IL2RA$rsid1)) #1460, the correct #

```


Format the pairwise LD matrix back to LD correlation matrix form
```{r}
ldmat_IL2RA_corr <- ldmat_IL2RA  %>%
  as_tibble() %>%
  select(-variant1, -variant2) %>%
  pivot_wider(names_from = "rsid2", values_from = "r2")

#test for missing
ldmat_IL2RA_corr %>% anyNA() #none

dim(ldmat_IL2RA)
```

Format LD-matrix
```{r}
rownames <- ldmat_IL2RA_corr$rsid1
ldmat_IL2RA_corr <- ldmat_IL2RA_corr %>%
  select(-rsid1)

ldmat_IL2RA_corr <- ldmat_IL2RA_corr %>%
  as.matrix()

dimnames(ldmat_IL2RA_corr)[1] <- list(rownames)
```

### Prepare coloc data & run coloc.susie

```{r}
D3 <- list(
  type = "quant", # quantitative trait
  beta = IL2RA$beta.exposure,
  varbeta = IL2RA$se.exposure^2,
  pvalues = IL2RA$pval.exposure,
  n = IL2RA$samplesize.exposure,
  MAF = IL2RA$maf.exposure,
  position = IL2RA$pos.exposure,
  snp = IL2RA$SNP,
  LD = ldmat_IL2RA_corr,
  sdY = 1)

D4 <- list(
  type = "cc", # case-control trait
  beta = IL2RA$beta.outcome,
  varbeta = IL2RA$se.outcome^2,
  pvalues = IL2RA$pval.outcome,
  n = 18942+501638, # Case-control study (Chiou et al. 2021 Nature)
  s = 18942/(18942+501638), # N_case/(N_case+ N_ctrl)
  MAF = IL2RA$maf.exposure,
  LD = ldmat_IL2RA_corr,
  position = IL2RA$pos.outcome,
  snp = IL2RA$SNP)

check_dataset(D3, req = c("beta", "varbeta", "LD", "snp"))
check_dataset(D4, req = c("beta", "varbeta", "LD", "snp"))

check_alignment(D3)
check_alignment(D4)

```


### Run coloc.susie
```{r}
S3 <- runsusie(D3, n = IL2RA$samplesize.exposure[1])
S4 <- runsusie(D4, n = 18942+501638)

summary(S3)
summary(S4)

if(requireNamespace("susieR",quietly=TRUE)) {
  susie.res=coloc.susie(S3,S4)
  print(susie.res$summary)
}

if(requireNamespace("susieR",quietly=TRUE)) {
  sensitivity(susie.res,"H4 > 0.7",row=1,dataset1=D3,dataset2=D4)
  sensitivity(susie.res,"H4 > 0.7",row=2,dataset1=D3,dataset2=D4)
  sensitivity(susie.res,"H4 > 0.7",row=3,dataset1=D3,dataset2=D4)
  sensitivity(susie.res,"H4 > 0.7",row=4,dataset1=D3,dataset2=D4)
}

susie.res

```


## MR


### Import GWAS of protein and T1D risk
```{r}

#Import data

IL2RA_clumped <- clump_data(IL2RA)
MR_IL2RA_res <- TwoSampleMR::mr(IL2RA_clumped)

MR_IL2RA_res_scatter <- mr_scatter_plot(MR_IL2RA_res, IL2RA_clumped)
MR_IL2RA_res_single <- mr_singlesnp(IL2RA_clumped)
MR_IL2RA_res_forest <- mr_forest_plot(MR_IL2RA_res_single)

exp(MR_IL2RA_res[3,]$b)
exp(MR_IL2RA_res[3,]$b)
exp(MR_IL2RA_res[3,]$b - 1.96*MR_IL2RA_res[3,]$se)
exp(MR_IL2RA_res[3,]$b + 1.96*MR_IL2RA_res[3,]$se)

```


