---
title: "IL6R"
author: "Tea Heikkil√§, Emilia Kaiser & Jaakko Koskenniemi"
output: html_document
date: '2022-26-04'
editor_options:
  chunk_output_type: console
---

  Load packages and import data
```{r}
pacman::p_load(tidyverse, data.table, coloc, TwoSampleMR, ggpubr, 
               ieugwasr, rio, here)

here::i_am("analyses/IL6R.Rmd")

source("code/Functions.R")

```

# eQTL

```{r}
IL6R_eqtl <- import_data("data/export/IL6R_eqtl_TwoSampleMR.csv", 
                         "data/export/IL6R_T1D_TwoSampleMR.csv", has_beta = TRUE)

```


## Manhattan plots
```{r}
manhattan_plots(IL6R_eqtl, gene_start=154405193, gene_end=154469450, title = "IL6R")
IL6R_eqtl <- IL6R_eqtl %>% filter(pos.outcome > 154300000 & pos.outcome < 154700000) 
manhattan_plots(IL6R_eqtl, gene_start=154405193, gene_end=154469450, title = "IL6R")

ggsave("figures/IL6R-eQTL-T1D-manhattan.png", height = 6, width = 6)

```

## Coloc assuming 1 causal variant
```{r}
coloc_results_IL6R_eqtl <- analyze_coloc_eqtl(IL6R_eqtl)
# coloc_results_IL6R_eqtl$coloc_D1D2 #coloc results
# coloc_results_IL6R_eqtl$D1 #data eQTL
# coloc_results_IL6R_eqtl$D2 #data t1d

```

## Regional manhattan plot

```{r}

load("data/export/IL6R_LDmat.RData")

plot_coloc_IL6R(genechr = 1,
           fig_start = 154300000,
           fig_end = 154700000,
           coloc_results = coloc_results_IL6R_eqtl$coloc_D1D2,
           data = IL6R_eqtl,
           D1 = coloc_results_IL6R_eqtl$D1,
           D2 = coloc_results_IL6R_eqtl$D2,
           LDmat = IL6R)

ggsave(filename = "IL6R.png", height = 8, width = 6)
```

## MR
```{r}

#Find top SNP
coloc_results_IL6R_eqtl$coloc_D1D2$results$snp[which.max(coloc_results_IL6R_eqtl$coloc_D1D2$results$SNP.PP.H4)] #rs10908839

#MR
IL6R_eqtl_rs10908839   <- IL6R_eqtl %>% filter(SNP == "rs10908839")

exp(IL6R_eqtl_rs10908839$beta.outcome)
exp(IL6R_eqtl_rs10908839$beta.outcome - 1.96 * IL6R_eqtl_rs10908839$se.outcome)
exp(IL6R_eqtl_rs10908839$beta.outcome + 1.96 * IL6R_eqtl_rs10908839$se.outcome)

MR_IL6R_eqtl_rs10908839   <- TwoSampleMR::mr(IL6R_eqtl_rs10908839)

MR_IL6R_eqtl_rs10908839

exp(MR_IL6R_eqtl_rs10908839$b)
exp(MR_IL6R_eqtl_rs10908839$b - 1.96*MR_IL6R_eqtl_rs10908839$se)
exp(MR_IL6R_eqtl_rs10908839$b + 1.96*MR_IL6R_eqtl_rs10908839$se)
```


## Functional variants

```{r}
load("data/export/phenoscanner-functional-variants-rsids.RDS")

IL6R_eqtl_orig <- read_csv("data/export/IL6R_eqtl_TwoSampleMR.csv")

functional_variants <- as.data.frame(functional_variants)
IL6R_eqtl_orig %>%
 filter(SNP %in% filter(functional_variants, gene == "IL6R")$rsid) #only 1

IL6R_eqtl %>%
 filter(SNP %in% filter(functional_variants, gene == "IL6R")$rsid) #only 1


IL6R_eqtl_orig <- read_csv("data/export/IL6R_eqtl_TwoSampleMR.csv")
IL6R_eqtl_orig %>%
  filter(SNP == "154454556") #no

IL6R_T1D <- read_csv("data/export/IL6R_T1D_TwoSampleMR.csv")
IL6R_T1D %>%
  filter(SNP == "154454556") #no



```



## MR
```{r}

IL6R_eqtl_rs72650666   <- IL6R_eqtl %>% filter(SNP == "rs72650666")
MR_IL6R_eqtl_rs72650666   <- TwoSampleMR::mr(IL6R_eqtl_rs72650666)

MR_IL6R_eqtl_rs72650666

exp(MR_IL6R_eqtl_rs72650666$b)
exp(MR_IL6R_eqtl_rs72650666$b - 1.96*MR_IL6R_eqtl_rs72650666$se)
exp(MR_IL6R_eqtl_rs72650666$b + 1.96*MR_IL6R_eqtl_rs72650666$se)
```



# CRP

```{r}
IL6R_pqtl <- import_data("data/export/IL6R_crp_TwoSampleMR.csv",
                        "data/export/IL6R_T1D_TwoSampleMR.csv", has_beta = TRUE)

IL6R_pqtl %>%
  filter(SNP %in% filter(functional_variants, gene == "IL6R")$rsid) %>%
  mr()


```



## Manhattan plots
```{r}

#manhattan_plots(IL6R_pqtl, gene_start = 154377669, gene_end = 154441926,
#                title = "IL6R")
# ggsave("figures/IL6R-manhattan-combined-fig.pdf",
#        height = 6, width = 8, units = "in")
```

## Coloc assuming 1 causal variant
```{r}
#analyze_coloc_pqtl(IL6R_pqtl)
# pdf("figures/IL6R-eQTL-T1D-coloc.pdf", height = 6, width = 8)
# dev.off()
```

## MR

```{r}

# IL6R_pqtl_clumped     <- clump_data(IL6R_pqtl)
# IL6R_pqtl_rs7536152   <- IL6R_pqtl %>% filter(SNP == "rs7536152")
# IL6R_pqtl_rs10908839  <- IL6R_pqtl %>% filter(SNP == "rs10908839")
# 
# MR_IL6R_res             <- TwoSampleMR::mr(IL6R_pqtl_clumped)
# MR_IL6R_res_rs7536152   <- TwoSampleMR::mr(IL6R_pqtl_rs7536152)
# MR_IL6R_res_rs10908839  <- TwoSampleMR::mr(IL6R_pqtl_rs10908839)
# 
# mr_scatter_plot(MR_IL6R_res, IL6R_pqtl_clumped)
# MR_IL6R_res_single <- mr_singlesnp(IL6R_pqtl_clumped)
# mr_forest_plot(MR_IL6R_res_single)
# 
# MR_IL6R_res
# MR_IL6R_res_scatter
# MR_IL6R_res_single
# MR_IL6R_res_forest
# 
# exp(MR_IL6R_res[3,]$b)
# exp(MR_IL6R_res[3,]$b - 1.96*MR_IL6R_res[3,]$se)
# exp(MR_IL6R_res[3,]$b + 1.96*MR_IL6R_res[3,]$se)

```


## LD matrix with previous SNPs

```{r}

ld_matrix_l() %>%
  
  signif(2) %>%
  
IL6R_LD_EUR <- ld_matrix(
    variants = c("rs10908839", "rs2229238", "rs2228145"),
    pop = "EUR",
    plink_bin = genetics.binaRies::get_plink_binary(),
    bfile = "C:/Users/jajoko/Documents/plinkref/EUR")

IL6R_LD_AFR <- ld_matrix(
    variants = c("rs10908839", "rs2229238", "rs2228145"),
    pop = "AFR",
    plink_bin = genetics.binaRies::get_plink_binary(),
    bfile = "C:/Users/jajoko/Documents/plinkref/AFR")

IL6R_LD_SAS <- ld_matrix(
    variants = c("rs10908839", "rs2229238", "rs2228145"),
    pop = "SAS",
    plink_bin = genetics.binaRies::get_plink_binary(),
    bfile = "C:/Users/jajoko/Documents/plinkref/SAS")

IL6R_LD_AMR <- ld_matrix(
    variants = c("rs10908839", "rs2229238", "rs2228145"),
    pop = "SAS",
    plink_bin = genetics.binaRies::get_plink_binary(),
    bfile = "C:/Users/jajoko/Documents/plinkref/AMR")

IL6R_LD_EAS <- ld_matrix(
    variants = c("rs10908839", "rs2229238", "rs2228145"),
    pop = "EAS",
    plink_bin = genetics.binaRies::get_plink_binary(),
    bfile = "C:/Users/jajoko/Documents/plinkref/EAS")

IL6R_LD <- ld_matrix(
    variants = c("rs10908839", "rs2229238", "rs2228145"),
    plink_bin = genetics.binaRies::get_plink_binary(),
    bfile = "C:/Users/jajoko/Documents/plinkref/")

IL6R_LD <- ld_matrix(
    variants = c("rs10908839", "rs2229238", "rs2228145"))

signif(IL6R_LD_EUR * IL6R_LD_EUR, 2)


LD_across <- ld_matrix(
    variants = c("rs41316003", "rs2304256", "rs141500365"))



```




