---
title: "IL6ST"
author: "Tea Heikkil√§, Emilia Kaiser & Jaakko Koskenniemi"
output: html_document
date: '2022-24-04'
editor_options:
  chunk_output_type: console
---

  Load packages and import data
```{r}
pacman::p_load(tidyverse, data.table, coloc, TwoSampleMR, ggpubr, 
               ieugwasr, rio, here)
source("code/Functions.R")
here::i_am("analyses/IL6ST.Rmd")
```


```{r}
IL6ST_pqtl <- import_data("data/export/IL6ST_prot_anno_TwoSampleMR.csv",
                          "data/export/IL6ST_T1D_TwoSampleMR.csv", 
                          has_beta=TRUE)
```

## Manhattan plots

No GWAS hits in T1D. MR is not done.
```{r}
manhattan_plots(IL6ST_pqtl, gene_start = 55935095, gene_end = 55995022,
               title = "IL6ST")

IL6ST_pqtl <- IL6ST_pqtl %>% filter(pos.outcome > 55600000 & pos.outcome < 56200000) 
manhattan_plots(IL6ST_pqtl, gene_start = 55935095, gene_end = 55995022,
               title = "IL6ST")

```


## Coloc

```{r}
coloc_results_IL6ST_pqtl <- analyze_coloc_pqtl(IL6ST_pqtl)


coloc_results_IL6ST_pqtl$coloc_D1D2

```


```{r}

load("data/export/IL6ST_LDmat.RData")

plot_coloc(genechr = 5,
           fig_start = 55600000,
           fig_end = 56200000,
           coloc_results = coloc_results_IL6ST_pqtl$coloc_D1D2,
           data = IL6ST_pqtl,
           D1 = coloc_results_IL6ST_pqtl$D1,
           D2 = coloc_results_IL6ST_pqtl$D2,
           LDmat = IL6ST)

ggsave(filename = "IL6ST.png", height = 8, width = 6)
```


## Functional variants

```{r}
load("data/export/phenoscanner-functional-variants-rsids.RDS")

IL6ST_pqtl <- read_csv("data/export/IL6ST_prot_anno_TwoSampleMR.csv")
IL6ST_pqtl %>%
  filter(SNP == "rs61748224") #yes

IL6ST_T1D <- read_csv("data/export/IL6ST_T1D_TwoSampleMR.csv")
IL6ST_T1D %>%
  filter(SNP == "rs61748224") #yes


IL6ST_pqtl_funcvar <- IL6ST_pqtl %>%
  filter(SNP %in% c("rs61748224", "rs61755738", "rs34417936", "rs141500365"))

IL6ST_pqtl_funcvar %>%
  select(SNP, beta.outcome, se.outcome, pval.outcome, beta.exposure, pval.exposure) %>%
  mutate(across(!matches("SNP"), sig2))

IL6ST_pqtl_funcvar_MR <- IL6ST_pqtl_funcvar %>%
  filter(pval.exposure < 1e-4) %>% 
  TwoSampleMR::mr()

IL6ST_pqtl_funcvar_MR

exp(IL6ST_pqtl_funcvar_MR$b)
exp(IL6ST_pqtl_funcvar_MR$b - 1.96*IL6ST_pqtl_funcvar_MR$se)
exp(IL6ST_pqtl_funcvar_MR$b + 1.96*IL6ST_pqtl_funcvar_MR$se)

```
