---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
bibliography: references.bib 
editor_options:
  chunk_output_type: console
---

```{r}
library(tidyverse)
```

# Introduction

A research project aimed to evaluate the available genetic evidence
for efficacy and safety of existing drugs to prevent type 1 diabetes.

```{r, echo = FALSE}
data.frame("gene"       = c("IFNAR2", "IL12B", "IL23A", "IL2RA", "IL2RB", 
                          "IL2RG",  "IL6R",  "IL6ST", "JAK1",  "JAK2", 
                          "JAK3",   "TYK2"),
           "protein"    = c("p40",    "",      "", "IL-2Rα/CD25", "IL-2Rβ/CD122",
                          "γc/CD132", "CRP as a proxy", "Gp130", "", "",
                          "", ""),
           "T1D"   = c("", "0", "", "X", "X", 
                            "0", "(X)", "0", "(X)", "X", 
                            "X", "X"),
           "pQTL"       = c("", "X", "", "X", "X", 
                            "0", "X", "X", "-", "0", 
                            "-", "0"),
           "eQTL"       = c("", "", "", "X", "X", 
                            "", "X", "", "X", "X", 
                            "(X)", "X"),
           "coloc_eQTL" = c("", "", "", "H4", "H3", 
                            "", "H4", "", "H3", "H3", 
                            "H3", "H3"),
           "coloc_pQTL" = c("", "", "", "H3", "H3", 
                            "", "H3", "-", "-", "-", 
                            "-", "-")) %>%
  knitr::kable(caption = "Summary of study results")
```

# Build script

Linux script to get data
```{bash, eval = FALSE}

#Get data (in unix)
wget https://molgenis26.gcc.rug.nl/downloads/eqtlgen/cis-eqtl/2019-12-11-cis-eQTLsFDR0.05-ProbeLevel-CohortInfoRemoved-BonferroniAdded.txt.gzwget https://molgenis26.gcc.rug.nl/downloads/eqtlgen/cis-eqtl/2019-12-11-cis-eQTLsFDR0.05-ProbeLevel-CohortInfoRemoved-BonferroniAdded.txt.gz
wget https://molgenis26.gcc.rug.nl/downloads/eqtlgen/cis-eqtl/2019-12-11-cis-eQTLsFDR-ProbeLevel-CohortInfoRemoved-BonferroniAdded.txt.gz
wget https://molgenis26.gcc.rug.nl/downloads/eqtlgen/cis-eqtl/2018-07-18_SNP_AF_for_AlleleB_combined_allele_counts_and_MAF_pos_added.txt.gz

#decompress data
zcat -d 2019-12-11-cis-eQTLsFDR0.05-ProbeLevel-CohortInfoRemoved-BonferroniAdded.txt.gz > cis-EQTL-FDR0.05.txt
zcat -d 2019-12-11-cis-eQTLsFDR-ProbeLevel-CohortInfoRemoved-BonferroniAdded.txt.gz > cis-EQTL-full.txt
zcat -d 2018-07-18_SNP_AF_for_AlleleB_combined_allele_counts_and_MAF_pos_added.txt.gz > cis-EQTL-AF.txt
```


Targets data script
```{r}

```



Build script for workflowr
```{r}
workflowr::wflow_build()
workflowr::wflow_publish(all = TRUE)
```




# Manuscript items


# (Supplementary) Figure: flow chart of the selection of the targets

# (Supplementary) Table: GWAS summary statistics sources


# Table: genomic regions under investigation (gene, chr, start, end, number of variants available for colocalization)

# Table: All colocalization results

# Figures (either separate or different panels): regional Manhattan plots for the colocalizing regions

# (Supplementary Figures: regional Manhattan plots for non-colocalizing regions)

# Table and/or Figure: MR results for the colocalizing regions




