---
title: "JAK2 eQTL"
author: "Tea Heikkil√§, Emilia Kaiser & Jaakko Koskenniemi"
output: html_document
date: '2022-05-03'
editor_options:
  chunk_output_type: console
---

  Load packages and import data
```{r}
pacman::p_load(tidyverse, data.table, coloc,TwoSampleMR, ggpubr, 
               ieugwasr, rio, here)

source("code/Functions.R")

here::i_am("analyses/JAK2.Rmd")

```

# eQTL

Import data 
```{r}
JAK2_eqtl <- import_data("data/export/JAK2_eqtl_TwoSampleMR.csv", 
                         "data/export/JAK2_T1D_TwoSampleMR.csv", has_beta = TRUE)
```

## Initial Manhattan plots
```{r}
manhattan_plots(JAK2_eqtl, gene_start=4985033, gene_end=5128183, title = "JAK2")
JAK2_eqtl <- JAK2_eqtl %>% filter(pos.outcome > 4000000 & pos.outcome < 5500000) 
manhattan_plots(JAK2_eqtl, gene_start=4985033, gene_end=5128183, title = "JAK2")
```

## Coloc assuming 1 causal variant
```{r}
coloc_results_JAK2_eqtl <- analyze_coloc_eqtl(JAK2_eqtl)
# coloc_results_JAK2_eqtl$coloc_D1D2 #coloc results
# coloc_results_JAK2_eqtl$D1 #data eQTL
# coloc_results_JAK2_eqtl$D2 #data t1d

```

## Regional manhattan plot

```{r}

load("data/export/JAK2_LDmat.RData")

plot_coloc_2(genechr = 9,
           fig_start = 4000000,
           fig_end = 5500000,
           coloc_results = coloc_results_JAK2_eqtl$coloc_D1D2,
           data = JAK2_eqtl,
           D1 = coloc_results_JAK2_eqtl$D1,
           D2 = coloc_results_JAK2_eqtl$D2,
           LDmat = JAK2)

ggsave(filename = "JAK2.png", height = 8, width = 6)
```



## Functional variants


```{r}
load("data/export/phenoscanner-functional-variants-rsids.RDS")

functional_variants_JAK2 <- functional_variants %>% filter(gene == "JAK2")



JAK2_eqtl_orig <- read_csv("data/export/JAK2_eqtl_TwoSampleMR.csv")
JAK2_eqtl_orig %>%
  filter(SNP %in% functional_variants_JAK2$rsid) %>%
  select(SNP) #yes, rs41316003

JAK2_T1D <- read_csv("data/export/JAK2_T1D_TwoSampleMR.csv")
JAK2_T1D %>%
  filter(SNP %in% functional_variants_JAK2$rsid) %>%
  select(SNP) #yes, rs143103233, rs150221602, and rs41316003 

JAK2_T1D %>%
  filter(SNP == "rs41316003") #yes

## MR

JAK2_eqtl_rs41316003   <- JAK2_eqtl %>% filter(SNP == "rs41316003")
MR_JAK2_eqtl_rs41316003   <- TwoSampleMR::mr(JAK2_eqtl_rs41316003)

MR_JAK2_eqtl_rs41316003

exp(MR_JAK2_eqtl_rs41316003$b)
exp(MR_JAK2_eqtl_rs41316003$b - 1.96*MR_JAK2_eqtl_rs41316003$se)
exp(MR_JAK2_eqtl_rs41316003$b + 1.96*MR_JAK2_eqtl_rs41316003$se)


```


