---
title: "TYK2"
author: "Tea Heikkil√§, Emilia Kaiser & Jaakko Koskenniemi"
output: html_document
date: '2022-26-04'
editor_options:
  chunk_output_type: console
---

  Load packages and import data
```{r}
pacman::p_load(tidyverse, data.table, coloc, TwoSampleMR, ggpubr, ieugwasr, rio, here)

here::i_am("analyses/TYK2.Rmd")

source("code/Functions.R")

```


#eQTL

Load packages and import data
```{r}

TYK2_eqtl <- import_data("data/export/TYK2_eqtl_TwoSampleMR.csv", 
                         "data/export/TYK2_T1D_TwoSampleMR.csv", has_beta = TRUE)

```

## Initial Manhattan plots
```{r}
manhattan_plots(TYK2_eqtl, gene_start=10350533, gene_end=10380608, title = "TYK2")
TYK2_eqtl <- TYK2_eqtl %>% filter(pos.outcome > 10200000 & pos.outcome < 10700000) 
manhattan_plots(TYK2_eqtl, gene_start=10350533, gene_end=10380608, title = "TYK2")

#ggsave("figures/TYK2-eqtl-manhattan-fig.png",
#       height = 6, width = 6, units = "in")
```

## Coloc assuming 1 causal variant
```{r}
coloc_results_TYK2_eqtl <- analyze_coloc_eqtl(TYK2_eqtl)

```

## Regional manhattan plot

```{r}
load("data/export/TYK2_LDmat.RData")

plot_coloc_2(genechr = 19,
           fig_start = 10200000,
           fig_end = 10700000,
           coloc_results = coloc_results_TYK2_eqtl$coloc_D1D2,
           data = TYK2_eqtl,
           D1 = coloc_results_TYK2_eqtl$D1,
           D2 = coloc_results_TYK2_eqtl$D2,
           LDmat = TYK2)

ggsave(filename = "TYK2.png", height = 8, width = 6)
```

## MR
```{r}

#Find top SNP
coloc_results_TYK2_eqtl$coloc_D1D2$results$snp[which.max(coloc_results_TYK2_eqtl$coloc_D1D2$results$SNP.PP.H4)] #rs34725611

#MR
TYK2_eqtl_rs34725611   <- TYK2_eqtl %>% filter(SNP == "rs34725611")
MR_TYK2_eqtl_rs34725611   <- TwoSampleMR::mr(TYK2_eqtl_rs34725611)

MR_TYK2_eqtl_rs34725611

exp(MR_TYK2_eqtl_rs34725611$b)
exp(MR_TYK2_eqtl_rs34725611$b - 1.96*MR_TYK2_eqtl_rs34725611$se)
exp(MR_TYK2_eqtl_rs34725611$b + 1.96*MR_TYK2_eqtl_rs34725611$se)
```


## Functional variants
```{r}




sig2 <- function(x) signif(x, digits = 2)
TYK2_eqtl %>%
  filter(SNP %in% c("rs55886939",   "rs34536443", "rs35018800",  "rs55882956",
                    "rs12720356", "rs139954108", "rs2304256","rs188833543", 
                    "rs55762744", "rs144960992")) %>%
  select(SNP, beta.outcome, se.outcome, pval.outcome, beta.exposure, pval.exposure) %>%
  mutate(across(!matches(c("SNP", "pval.outcome", "pval.exposure")), sig2))


TYK2_eqtl_funcvar_mr <- TYK2_eqtl %>%
  filter(SNP %in% c("rs12720356", "rs2304256")) %>%
  TwoSampleMR::mr_singlesnp() %>%
  mutate(exp_b = exp(b), 
         lo_95 = exp(b - (1.96 * se)),
         hi_95 = exp(b + (1.96 * se))) %>%
    mutate(across(starts_with(c("b", "se","p", "exp_b", "lo", "hi")), sig2))

  TYK2_eqtl_funcvar_mr
```


```{r}
#LD matrix for interesting variants
functional_variants_TYK2 <- functional_variants %>% 
  filter(gene == "TYK2")

library(ieugwasr)
TYK2_LD <- ld_matrix(
    variants = c(functional_variants_TYK2$rsid, "rs144309607", "rs34725611"),
    pop = "EUR",
    plink_bin = genetics.binaRies::get_plink_binary(),
    bfile = "C:/Users/jajoko/Documents/plinkref/EUR",
    with_alleles = FALSE)

```

