---
title: "TYK2"
author: "Tea Heikkil√§, Emilia Kaiser & Jaakko Koskenniemi"
output: html_document
date: '2022-26-04'
editor_options:
  chunk_output_type: console
---

  Load packages and import data
```{r}
pacman::p_load(tidyverse, data.table, coloc, TwoSampleMR, ggpubr, ieugwasr, rio, here)

here::i_am("analyses/TYK2.Rmd")

source("code/Functions.R")

```


#eQTL

Load packages and import data
```{r}

TYK2_eqtl <- import_data("data/export/TYK2_eqtl_TwoSampleMR.csv", 
                         "data/export/TYK2_T1D_TwoSampleMR.csv", has_beta = FALSE)

```

## Initial Manhattan plots
```{r}
manhattan_plots(TYK2_eqtl, gene_start=10350533, gene_end=10380608, title = "TYK2")
TYK2_eqtl <- TYK2_eqtl %>% filter(pos.outcome > 10200000 & pos.outcome < 10700000) 
manhattan_plots(TYK2_eqtl, gene_start=10350533, gene_end=10380608, title = "TYK2")

#ggsave("figures/TYK2-eqtl-manhattan-fig.png",
#       height = 6, width = 6, units = "in")
```

## Coloc assuming 1 causal variant
```{r}
coloc_results_TYK2_eqtl <- analyze_coloc_eqtl(TYK2_eqtl)

```

## Regional manhattan plot

```{r}
load("data/export/TYK2_LDmat.RData")

plot_coloc_2(genechr = 19,
           fig_start = 10200000,
           fig_end = 10700000,
           coloc_results = coloc_results_TYK2_eqtl$coloc_D1D2,
           data = TYK2_eqtl,
           D1 = coloc_results_TYK2_eqtl$D1,
           D2 = coloc_results_TYK2_eqtl$D2,
           LDmat = TYK2)

# ggsave(filename = "figures/TYK2.png", height = 8, width = 6)
```


```{r}
load("data/export/phenoscanner-functional-variants-rsids.RDS")

TYK2_eqtl_orig <- read_csv("data/export/TYK2_eqtl_TwoSampleMR.csv")
TYK2_eqtl_orig %>%
  filter(SNP %in% filter(functional_variants, gene == "TYK2")$rsid) #only 11

```


```{r}
TYK2_eqtl %>%
  filter(SNP %in% c("rs2304256", "rs12720356", "rs35018800"))
```

