<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jaakko Koskenniemi" />


<title>Script to merge data</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">T1DSCREEN</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="Harmonize.html">Harmonize</a>
</li>
<li>
  <a href="IL2RA.html">IL2RA</a>
</li>
<li>
  <a href="IL2RB.html">IL2RB</a>
</li>
<li>
  <a href="IL2RG.html">IL2RG</a>
</li>
<li>
  <a href="IL6R.html">IL6R</a>
</li>
<li>
  <a href="CRP.html">CRP</a>
</li>
<li>
  <a href="IL12B.html">IL12B</a>
</li>
<li>
  <a href="JAK1.html">JAK1</a>
</li>
<li>
  <a href="JAK2.html">JAK2</a>
</li>
<li>
  <a href="JAK3.html">JAK3</a>
</li>
<li>
  <a href="TYK2.html">TYK2</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jkoskenniemi/T1DSCREEN">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Script to merge data</h1>
<h4 class="author">Jaakko Koskenniemi</h4>
<h4 class="date">2022-24-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-05-06
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>T1DSCREEN/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220313code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220313)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220313code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220313)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjkoskenniemiT1DSCREENtree0d3f37017de4946514ff0df852a8abf92c2d0d7etargetblank0d3f370a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/tree/0d3f37017de4946514ff0df852a8abf92c2d0d7e" target="_blank">0d3f370</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomjkoskenniemiT1DSCREENtree0d3f37017de4946514ff0df852a8abf92c2d0d7etargetblank0d3f370a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/jkoskenniemi/T1DSCREEN/tree/0d3f37017de4946514ff0df852a8abf92c2d0d7e" target="_blank">0d3f370</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/README_cis_eqtl.txt
    Ignored:    data/README_cis_eqtl_AF.txt
    Ignored:    data/cis-EQTL-AF.txt
    Ignored:    data/cis_eqtl_sumstats/2019-12-11-cis-eQTLsFDR-ProbeLevel-CohortInfoRemoved-BonferroniAdded.txt
    Ignored:    data/cis_eqtl_sumstats/2019-12-11-cis-eQTLsFDR0.05-ProbeLevel-CohortInfoRemoved-BonferroniAdded.txt/
    Ignored:    data/cis_eqtl_sumstats/cis-eQTL-SMR_20191212.tar.gz
    Ignored:    data/cis_sumstats/2018-07-18_SNP_AF_for_AlleleB_combined_allele_counts_and_MAF_pos_added.txt
    Ignored:    refs/

Untracked files:
    Untracked:  analyses/IFNAR2.Rmd
    Untracked:  code/Functions.R
    Untracked:  code/IL2RA-säästö.Rmd
    Untracked:  code/IL2RB-säästö-susie-mr.Rmd
    Untracked:  code/IL6R-säästö-susie-mr.R
    Untracked:  data/cis_sumstats/IFNAR2_T1D.rds
    Untracked:  data/export/IFNAR2_T1D_TwoSampleMR.csv
    Untracked:  data/previous-studies/
    Untracked:  figures/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analyses/Harmonize.Rmd</code>) and HTML
(<code>docs/Harmonize.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/0d3f37017de4946514ff0df852a8abf92c2d0d7e/analyses/Harmonize.Rmd" target="_blank">0d3f370</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-05-06
</td>
<td>
functional programming added to result scripts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/b49016a04b5830e2336cc7f1be6dd2ba7b0943b1/docs/Harmonize.html" target="_blank">b49016a</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-05-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/a6eecc26d14712c1121c58759b0a579efed8e76f/docs/Harmonize.html" target="_blank">a6eecc2</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-05-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/c541480d23ea12e6e28d6b3a11d930db54659c36/docs/Harmonize.html" target="_blank">c541480</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-05-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/dae399b977d520e7351f37947062dd773dc5125d/docs/Harmonize.html" target="_blank">dae399b</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-05-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/cecc956cdafd8b7cd8686b43882d1ac6181e4ff0/analyses/Harmonize.Rmd" target="_blank">cecc956</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-05-03
</td>
<td>
Added eQTL data, removed CXCL10 &amp; TNF
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/e0d3987fcbf69cf5b42d345572022434bb91990f/docs/Harmonize.html" target="_blank">e0d3987</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-04-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/cfd36db7f48c142d3da502efdb244319c068660c/docs/Harmonize.html" target="_blank">cfd36db</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/b91e2516e9492e66c55da9d43b75156b2a15873a/docs/Harmonize.html" target="_blank">b91e251</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/c132eaa6f8c552f37374aff1c63b23ee2eb3bc5e/docs/Harmonize.html" target="_blank">c132eaa</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/85188f2ca9b182da9a9113791c5ecaca15e1d0c5/docs/Harmonize.html" target="_blank">85188f2</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/634a9455d60398253a95fceec714c63f01ed3d12/docs/Harmonize.html" target="_blank">634a945</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/9aff4fceef453ed1b475060e03af0d35579b46d3/analyses/Harmonize.Rmd" target="_blank">9aff4fc</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Fixed IL6ST_prot and IL6R_prot, added IL6R_crp, removed one extra
passage of IL2RG_T1D
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/61a6ddce4ed0aefcff000b56e245467a2a12854a/docs/Harmonize.html" target="_blank">61a6ddc</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/258ee8542cbcf7d1df99aef83281b229da8456b3/analyses/Harmonize.Rmd" target="_blank">258ee85</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
fixed IL6R_T1D and IL6ST_T1D
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/c8d146f5d34c2ded365b3a66c1596afc77c502b7/docs/Harmonize.html" target="_blank">c8d146f</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/f0fed8ba64ecc968f1d943bc16181b72b82e6c12/docs/Harmonize.html" target="_blank">f0fed8b</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/0e60a636af7914fea3cce2bc307614b7b862b077/analyses/Harmonize.Rmd" target="_blank">0e60a63</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
fixed an error that all Phenotypes in exposure data were labeled IL2RG
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/4fdafebcd49bf6720555f12ae4e782afded7b00e/docs/Harmonize.html" target="_blank">4fdafeb</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/83b8ba5bb0efac2d85dec5d2664e283026741086/analyses/Harmonize.Rmd" target="_blank">83b8ba5</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-25
</td>
<td>
Added TOC
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/169df25a671f164303bf018173266a2fb7940ae6/docs/Harmonize.html" target="_blank">169df25</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/3b88c4775c28aff2726502067a5dbf511fb05c2d/analyses/Harmonize.Rmd" target="_blank">3b88c47</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-25
</td>
<td>
Quick corrections to enhance how results are printed
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/0c334fc6c19696faa2e61b84183e73cc2ea8f0ca/docs/Harmonize.html" target="_blank">0c334fc</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/64dfa0b12a837aba74be9879982cbf7c099d324c/analyses/Harmonize.Rmd" target="_blank">64dfa0b</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-25
</td>
<td>
First commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>pacman::p_load(tidyverse, rio, here, TwoSampleMR)
here::i_am(&quot;analyses/Harmonize.Rmd&quot;)</code></pre>
<pre><code>here() starts at C:/Users/jajoko/Documents/T1DSCREEN</code></pre>
<p>This script merges the annotation and protein data.</p>
<p>GWAS sumstat of type 1 diabetes was obtained from the study by Chiou
et al.  (Nature 2021, <a
href="https://doi.org/10.1038/s41586-021-03552-w"
class="uri">https://doi.org/10.1038/s41586-021-03552-w</a>). GWAS
sumstat for protein levels of IL2RG was obtained from the study by
Ferkingstad et al.  (Nature 2021, <a
href="https://doi.org/10.1038/s41588-021-00978-w"
class="uri">https://doi.org/10.1038/s41588-021-00978-w</a>).</p>
<p>Variants within 1 megabase before the start site of IL2RG and 1
megabase after the stop site are included. See ‘/code/gather.R’ on how
the the variants within this area were filtered in each GWAS. The
gather-script was run in the high performance computing cluster using
the ‘run_gather.sh’ script. The output of ‘/code/Gather.R’ is saved in
‘/data/cis_sumstats/’, and for each possible target gene there are three
files:</p>
<ol style="list-style-type: decimal">
<li>*_T1D_data.rds: GWAS sumstat for the risk of T1D (Chiou et
al. 2021)</li>
<li>*_prot_data.rds: GWAS sumstat of serum protein level (Ferkingstad et
al. 2021)</li>
<li>*_anno_data.rds: further annotation for GWAS of serum protein
level<br />
Where * is a gene of interest (IL2RA, IL2RG, IL2RG, IL6ST, IL6R,
etc.)</li>
</ol>
<p>except for IL6R for which IL6R_prot_data.rds is substituted with
IL6R_crp_data.rds</p>
<pre class="r"><code>#add here a figure that shows the project workflow</code></pre>
<div id="pqtl-data" class="section level1">
<h1>pQTL data</h1>
<div id="import-pqtl-data" class="section level2">
<h2>Import pQTL data</h2>
<pre class="r"><code>IL2RG_anno  &lt;- import(&quot;data/cis_sumstats/IL2RG_anno.rds&quot;)
IL2RG_prot  &lt;- import(&quot;data/cis_sumstats/IL2RG_gwas.rds&quot;)
IL12B_anno  &lt;- import(&quot;data/cis_sumstats/IL12B_anno.rds&quot;)  
IL12B_prot  &lt;- import(&quot;data/cis_sumstats/IL12B_prot.rds&quot;)  
IL2RA_anno  &lt;- import(&quot;data/cis_sumstats/IL2RA_anno.rds&quot;)  
IL2RA_prot  &lt;- import(&quot;data/cis_sumstats/IL2RA_prot.rds&quot;)  
IL2RB_prot  &lt;- import(&quot;data/cis_sumstats/IL2RB_prot.rds&quot;)  
IL2RB_anno  &lt;- import(&quot;data/cis_sumstats/IL2RB_anno.rds&quot;)  
IL6R_anno  &lt;- import(&quot;data/cis_sumstats/IL6R_anno.rds&quot;)  
IL6R_prot  &lt;- import(&quot;data/cis_sumstats/IL6R_prot.rds&quot;)
IL6ST_anno  &lt;- import(&quot;data/cis_sumstats/IL6ST_anno.rds&quot;)  
IL6ST_prot  &lt;- import(&quot;data/cis_sumstats/IL6ST_prot.rds&quot;)
JAK2_anno   &lt;- import(&quot;data/cis_sumstats/JAK2_anno.rds&quot;)  
JAK2_prot   &lt;- import(&quot;data/cis_sumstats/JAK2_prot.rds&quot;)  
TYK2_anno   &lt;- import(&quot;data/cis_sumstats/TYK2_anno.rds&quot;)
TYK2_prot   &lt;- import(&quot;data/cis_sumstats/TYK2_prot.rds&quot;)</code></pre>
</div>
<div id="merge-protein-data-with-annotations" class="section level2">
<h2>Merge protein data with annotations</h2>
<p>Ferkingstad et al. state in their README file</p>
<blockquote>
<p>A subset of the variants in the summary statistics files should be
excluded due to quality issues. These variants are listed in a separate
Excluded variants file (assocvariants.excluded.txt.gz). The Extra
annotation file (assocvariants.annotated.txt.gz) does not include these
variants.</p>
</blockquote>
<p>Therefore in the join process, only the data available in the
annotation (IL2RG_anno) is used (hence, left_join)</p>
<pre class="r"><code>IL12B_prot_anno  &lt;- left_join(IL12B_prot, IL12B_anno, by = c(&quot;Name&quot;, &quot;Chrom&quot;, &quot;Pos&quot;))
IL2RA_prot_anno  &lt;- left_join(IL2RA_prot, IL2RA_anno, by = c(&quot;Name&quot;, &quot;Chrom&quot;, &quot;Pos&quot;))
IL2RB_prot_anno  &lt;- left_join(IL2RB_prot, IL2RB_anno, by = c(&quot;Name&quot;, &quot;Chrom&quot;, &quot;Pos&quot;))
IL2RG_prot_anno  &lt;- left_join(IL2RG_prot, IL2RG_anno, by = c(&quot;Name&quot;, &quot;Chrom&quot;, &quot;Pos&quot;))
IL6R_prot_anno   &lt;- left_join(IL6R_prot,  IL6R_anno, by = c(&quot;Name&quot;, &quot;Chrom&quot;, &quot;Pos&quot;))
IL6ST_prot_anno  &lt;- left_join(IL6ST_prot, IL6ST_anno, by = c(&quot;Name&quot;, &quot;Chrom&quot;, &quot;Pos&quot;))
JAK2_prot_anno   &lt;- left_join(JAK2_prot,  JAK2_anno, by = c(&quot;Name&quot;, &quot;Chrom&quot;, &quot;Pos&quot;))
TYK2_prot_anno   &lt;- left_join(TYK2_prot,  TYK2_anno, by = c(&quot;Name&quot;, &quot;Chrom&quot;, &quot;Pos&quot;))

rm(&quot;IL12B_anno&quot;, &quot;IL12B_prot&quot;, 
   &quot;IL2RA_anno&quot;, &quot;IL2RA_prot&quot;, &quot;IL2RB_anno&quot;, &quot;IL2RB_prot&quot;,  
   &quot;IL2RG_anno&quot;, &quot;IL2RG_prot&quot;, &quot;IL6R_anno&quot;, &quot;IL6R_prot&quot;, 
   &quot;IL6ST_anno&quot;, &quot;IL6ST_prot&quot;, &quot;JAK2_anno&quot;, &quot;JAK2_prot&quot;, 
   &quot;TYK2_anno&quot;, &quot;TYK2_prot&quot;)</code></pre>
</div>
<div id="sort-conflicts" class="section level2">
<h2>Sort conflicts</h2>
<p>Further in the README file of Ferkingstad et al. </p>
<blockquote>
<p>Also, the summary statistics files sometimes incorrectly have
effectAllele=otherAllele for multiallelic variants. In these cases the
effectAllele is correct, but the otherAllele should be ‘!’, meaning that
the effectAllele is tested against the other (two or more) alleles
(using the ‘!’ sign as shorthand for ‘not effectAllele’). This has been
corrected in the file Extra annotation file
(assocvariants.annotated.txt.gz).</p>
</blockquote>
<p>Therefore, multiallelic variants were removed.</p>
<pre class="r"><code>IL12B_prot_anno  &lt;- IL12B_prot_anno  %&gt;% filter(otherAllele.y != &quot;!&quot;)
IL2RA_prot_anno  &lt;- IL2RA_prot_anno  %&gt;% filter(otherAllele.y != &quot;!&quot;)
IL2RB_prot_anno  &lt;- IL2RB_prot_anno  %&gt;% filter(otherAllele.y != &quot;!&quot;)
IL2RG_prot_anno  &lt;- IL2RG_prot_anno  %&gt;% filter(otherAllele.y != &quot;!&quot;)
IL6R_prot_anno   &lt;- IL6ST_prot_anno  %&gt;% filter(otherAllele.y != &quot;!&quot;)
IL6ST_prot_anno  &lt;- IL6ST_prot_anno  %&gt;% filter(otherAllele.y != &quot;!&quot;)
JAK2_prot_anno   &lt;- JAK2_prot_anno   %&gt;% filter(otherAllele.y != &quot;!&quot;)
TYK2_prot_anno   &lt;- TYK2_prot_anno   %&gt;% filter(otherAllele.y != &quot;!&quot;)</code></pre>
<p>After removal of multiallelic variants there were no conflicts
between effectAllele.x and effectAllele.y or between otherAllele.x and
otherAllele.y (results suppressed, please see
“analyses/Harmonize.Rmd”)</p>
<pre class="r"><code>#Following function is simply to avoid repetition
check_conflicts_ea_oa &lt;- function(data, effectAllele.x, effectAllele.y) {
  filter(data, effectAllele.x != effectAllele.y | 
           is.na(effectAllele.x) &amp; !is.na(effectAllele.y) |
           !is.na(effectAllele.x) &amp;  is.na(effectAllele.y) | 
           otherAllele.x != otherAllele.y | 
            is.na(otherAllele.x) &amp; !is.na(otherAllele.y) |
           !is.na(otherAllele.x) &amp;  is.na(otherAllele.y))
}

#List those with conflicts in effectAllele or otherAllele

IL12B_prot_anno  %&gt;% check_conflicts_ea_oa() #0 conflicts
IL2RA_prot_anno  %&gt;% check_conflicts_ea_oa() #0 conflicts
IL2RB_prot_anno  %&gt;% check_conflicts_ea_oa() #0 conflicts
IL2RG_prot_anno  %&gt;% check_conflicts_ea_oa() #0 conflicts
IL6R_prot_anno   %&gt;% check_conflicts_ea_oa() #0 conflicts
IL6ST_prot_anno  %&gt;% check_conflicts_ea_oa() #0 conflicts
JAK2_prot_anno   %&gt;% check_conflicts_ea_oa() #0 conflicts
TYK2_prot_anno   %&gt;% check_conflicts_ea_oa() #0 conflicts</code></pre>
<p>After this, the remaining discrepancies between the variables
obtained from annotation and protein data files were because those
variants without an rsid were listed as “.” in the annotation file and
as “NA” in the main GWAS sumstat file (please see
‘/analyses/Harmonize.Rmd’)</p>
<pre class="r"><code>#List subjects which had conflicting rsid information and show
#which entries there are for these
check_conflicts_rsid &lt;- function(data, effectAllele.x, effectAllele.y) {
  filter(data, rsids.x != rsids.y | 
           is.na(rsids.x) &amp; !is.na(rsids.y) |
           !is.na(rsids.x) &amp;  is.na(rsids.y)) %&gt;%
    mutate(rsids.x = factor(rsids.x),
         rsids.y = factor(rsids.y)) %&gt;%
  select(rsids.x, rsids.y) %&gt;%
  summary()
}

#All the conflicts in rsid.x and rsid.y are because rsid.x is NA and rsid.y is &quot;.&quot;
IL12B_prot_anno  %&gt;% check_conflicts_rsid() #0 conflicts
IL2RA_prot_anno  %&gt;% check_conflicts_rsid() #0 conflicts
IL2RB_prot_anno  %&gt;% check_conflicts_rsid() #0 conflicts
IL2RG_prot_anno  %&gt;% check_conflicts_rsid() #0 conflicts
IL6R_prot_anno   %&gt;% check_conflicts_rsid() #0 conflicts
IL6ST_prot_anno  %&gt;% check_conflicts_rsid() #0 conflicts
JAK2_prot_anno   %&gt;% check_conflicts_rsid() #0 conflicts
TYK2_prot_anno   %&gt;% check_conflicts_rsid() #0 conflicts</code></pre>
<p>Thus, only those values variables with .x in their name are
preserved. Here, we also remove the remaining multiallelic SNPs (where
effect allele equals the other allele.)</p>
<pre class="r"><code>#Discard all variables that end with an &quot;.y&quot;
IL12B_prot_anno   &lt;- IL12B_prot_anno  %&gt;% select(!ends_with(&quot;.y&quot;))
IL2RA_prot_anno   &lt;- IL2RA_prot_anno  %&gt;% select(!ends_with(&quot;.y&quot;))
IL2RB_prot_anno   &lt;- IL2RB_prot_anno  %&gt;% select(!ends_with(&quot;.y&quot;))
IL2RG_prot_anno   &lt;- IL2RG_prot_anno  %&gt;% select(!ends_with(&quot;.y&quot;))
IL6R_prot_anno    &lt;- IL6ST_prot_anno  %&gt;% select(!ends_with(&quot;.y&quot;))
IL6ST_prot_anno   &lt;- IL6ST_prot_anno  %&gt;% select(!ends_with(&quot;.y&quot;))
JAK2_prot_anno    &lt;- JAK2_prot_anno   %&gt;% select(!ends_with(&quot;.y&quot;))
TYK2_prot_anno    &lt;- TYK2_prot_anno   %&gt;% select(!ends_with(&quot;.y&quot;))

#Remove &quot;.x&quot; from variable names
IL12B_prot_anno   &lt;- IL12B_prot_anno  %&gt;% rename_with(~gsub(&quot;.x&quot;, &quot;&quot;, .x, fixed = TRUE)) 
IL2RA_prot_anno   &lt;- IL2RA_prot_anno  %&gt;% rename_with(~gsub(&quot;.x&quot;, &quot;&quot;, .x, fixed = TRUE)) 
IL2RB_prot_anno   &lt;- IL2RB_prot_anno  %&gt;% rename_with(~gsub(&quot;.x&quot;, &quot;&quot;, .x, fixed = TRUE)) 
IL2RG_prot_anno   &lt;- IL2RG_prot_anno  %&gt;% rename_with(~gsub(&quot;.x&quot;, &quot;&quot;, .x, fixed = TRUE)) 
IL6ST_prot_anno   &lt;- IL6ST_prot_anno  %&gt;% rename_with(~gsub(&quot;.x&quot;, &quot;&quot;, .x, fixed = TRUE)) 
IL6R_prot_anno    &lt;- IL6R_prot_anno   %&gt;% rename_with(~gsub(&quot;.x&quot;, &quot;&quot;, .x, fixed = TRUE)) 
JAK2_prot_anno    &lt;- JAK2_prot_anno   %&gt;% rename_with(~gsub(&quot;.x&quot;, &quot;&quot;, .x, fixed = TRUE)) 
TYK2_prot_anno    &lt;- TYK2_prot_anno   %&gt;% rename_with(~gsub(&quot;.x&quot;, &quot;&quot;, .x, fixed = TRUE)) </code></pre>
<pre class="r"><code>#Sanity checks
colnames(IL2RG_prot_anno)   
colnames(IL12B_prot_anno)
colnames(IL2RA_prot_anno)
colnames(IL2RB_prot_anno)
colnames(IL2RG_prot_anno)
colnames(IL6ST_prot_anno)
colnames(IL6R_prot_anno)
colnames(JAK2_prot_anno)
colnames(TYK2_prot_anno)</code></pre>
<p>All data files seem to have entries without rsids (suppressed, please
see ‘/analyses/Harmonize.Rmd’). They are now removed</p>
<pre class="r"><code>anyNA(IL2RG_prot_anno$rsids)   
anyNA(IL12B_prot_anno$rsids)
anyNA(IL2RA_prot_anno$rsids)
anyNA(IL2RB_prot_anno$rsids)
anyNA(IL2RG_prot_anno$rsids)
anyNA(IL6ST_prot_anno$rsids)
anyNA(IL6R_prot_anno$rsids)
anyNA(JAK2_prot_anno$rsids)
anyNA(TYK2_prot_anno$rsids)

IL2RG_prot_anno   &lt;- IL2RG_prot_anno  %&gt;% filter(!is.na(rsids))
IL12B_prot_anno   &lt;- IL12B_prot_anno  %&gt;% filter(!is.na(rsids))
IL2RA_prot_anno   &lt;- IL2RA_prot_anno  %&gt;% filter(!is.na(rsids))
IL2RB_prot_anno   &lt;- IL2RB_prot_anno  %&gt;% filter(!is.na(rsids))
IL2RG_prot_anno   &lt;- IL2RG_prot_anno  %&gt;% filter(!is.na(rsids))
IL6R_prot_anno    &lt;- IL6ST_prot_anno  %&gt;% filter(!is.na(rsids))
IL6ST_prot_anno   &lt;- IL6ST_prot_anno  %&gt;% filter(!is.na(rsids))
JAK2_prot_anno    &lt;- JAK2_prot_anno   %&gt;% filter(!is.na(rsids))
TYK2_prot_anno    &lt;- TYK2_prot_anno   %&gt;% filter(!is.na(rsids))</code></pre>
<p>There were no duplicated rsids in data after exclusion of missing
rsids (suppressed, please see /analyses/Harmonize.Rmd)</p>
<pre class="r"><code>#Sanity check
anyDuplicated(IL2RG_prot_anno$rsids)   
anyDuplicated(IL12B_prot_anno$rsids)
anyDuplicated(IL2RA_prot_anno$rsids)
anyDuplicated(IL2RB_prot_anno$rsids)
anyDuplicated(IL2RG_prot_anno$rsids)
anyDuplicated(IL6ST_prot_anno$rsids)
anyDuplicated(IL6R_prot_anno$rsids)
anyDuplicated(JAK2_prot_anno$rsids)
anyDuplicated(TYK2_prot_anno$rsids)</code></pre>
</div>
<div id="write-data-in-twosamplemr-format" class="section level2">
<h2>Write data in TwoSampleMR format</h2>
<p>Data are written in TwoSampleMR format so that they can be easily
opened in the main analyses scripts.</p>
<p>Rename files</p>
<pre class="r"><code>rename_for_TwoSampleMR_e  &lt;- function(data, rsids, Beta, SE, effectAlleleFreq,
                                    effectAllele, otherAllele,
                                    Pval, Pos, N) {
  data %&gt;%
    rename(SNP = rsids, beta = Beta, se = SE, eaf = effectAlleleFreq,
         effect_allele = effectAllele, other_allele = otherAllele,
         pval = Pval, pos = Pos) %&gt;%
    mutate(samplesize = N) 
  }

#Rename
IL12B_prot_anno   &lt;- IL12B_prot_anno  %&gt;% rename_for_TwoSampleMR_e()
IL2RA_prot_anno   &lt;- IL2RA_prot_anno  %&gt;% rename_for_TwoSampleMR_e()
IL2RB_prot_anno   &lt;- IL2RB_prot_anno  %&gt;% rename_for_TwoSampleMR_e()
IL2RG_prot_anno   &lt;- IL2RG_prot_anno  %&gt;% rename_for_TwoSampleMR_e()
IL6R_prot_anno    &lt;- IL6R_prot_anno   %&gt;% rename_for_TwoSampleMR_e()
IL6ST_prot_anno   &lt;- IL6ST_prot_anno  %&gt;% rename_for_TwoSampleMR_e()
JAK2_prot_anno    &lt;- JAK2_prot_anno   %&gt;% rename_for_TwoSampleMR_e()
TYK2_prot_anno    &lt;- TYK2_prot_anno   %&gt;% rename_for_TwoSampleMR_e()

#Add phenotype information
IL12B_prot_anno   &lt;- IL12B_prot_anno  %&gt;%  
  mutate(Phenotype = &quot;Serum IL12B level&quot;)
IL2RA_prot_anno   &lt;- IL2RA_prot_anno  %&gt;%  
  mutate(Phenotype = &quot;Serum IL2RA level&quot;)
IL2RB_prot_anno   &lt;- IL2RB_prot_anno  %&gt;%  
  mutate(Phenotype = &quot;Serum IL2RB level&quot;)
IL2RG_prot_anno   &lt;- IL2RG_prot_anno  %&gt;%  
  mutate(Phenotype = &quot;Serum IL2RG level&quot;)
IL6R_prot_anno    &lt;- IL6R_prot_anno  %&gt;%
  mutate(Phenotype = &quot;Serum IL6R level&quot;)
IL6ST_prot_anno   &lt;- IL6ST_prot_anno  %&gt;%
  mutate(Phenotype = &quot;Serum IL6ST level&quot;)
JAK2_prot_anno    &lt;- JAK2_prot_anno   %&gt;%  
  mutate(Phenotype = &quot;Serum JAK2 level&quot;)
TYK2_prot_anno    &lt;- TYK2_prot_anno   %&gt;%  
  mutate(Phenotype = &quot;Serum TYK2 level&quot;)</code></pre>
<p>Export filest to a CSV file with default parameters of
TwoSampleMR::read_exposure_data()</p>
<pre class="r"><code>#Export
IL12B_prot_anno  %&gt;% 
  export(&quot;data/export/IL12B_prot_anno_TwoSampleMR.csv&quot;)
IL2RA_prot_anno  %&gt;% 
  export(&quot;data/export/IL2RA_prot_anno_TwoSampleMR.csv&quot;)
IL2RB_prot_anno  %&gt;% 
  export(&quot;data/export/IL2RB_prot_anno_TwoSampleMR.csv&quot;)
IL2RG_prot_anno  %&gt;% 
  export(&quot;data/export/IL2RG_prot_anno_TwoSampleMR.csv&quot;)
IL6R_prot_anno  %&gt;%
  export(&quot;data/export/IL6R_prot_anno_TwoSampleMR.csv&quot;)
IL6ST_prot_anno  %&gt;%
  export(&quot;data/export/IL6ST_prot_anno_TwoSampleMR.csv&quot;)
JAK2_prot_anno   %&gt;% 
  export(&quot;data/export/JAK2_prot_anno_TwoSampleMR.csv&quot;)
TYK2_prot_anno   %&gt;% 
  export(&quot;data/export/TYK2_prot_anno_TwoSampleMR.csv&quot;)</code></pre>
<pre class="r"><code>#Sanity check: no warnings
# read_exposure_data(&quot;data/export/IL12B_prot_anno_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_exposure_data(&quot;data/export/IL2RA_prot_anno_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_exposure_data(&quot;data/export/IL2RB_prot_anno_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_exposure_data(&quot;data/export/IL2RG_prot_anno_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_exposure_data(&quot;data/export/IL6ST_prot_anno_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_exposure_data(&quot;data/export/IL6R_prot_anno_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_exposure_data(&quot;data/export/JAK2_prot_anno_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_exposure_data(&quot;data/export/TYK2_prot_anno_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)</code></pre>
</div>
</div>
<div id="il6-crp-data" class="section level1">
<h1>IL6 CRP data</h1>
<div id="write-il6r_crp-in-twosamplemr-format" class="section level2">
<h2>Write IL6R_crp in TwoSampleMR format</h2>
<pre class="r"><code>IL6R_crp   &lt;- import(&quot;data/cis_sumstats/IL6R_crp.rds&quot;)  

# colnames(IL6R_crp)
IL6R_crp &lt;- IL6R_crp %&gt;%
  select(-other_allele, -effect_allele, -beta) %&gt;%
  rename(SNP = hm_rsid, beta = hm_beta, se = standard_error, 
           eaf = hm_effect_allele_frequency,
           effect_allele = hm_effect_allele, 
           other_allele = hm_other_allele,
           pval = p_value, pos = hm_pos) %&gt;%
  mutate(pval = as.numeric(pval),
         eaf = as.numeric(eaf),
         samplesize = 575531,
         Phenotype = &quot;Serum C-reactive protein levels&quot;) 

IL6R_crp  %&gt;% 
  export(&quot;data/export/IL6R_crp_TwoSampleMR.csv&quot;)</code></pre>
<p>#eQTL data</p>
</div>
<div id="import-data" class="section level2">
<h2>Import data</h2>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:lubridate&#39;:

    hour, isoweek, mday, minute, month, quarter, second, wday, week,
    yday, year</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(dtplyr)

#Read eqtl sumstat
IFNAR2_eqtl &lt;- import(&quot;data/cis_eqtl_sumstats/eqtl_IFNAR2-Gather.rds&quot;) %&gt;% as_tibble()
IL2RA_eqtl  &lt;- import(&quot;data/cis_eqtl_sumstats/eqtl_IL2RA-Gather.rds&quot;) %&gt;% as_tibble()
IL2RG_eqtl  &lt;- import(&quot;data/cis_eqtl_sumstats/eqtl_IL2RG-Gather.rds&quot;) %&gt;% as_tibble()
IL2RB_eqtl  &lt;- import(&quot;data/cis_eqtl_sumstats/eqtl_IL2RB-Gather.rds&quot;) %&gt;% as_tibble()
IL6R_eqtl   &lt;- import(&quot;data/cis_eqtl_sumstats/eqtl_IL6R-Gather.rds&quot;) %&gt;% as_tibble()
JAK1_eqtl   &lt;- import(&quot;data/cis_eqtl_sumstats/eqtl_JAK1-Gather.rds&quot;) %&gt;% as_tibble()
JAK2_eqtl   &lt;- import(&quot;data/cis_eqtl_sumstats/eqtl_JAK2-Gather.rds&quot;) %&gt;% as_tibble()
JAK3_eqtl   &lt;- import(&quot;data/cis_eqtl_sumstats/eqtl_JAK3-Gather.rds&quot;) %&gt;% as_tibble()
TYK2_eqtl   &lt;- import(&quot;data/cis_eqtl_sumstats/eqtl_TYK2-Gather.rds&quot;) %&gt;% as_tibble()


names_eqtl &lt;- c(&quot;IFNAR2_eqtl&quot;, 
&quot;IL2RA_eqtl&quot;, 
&quot;IL2RG_eqtl&quot;, 
&quot;IL2RB_eqtl&quot;, 
&quot;IL6R_eqtl&quot;, 
&quot;JAK1_eqtl&quot;, 
&quot;JAK2_eqtl&quot;, 
&quot;JAK3_eqtl&quot;, 
&quot;TYK2_eqtl&quot;)</code></pre>
<p>According to the readme files of eQTLGEN phase I
(data/REAMDE_cis_eqtl.txt and
data/README_cis_eqtl_allele_frequency.txt),</p>
<table style="width:100%;">
<caption>sources: README_cis_eqtl_AF.txt = <a
href="https://molgenis26.gcc.rug.nl/downloads/eqtlgen/cis-eqtl/README_allele_frequency"
class="uri">https://molgenis26.gcc.rug.nl/downloads/eqtlgen/cis-eqtl/README_allele_frequency</a>,
README_cis_eqtl.txt = <a
href="https://molgenis26.gcc.rug.nl/downloads/eqtlgen/cis-eqtl/README_cis"
class="uri">https://molgenis26.gcc.rug.nl/downloads/eqtlgen/cis-eqtl/README_cis</a></caption>
<colgroup>
<col width="14%" />
<col width="63%" />
<col width="21%" />
</colgroup>
<tbody>
<tr class="odd">
<td align="left">Pvalue</td>
<td align="left">P-value</td>
<td align="left">README_cis_eqtl.txt</td>
</tr>
<tr class="even">
<td align="left">SNP</td>
<td align="left">SNP rs ID</td>
<td align="left">README_cis_eqtl.txt</td>
</tr>
<tr class="odd">
<td align="left">SNPChr</td>
<td align="left">SNP chromosome</td>
<td align="left">README_cis_eqtl.txt</td>
</tr>
<tr class="even">
<td align="left">SNPPos</td>
<td align="left">SNP position</td>
<td align="left">README_cis_eqtl.txt</td>
</tr>
<tr class="odd">
<td align="left">AssessedAllele</td>
<td align="left">Assessed allele, the Z-score refers to this allele</td>
<td align="left">README_cis_eqtl.txt</td>
</tr>
<tr class="even">
<td align="left">OtherAllele</td>
<td align="left">Not assessed allele</td>
<td align="left">README_cis_eqtl.txt</td>
</tr>
<tr class="odd">
<td align="left">Zscore</td>
<td align="left">Z-score</td>
<td align="left">README_cis_eqtl.txt</td>
</tr>
<tr class="even">
<td align="left">Gene</td>
<td align="left">ENSG name (Ensembl v71) of the eQTL gene</td>
<td align="left">README_cis_eqtl.txt</td>
</tr>
<tr class="odd">
<td align="left">GeneSymbol</td>
<td align="left">HGNC name of the gene</td>
<td align="left">README_cis_eqtl.txt</td>
</tr>
<tr class="even">
<td align="left">GeneChr</td>
<td align="left">Gene chromosome</td>
<td align="left">README_cis_eqtl.txt</td>
</tr>
<tr class="odd">
<td align="left">GenePos</td>
<td align="left">Centre of gene position</td>
<td align="left">README_cis_eqtl.txt</td>
</tr>
<tr class="even">
<td align="left">NrCohorts</td>
<td align="left">Total number of cohorts where this SNP-gene combination
was tested</td>
<td align="left">README_cis_eqtl.txt</td>
</tr>
<tr class="odd">
<td align="left">NrSamples</td>
<td align="left">Total number of samples where this SNP-gene combination
was tested</td>
<td align="left">README_cis_eqtl.txt</td>
</tr>
<tr class="even">
<td align="left">FDR</td>
<td align="left">False discovery rate estimated based on
permutations</td>
<td align="left">README_cis_eqtl.txt</td>
</tr>
<tr class="odd">
<td align="left">BonferroniP</td>
<td align="left">P-value after Bonferroni correction</td>
<td align="left">README_cis_eqtl.txt</td>
</tr>
<tr class="even">
<td align="left">SNP</td>
<td align="left">SNP rs ID</td>
<td align="left">README_cis_eqtl_AF.txt</td>
</tr>
<tr class="odd">
<td align="left">hg19_chr</td>
<td align="left">chr number</td>
<td align="left">README_cis_eqtl_AF.txt</td>
</tr>
<tr class="even">
<td align="left">hg19_pos</td>
<td align="left">SNP position (hg19)</td>
<td align="left">README_cis_eqtl_AF.txt</td>
</tr>
<tr class="odd">
<td align="left">AlleleA</td>
<td align="left">Other allele</td>
<td align="left">README_cis_eqtl_AF.txt</td>
</tr>
<tr class="even">
<td align="left">AlleleB</td>
<td align="left">Assessed allele</td>
<td align="left">README_cis_eqtl_AF.txt</td>
</tr>
<tr class="odd">
<td align="left">allA_total</td>
<td align="left">Total allele count of genotype AA</td>
<td align="left">README_cis_eqtl_AF.txt</td>
</tr>
<tr class="even">
<td align="left">allAB_total</td>
<td align="left">Total allele count of genotype AB</td>
<td align="left">README_cis_eqtl_AF.txt</td>
</tr>
<tr class="odd">
<td align="left">allB_total</td>
<td align="left">Total allele count of genotype BB</td>
<td align="left">README_cis_eqtl_AF.txt</td>
</tr>
<tr class="even">
<td align="left">AlleleB_all</td>
<td align="left">Allele frequency of assessed allele</td>
<td align="left">README_cis_eqtl_AF.txt</td>
</tr>
</tbody>
</table>
<p>There is some disagreement between the readme and the original file.
E.g. there are 1,053 subjects near IL2RB which for which
‘AssessedAllele’ is not equal to ‘AlleleB’.</p>
<pre class="r"><code>colnames(IL2RB_eqtl)</code></pre>
<pre><code> [1] &quot;Pvalue&quot;         &quot;SNP&quot;            &quot;SNPChr&quot;         &quot;SNPPos&quot;        
 [5] &quot;AssessedAllele&quot; &quot;OtherAllele&quot;    &quot;Zscore&quot;         &quot;Gene&quot;          
 [9] &quot;GeneSymbol&quot;     &quot;GeneChr&quot;        &quot;GenePos&quot;        &quot;NrCohorts&quot;     
[13] &quot;NrSamples&quot;      &quot;FDR&quot;            &quot;BonferroniP&quot;    &quot;hg19_chr&quot;      
[17] &quot;hg19_pos&quot;       &quot;AlleleA&quot;        &quot;AlleleB&quot;        &quot;allA_total&quot;    
[21] &quot;allAB_total&quot;    &quot;allB_total&quot;     &quot;AlleleB_all&quot;   </code></pre>
<pre class="r"><code>get_n_conflicts  &lt;- function(data, AlleleB=AlleleB, AssessedAllele=AssessedAllele) {
  data %&gt;%
  filter(AlleleB != AssessedAllele) %&gt;% 
  pull(SNP) %&gt;% length()
}

table_conflicts_eqtl &lt;- c(&quot;IFNAR2&quot;, get_n_conflicts(IFNAR2_eqtl),
  &quot;IL2RA&quot;, get_n_conflicts(IL2RA_eqtl),
  &quot;IL2RG&quot;, get_n_conflicts(IL2RG_eqtl),
  &quot;IL2RB&quot;, get_n_conflicts(IL2RB_eqtl),
  &quot;IL6R&quot;, get_n_conflicts(IL6R_eqtl),
  &quot;JAK1&quot;, get_n_conflicts(JAK1_eqtl),
  &quot;JAK2&quot;, get_n_conflicts(JAK2_eqtl),
  &quot;JAK3&quot;, get_n_conflicts(JAK3_eqtl),
  &quot;TYK2&quot;, get_n_conflicts(TYK2_eqtl)) %&gt;%
  matrix(ncol = 2, byrow = TRUE)

colnames(table_conflicts_eqtl) &lt;- c(&quot;EQTL&quot;, &quot;n conflicts&quot;)

knitr::kable(table_conflicts_eqtl)</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">EQTL</th>
<th align="left">n conflicts</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">IFNAR2</td>
<td align="left">1981</td>
</tr>
<tr class="even">
<td align="left">IL2RA</td>
<td align="left">1695</td>
</tr>
<tr class="odd">
<td align="left">IL2RG</td>
<td align="left">0</td>
</tr>
<tr class="even">
<td align="left">IL2RB</td>
<td align="left">1053</td>
</tr>
<tr class="odd">
<td align="left">IL6R</td>
<td align="left">672</td>
</tr>
<tr class="even">
<td align="left">JAK1</td>
<td align="left">932</td>
</tr>
<tr class="odd">
<td align="left">JAK2</td>
<td align="left">1393</td>
</tr>
<tr class="even">
<td align="left">JAK3</td>
<td align="left">1025</td>
</tr>
<tr class="odd">
<td align="left">TYK2</td>
<td align="left">1353</td>
</tr>
</tbody>
</table>
<pre class="r"><code>summarize_agreement &lt;- function(data, AlleleB=AlleleB, 
                           AssessedAllele=AssessedAllele, 
                           AlleleB_all=AlleleB_all) {
  data %&gt;%
  mutate(AlleleB_vs_AssessedAllele = 
           ifelse(AlleleB == AssessedAllele, &quot;AlleleB == AssessedAllele&quot;, 
                  &quot;AlleleB != AssessedAllele&quot;)) %&gt;%
  group_by(AlleleB_vs_AssessedAllele) %&gt;%
  summarise(p_AF_m40  = round(sum(AlleleB_all &lt; 0.40) / length(AlleleB_all), 4),
            p_AF_4050 = round(sum(AlleleB_all &gt;= 0.40 &amp; AlleleB_all &lt; 0.50)  /
                                length(AlleleB_all), 4),
            p_AF_5060 = round(sum(AlleleB_all &gt; 0.50 &amp; AlleleB_all &lt;= 0.60)  /
                                length(AlleleB_all), 4),
            p_AF_p60  = round(sum(AlleleB_all &gt; 0.60) / length(AlleleB_all), 4),
            p_AF_50  = round(sum(AlleleB_all == 0.50) / length(AlleleB_all), 4),
            n_AF_NA = sum(is.na(AlleleB_all)))
}</code></pre>
<p>In a remarkably stable pattern between files, AlleleB == Assessed
allele when whenAF &lt; 0.50 and AlleleB != Assessed allele when AF &gt;
0.50.</p>
<pre class="r"><code>summarize_agreement(IFNAR2_eqtl) %&gt;% 
  knitr::kable(caption = &quot;IFNAR2_eqtl&quot;)</code></pre>
<table>
<caption>IFNAR2_eqtl</caption>
<colgroup>
<col width="32%" />
<col width="11%" />
<col width="12%" />
<col width="12%" />
<col width="11%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">AlleleB_vs_AssessedAllele</th>
<th align="right">p_AF_m40</th>
<th align="right">p_AF_4050</th>
<th align="right">p_AF_5060</th>
<th align="right">p_AF_p60</th>
<th align="right">p_AF_50</th>
<th align="right">n_AF_NA</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AlleleB != AssessedAllele</td>
<td align="right">0.0000</td>
<td align="right">0.0288</td>
<td align="right">0.3458</td>
<td align="right">0.6254</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">AlleleB == AssessedAllele</td>
<td align="right">0.9392</td>
<td align="right">0.0590</td>
<td align="right">0.0018</td>
<td align="right">0.0000</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>summarize_agreement(IL2RA_eqtl) %&gt;% 
  knitr::kable(caption = &quot;IL2RA_eqtl&quot;)</code></pre>
<table>
<caption>IL2RA_eqtl</caption>
<colgroup>
<col width="32%" />
<col width="11%" />
<col width="12%" />
<col width="12%" />
<col width="11%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">AlleleB_vs_AssessedAllele</th>
<th align="right">p_AF_m40</th>
<th align="right">p_AF_4050</th>
<th align="right">p_AF_5060</th>
<th align="right">p_AF_p60</th>
<th align="right">p_AF_50</th>
<th align="right">n_AF_NA</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AlleleB != AssessedAllele</td>
<td align="right">0.0006</td>
<td align="right">0.0265</td>
<td align="right">0.1971</td>
<td align="right">0.7758</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">AlleleB == AssessedAllele</td>
<td align="right">0.9302</td>
<td align="right">0.0637</td>
<td align="right">0.0058</td>
<td align="right">0.0004</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>summarize_agreement(IL2RB_eqtl) %&gt;% 
  knitr::kable(caption = &quot;IL2RB_eqtl&quot;)</code></pre>
<table>
<caption>IL2RB_eqtl</caption>
<colgroup>
<col width="32%" />
<col width="11%" />
<col width="12%" />
<col width="12%" />
<col width="11%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">AlleleB_vs_AssessedAllele</th>
<th align="right">p_AF_m40</th>
<th align="right">p_AF_4050</th>
<th align="right">p_AF_5060</th>
<th align="right">p_AF_p60</th>
<th align="right">p_AF_50</th>
<th align="right">n_AF_NA</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AlleleB != AssessedAllele</td>
<td align="right">0.0038</td>
<td align="right">0.0465</td>
<td align="right">0.3761</td>
<td align="right">0.5736</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">AlleleB == AssessedAllele</td>
<td align="right">0.9205</td>
<td align="right">0.0778</td>
<td align="right">0.0017</td>
<td align="right">0.0000</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>summarize_agreement(IL6R_eqtl) %&gt;% 
  knitr::kable(caption = &quot;IL6R_eqtl&quot;)</code></pre>
<table>
<caption>IL6R_eqtl</caption>
<colgroup>
<col width="32%" />
<col width="11%" />
<col width="12%" />
<col width="12%" />
<col width="11%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">AlleleB_vs_AssessedAllele</th>
<th align="right">p_AF_m40</th>
<th align="right">p_AF_4050</th>
<th align="right">p_AF_5060</th>
<th align="right">p_AF_p60</th>
<th align="right">p_AF_50</th>
<th align="right">n_AF_NA</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AlleleB != AssessedAllele</td>
<td align="right">0.0000</td>
<td align="right">0.0967</td>
<td align="right">0.5015</td>
<td align="right">0.4018</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">AlleleB == AssessedAllele</td>
<td align="right">0.9295</td>
<td align="right">0.0616</td>
<td align="right">0.0090</td>
<td align="right">0.0000</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>summarize_agreement(JAK1_eqtl) %&gt;% 
  knitr::kable(caption = &quot;JAK1_eqtl&quot;)</code></pre>
<table>
<caption>JAK1_eqtl</caption>
<colgroup>
<col width="32%" />
<col width="11%" />
<col width="12%" />
<col width="12%" />
<col width="11%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">AlleleB_vs_AssessedAllele</th>
<th align="right">p_AF_m40</th>
<th align="right">p_AF_4050</th>
<th align="right">p_AF_5060</th>
<th align="right">p_AF_p60</th>
<th align="right">p_AF_50</th>
<th align="right">n_AF_NA</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AlleleB != AssessedAllele</td>
<td align="right">0.0054</td>
<td align="right">0.0064</td>
<td align="right">0.2768</td>
<td align="right">0.7114</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">AlleleB == AssessedAllele</td>
<td align="right">0.9510</td>
<td align="right">0.0435</td>
<td align="right">0.0054</td>
<td align="right">0.0000</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>summarize_agreement(JAK2_eqtl) %&gt;% 
  knitr::kable(caption = &quot;JAK2_eqtl&quot;)</code></pre>
<table>
<caption>JAK2_eqtl</caption>
<colgroup>
<col width="32%" />
<col width="11%" />
<col width="12%" />
<col width="12%" />
<col width="11%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">AlleleB_vs_AssessedAllele</th>
<th align="right">p_AF_m40</th>
<th align="right">p_AF_4050</th>
<th align="right">p_AF_5060</th>
<th align="right">p_AF_p60</th>
<th align="right">p_AF_50</th>
<th align="right">n_AF_NA</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AlleleB != AssessedAllele</td>
<td align="right">0.0029</td>
<td align="right">0.0223</td>
<td align="right">0.2032</td>
<td align="right">0.7717</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">AlleleB == AssessedAllele</td>
<td align="right">0.9383</td>
<td align="right">0.0495</td>
<td align="right">0.0122</td>
<td align="right">0.0000</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>summarize_agreement(JAK3_eqtl) %&gt;% 
  knitr::kable(caption = &quot;JAK3_eqtl&quot;)</code></pre>
<table>
<caption>JAK3_eqtl</caption>
<colgroup>
<col width="32%" />
<col width="11%" />
<col width="12%" />
<col width="12%" />
<col width="11%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">AlleleB_vs_AssessedAllele</th>
<th align="right">p_AF_m40</th>
<th align="right">p_AF_4050</th>
<th align="right">p_AF_5060</th>
<th align="right">p_AF_p60</th>
<th align="right">p_AF_50</th>
<th align="right">n_AF_NA</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AlleleB != AssessedAllele</td>
<td align="right">0.0000</td>
<td align="right">0.0478</td>
<td align="right">0.3044</td>
<td align="right">0.6468</td>
<td align="right">0.001</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">AlleleB == AssessedAllele</td>
<td align="right">0.9429</td>
<td align="right">0.0531</td>
<td align="right">0.0030</td>
<td align="right">0.0009</td>
<td align="right">0.000</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<pre class="r"><code>summarize_agreement(TYK2_eqtl) %&gt;% 
  knitr::kable(caption = &quot;TYK2_eqtl&quot;)</code></pre>
<table>
<caption>TYK2_eqtl</caption>
<colgroup>
<col width="32%" />
<col width="11%" />
<col width="12%" />
<col width="12%" />
<col width="11%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">AlleleB_vs_AssessedAllele</th>
<th align="right">p_AF_m40</th>
<th align="right">p_AF_4050</th>
<th align="right">p_AF_5060</th>
<th align="right">p_AF_p60</th>
<th align="right">p_AF_50</th>
<th align="right">n_AF_NA</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">AlleleB != AssessedAllele</td>
<td align="right">0.0000</td>
<td align="right">0.0244</td>
<td align="right">0.1471</td>
<td align="right">0.8285</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">AlleleB == AssessedAllele</td>
<td align="right">0.9158</td>
<td align="right">0.0803</td>
<td align="right">0.0039</td>
<td align="right">0.0000</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>From this, it is taken that AF is given per effect alleles mentioned
in README_cis_eqtl_AF.txt. Therefore, EAFs are harmonized to align with
the variable ‘AssessedAllele’</p>
<pre class="r"><code>IFNAR2_eqtl &lt;- IFNAR2_eqtl %&gt;% rename(AF_B = AlleleB_all)
IL2RA_eqtl  &lt;- IL2RA_eqtl  %&gt;% rename(AF_B = AlleleB_all)
IL2RG_eqtl  &lt;- IL2RG_eqtl  %&gt;% rename(AF_B = AlleleB_all)
IL2RB_eqtl  &lt;- IL2RB_eqtl  %&gt;% rename(AF_B = AlleleB_all)
IL6R_eqtl   &lt;- IL6R_eqtl   %&gt;% rename(AF_B = AlleleB_all)
JAK1_eqtl   &lt;- JAK1_eqtl   %&gt;% rename(AF_B = AlleleB_all)
JAK2_eqtl   &lt;- JAK2_eqtl   %&gt;% rename(AF_B = AlleleB_all)
JAK3_eqtl   &lt;- JAK3_eqtl   %&gt;% rename(AF_B = AlleleB_all)
TYK2_eqtl   &lt;- TYK2_eqtl   %&gt;% rename(AF_B = AlleleB_all)

IFNAR2_eqtl &lt;- IFNAR2_eqtl %&gt;%
  mutate(EAF = ifelse(AlleleB == AssessedAllele, AF_B, 1 - AF_B))
IL2RA_eqtl  &lt;- IL2RA_eqtl %&gt;%
  mutate(EAF = ifelse(AlleleB == AssessedAllele, AF_B, 1 - AF_B))
IL2RG_eqtl  &lt;- IL2RG_eqtl %&gt;%
  mutate(EAF = ifelse(AlleleB == AssessedAllele, AF_B, 1 - AF_B))
IL2RB_eqtl  &lt;- IL2RB_eqtl %&gt;%
  mutate(EAF = ifelse(AlleleB == AssessedAllele, AF_B, 1 - AF_B))
IL6R_eqtl   &lt;- IL6R_eqtl %&gt;%
  mutate(EAF = ifelse(AlleleB == AssessedAllele, AF_B, 1 - AF_B))
JAK1_eqtl   &lt;- JAK1_eqtl %&gt;%
  mutate(EAF = ifelse(AlleleB == AssessedAllele, AF_B, 1 - AF_B))
JAK2_eqtl   &lt;- JAK2_eqtl %&gt;%
  mutate(EAF = ifelse(AlleleB == AssessedAllele, AF_B, 1 - AF_B))
JAK3_eqtl   &lt;- JAK3_eqtl %&gt;%
  mutate(EAF = ifelse(AlleleB == AssessedAllele, AF_B, 1 - AF_B))
TYK2_eqtl   &lt;- TYK2_eqtl %&gt;%
  mutate(EAF = ifelse(AlleleB == AssessedAllele, AF_B, 1 - AF_B))


IFNAR2_eqtl &lt;- IFNAR2_eqtl %&gt;%
  mutate(MAF = ifelse(EAF &lt; 0.5,  EAF, 1 - EAF))
IL2RA_eqtl  &lt;- IL2RA_eqtl %&gt;%
  mutate(MAF = ifelse(EAF &lt; 0.5,  EAF, 1 - EAF))
IL2RG_eqtl  &lt;- IL2RG_eqtl %&gt;%
  mutate(MAF = ifelse(EAF &lt; 0.5,  EAF, 1 - EAF))
IL2RB_eqtl  &lt;- IL2RB_eqtl %&gt;%
  mutate(MAF = ifelse(EAF &lt; 0.5,  EAF, 1 - EAF))
IL6R_eqtl   &lt;- IL6R_eqtl %&gt;%
  mutate(MAF = ifelse(EAF &lt; 0.5,  EAF, 1 - EAF))
JAK1_eqtl   &lt;- JAK1_eqtl %&gt;%
  mutate(MAF = ifelse(EAF &lt; 0.5,  EAF, 1 - EAF))
JAK2_eqtl   &lt;- JAK2_eqtl %&gt;%
  mutate(MAF = ifelse(EAF &lt; 0.5,  EAF, 1 - EAF))
JAK3_eqtl   &lt;- JAK3_eqtl %&gt;%
  mutate(MAF = ifelse(EAF &lt; 0.5,  EAF, 1 - EAF))
TYK2_eqtl   &lt;- TYK2_eqtl %&gt;%
  mutate(MAF = ifelse(EAF &lt; 0.5,  EAF, 1 - EAF))</code></pre>
<pre class="r"><code>rename_eqtl_data &lt;- function(data, Phenotype, Zscore, EAF, AssessedAllele, OtherAllele,
                              Pvalue, SNPPos, NrSamples) {
   data %&gt;%
  rename(beta = Zscore, 
         eaf = EAF,
         effect_allele = AssessedAllele, 
         other_allele = OtherAllele,
         pval = Pvalue, 
         pos = SNPPos,
         samplesize = NrSamples) %&gt;%
  mutate(pval = as.numeric(pval),
         eaf = as.numeric(eaf),
         Phenotype = Phenotype,
         se = NA) 
 }

IFNAR2_eqtl %&gt;%  rename_eqtl_data(Phenotype = &quot;Whole blood IFNAR2 mRNA&quot;) %&gt;% export(&quot;data/export/IFNAR2_eqtl_TwoSampleMR.csv&quot;)
IL2RA_eqtl  %&gt;%  rename_eqtl_data(Phenotype = &quot;Whole blood IL2RA mRNA&quot;) %&gt;% export(&quot;data/export/IL2RA_eqtl_TwoSampleMR.csv&quot;)
IL2RG_eqtl  %&gt;%  rename_eqtl_data(Phenotype = &quot;Whole blood IL2RG mRNA&quot;) %&gt;% export(&quot;data/export/IL2RG_eqtl_TwoSampleMR.csv&quot;)
IL2RB_eqtl  %&gt;%  rename_eqtl_data(Phenotype = &quot;Whole blood IL2RB mRNA&quot;) %&gt;% export(&quot;data/export/IL2RB_eqtl_TwoSampleMR.csv&quot;)
IL6R_eqtl   %&gt;%  rename_eqtl_data(Phenotype = &quot;Whole blood IL6L mRNA&quot;) %&gt;% export(&quot;data/export/IL6R_eqtl_TwoSampleMR.csv&quot;)
JAK1_eqtl   %&gt;%  rename_eqtl_data(Phenotype = &quot;Whole blood JAK1 mRNA&quot;) %&gt;% export(&quot;data/export/JAK1_eqtl_TwoSampleMR.csv&quot;)
JAK2_eqtl   %&gt;%  rename_eqtl_data(Phenotype = &quot;Whole blood JAK2 mRNA&quot;) %&gt;% export(&quot;data/export/JAK2_eqtl_TwoSampleMR.csv&quot;)
JAK3_eqtl   %&gt;%  rename_eqtl_data(Phenotype = &quot;Whole blood JAK3 mRNA&quot;) %&gt;% export(&quot;data/export/JAK3_eqtl_TwoSampleMR.csv&quot;)
TYK2_eqtl   %&gt;%  rename_eqtl_data(Phenotype = &quot;Whole blood TYK2 mRNA&quot;) %&gt;% export(&quot;data/export/TYK2_eqtl_TwoSampleMR.csv&quot;)</code></pre>
</div>
</div>
<div id="outcome-data" class="section level1">
<h1>Outcome data</h1>
<div id="import-outcome-data" class="section level2">
<h2>Import outcome data</h2>
<pre class="r"><code>IL12B_T1D   &lt;- import(&quot;data/cis_sumstats/IL12B_T1D.rds&quot;) %&gt;% as_tibble()  
TYK2_T1D    &lt;- import(&quot;data/cis_sumstats/TYK2_T1D.rds&quot;)  %&gt;% as_tibble()
JAK1_T1D    &lt;- import(&quot;data/cis_sumstats/JAK1_T1D.rds&quot;)  %&gt;% as_tibble()
JAK2_T1D    &lt;- import(&quot;data/cis_sumstats/JAK2_T1D.rds&quot;)  %&gt;% as_tibble()
JAK3_T1D    &lt;- import(&quot;data/cis_sumstats/JAK3_T1D.rds&quot;)  %&gt;% as_tibble()
IL6ST_T1D   &lt;- import(&quot;data/cis_sumstats/IL6ST_T1D.rds&quot;) %&gt;% as_tibble()
IL6R_T1D    &lt;- import(&quot;data/cis_sumstats/IL6R_T1D.rds&quot;)  %&gt;% as_tibble()
IL2RB_T1D   &lt;- import(&quot;data/cis_sumstats/IL2RB_T1D.rds&quot;) %&gt;% as_tibble()
IL2RG_T1D   &lt;- import(&quot;data/cis_sumstats/IL2RG_T1D.rds&quot;) %&gt;% as_tibble()
IL2RA_T1D   &lt;- import(&quot;data/cis_sumstats/IL2RA_T1D.rds&quot;) %&gt;% as_tibble()
IFNAR2_T1D  &lt;- import(&quot;data/cis_sumstats/IFNAR2_T1D.rds&quot;) %&gt;% as_tibble()</code></pre>
<p>There were no missing rsid values (suppressed)</p>
</div>
<div id="remove-duplicated-snps" class="section level2">
<h2>Remove duplicated SNPs</h2>
<p>All data frames had duplicated rsids (suppressed). They are
removed.</p>
<pre class="r"><code>IL12B_T1D  %&gt;% pull(hm_rsid) %&gt;% anyDuplicated()
TYK2_T1D   %&gt;% pull(hm_rsid) %&gt;% anyDuplicated() 
JAK1_T1D   %&gt;% pull(hm_rsid) %&gt;% anyDuplicated()
JAK2_T1D   %&gt;% pull(hm_rsid) %&gt;% anyDuplicated()
JAK3_T1D   %&gt;% pull(hm_rsid) %&gt;% anyDuplicated()
IL6ST_T1D  %&gt;% pull(hm_rsid) %&gt;% anyDuplicated()
IL6R_T1D   %&gt;% pull(hm_rsid) %&gt;% anyDuplicated()
IL2RB_T1D  %&gt;% pull(hm_rsid) %&gt;% anyDuplicated()
IL2RG_T1D  %&gt;% pull(hm_rsid) %&gt;% anyDuplicated()
IL2RA_T1D  %&gt;% pull(hm_rsid) %&gt;% anyDuplicated()
IFNAR2_T1D %&gt;% pull(hm_rsid) %&gt;% anyDuplicated()</code></pre>
<pre class="r"><code>duplicated_snps_IL12B  &lt;- IL12B_T1D  %&gt;% 
  filter(duplicated(hm_rsid)) %&gt;% pull(hm_rsid)
duplicated_snps_TYK2   &lt;- TYK2_T1D   %&gt;% 
  filter(duplicated(hm_rsid)) %&gt;% pull(hm_rsid) 
duplicated_snps_JAK1   &lt;- JAK1_T1D   %&gt;% 
  filter(duplicated(hm_rsid)) %&gt;% pull(hm_rsid)
duplicated_snps_JAK2   &lt;- JAK2_T1D   %&gt;% 
  filter(duplicated(hm_rsid)) %&gt;% pull(hm_rsid)
duplicated_snps_JAK3   &lt;- JAK3_T1D   %&gt;% 
  filter(duplicated(hm_rsid)) %&gt;% pull(hm_rsid)
duplicated_snps_IL6ST  &lt;- IL6ST_T1D  %&gt;%
  filter(duplicated(hm_rsid)) %&gt;% pull(hm_rsid)
duplicated_snps_IL6R   &lt;- IL6R_T1D   %&gt;% 
  filter(duplicated(hm_rsid)) %&gt;% pull(hm_rsid)
duplicated_snps_IL2RB  &lt;- IL2RB_T1D  %&gt;% 
  filter(duplicated(hm_rsid)) %&gt;% pull(hm_rsid)
duplicated_snps_IL2RG  &lt;- IL2RG_T1D  %&gt;% 
  filter(duplicated(hm_rsid)) %&gt;% pull(hm_rsid)
duplicated_snps_IL2RA  &lt;- IL2RA_T1D  %&gt;% 
  filter(duplicated(hm_rsid)) %&gt;% pull(hm_rsid)
duplicated_snps_IFNAR2 &lt;- IFNAR2_T1D  %&gt;% 
  filter(duplicated(hm_rsid)) %&gt;% pull(hm_rsid)

IL12B_T1D  &lt;- IL12B_T1D   %&gt;% filter(!hm_rsid %in% duplicated_snps_IL12B)
TYK2_T1D   &lt;- TYK2_T1D    %&gt;% filter(!hm_rsid %in% duplicated_snps_TYK2) 
JAK1_T1D   &lt;- JAK1_T1D    %&gt;% filter(!hm_rsid %in% duplicated_snps_JAK1)
JAK2_T1D   &lt;- JAK2_T1D    %&gt;% filter(!hm_rsid %in% duplicated_snps_JAK2)
JAK3_T1D   &lt;- JAK3_T1D    %&gt;% filter(!hm_rsid %in% duplicated_snps_JAK3)
IL6ST_T1D  &lt;- IL6ST_T1D %&gt;% filter(!hm_rsid %in% duplicated_snps_IL6ST)
IL6R_T1D   &lt;- IL6R_T1D    %&gt;% filter(!hm_rsid %in% duplicated_snps_IL6R)
IL2RB_T1D  &lt;- IL2RB_T1D   %&gt;% filter(!hm_rsid %in% duplicated_snps_IL2RB)
IL2RG_T1D  &lt;- IL2RG_T1D   %&gt;% filter(!hm_rsid %in% duplicated_snps_IL2RG)
IL2RA_T1D  &lt;- IL2RA_T1D   %&gt;% filter(!hm_rsid %in% duplicated_snps_IL2RA)
IFNAR2_T1D &lt;- IFNAR2_T1D  %&gt;% filter(!hm_rsid %in% duplicated_snps_IFNAR2)</code></pre>
</div>
<div id="write-data-in-twosamplemr-format-1" class="section level2">
<h2>Write data in TwoSampleMR format</h2>
<pre class="r"><code>#Change to numeric
rename_for_TwoSampleMR_o  &lt;- function(data, beta, other_allele, effect_allele,
                                      hm_rsid, hm_beta, standard_error, 
                                      hm_effect_allele_frequency,
                                      hm_effectAllele, hm_otherAllele,
                                      p_value, hm_pos) {
  data %&gt;%
  select(-beta, -other_allele, -effect_allele) %&gt;%
  rename(SNP = hm_rsid, beta = hm_beta, se = standard_error, eaf = hm_effect_allele_frequency,
         effect_allele = hm_effect_allele, other_allele = hm_other_allele,
         pval = p_value, pos = hm_pos) %&gt;%
  mutate(Phenotype = &quot;Risk of type 1 diabetes&quot;,
         ncase = 18942, ncontrol = 501638, samplesize = 18942+501638)
  }

IL12B_T1D   &lt;- IL12B_T1D  %&gt;% rename_for_TwoSampleMR_o()
IL2RA_T1D   &lt;- IL2RA_T1D  %&gt;% rename_for_TwoSampleMR_o()
IL2RB_T1D   &lt;- IL2RB_T1D  %&gt;% rename_for_TwoSampleMR_o()
IL2RG_T1D   &lt;- IL2RG_T1D  %&gt;% rename_for_TwoSampleMR_o()
IL6ST_T1D   &lt;- IL6ST_T1D  %&gt;% rename_for_TwoSampleMR_o()
IL6R_T1D    &lt;- IL6R_T1D   %&gt;% rename_for_TwoSampleMR_o()
JAK1_T1D    &lt;- JAK1_T1D   %&gt;% rename_for_TwoSampleMR_o()
JAK2_T1D    &lt;- JAK2_T1D   %&gt;% rename_for_TwoSampleMR_o()
JAK3_T1D    &lt;- JAK3_T1D   %&gt;% rename_for_TwoSampleMR_o()
TYK2_T1D    &lt;- TYK2_T1D   %&gt;% rename_for_TwoSampleMR_o()
IFNAR2_T1D  &lt;- IFNAR2_T1D %&gt;% rename_for_TwoSampleMR_o()</code></pre>
<p>Write data in TwoSampleMR data format</p>
<pre class="r"><code>IL12B_T1D  %&gt;% 
  export(&quot;data/export/IL12B_T1D_TwoSampleMR.csv&quot;)
IL2RA_T1D  %&gt;% 
  export(&quot;data/export/IL2RA_T1D_TwoSampleMR.csv&quot;)
IL2RB_T1D  %&gt;% 
  export(&quot;data/export/IL2RB_T1D_TwoSampleMR.csv&quot;)
IL2RG_T1D  %&gt;% 
  export(&quot;data/export/IL2RG_T1D_TwoSampleMR.csv&quot;)
IL6ST_T1D  %&gt;%
  export(&quot;data/export/IL6ST_T1D_TwoSampleMR.csv&quot;)
IL6R_T1D  %&gt;%
  export(&quot;data/export/IL6R_T1D_TwoSampleMR.csv&quot;)
JAK1_T1D   %&gt;% 
  export(&quot;data/export/JAK1_T1D_TwoSampleMR.csv&quot;)
JAK2_T1D   %&gt;% 
  export(&quot;data/export/JAK2_T1D_TwoSampleMR.csv&quot;)
JAK3_T1D   %&gt;% 
  export(&quot;data/export/JAK3_T1D_TwoSampleMR.csv&quot;)
TYK2_T1D   %&gt;% 
  export(&quot;data/export/TYK2_T1D_TwoSampleMR.csv&quot;)
IFNAR2_T1D   %&gt;% 
  export(&quot;data/export/IFNAR2_T1D_TwoSampleMR.csv&quot;)</code></pre>
<pre class="r"><code>#Sanity check: no warnings

# read_outcome_data(&quot;data/export/IL12B_T1D_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_outcome_data(&quot;data/export/IL2RA_T1D_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_outcome_data(&quot;data/export/IL2RB_T1D_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_outcome_data(&quot;data/export/IL2RG_T1D_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_outcome_data(&quot;data/export/IL6ST_T1D_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_outcome_data(&quot;data/export/IL6R_T1D_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_outcome_data(&quot;data/export/JAK1_T1D_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_outcome_data(&quot;data/export/JAK2_T1D_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_outcome_data(&quot;data/export/JAK3_T1D_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_outcome_data(&quot;data/export/TYK2_T1D_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)
# read_outcome_data(&quot;data/export/IFNAR2_T1D_TwoSampleMR.csv&quot;, sep = &quot;,&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 14393)

Matrix products: default


locale:
[1] LC_COLLATE=Finnish_Finland.1252  LC_CTYPE=Finnish_Finland.1252   
[3] LC_MONETARY=Finnish_Finland.1252 LC_NUMERIC=C                    
[5] LC_TIME=Finnish_Finland.1252    

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] dtplyr_1.3.1      data.table_1.14.8 TwoSampleMR_0.5.6 here_1.0.1       
 [5] rio_0.5.29        lubridate_1.9.2   forcats_1.0.0     stringr_1.5.0    
 [9] dplyr_1.1.2       purrr_1.0.1       readr_2.1.4       tidyr_1.3.0      
[13] tibble_3.2.1      ggplot2_3.4.2     tidyverse_2.0.0   workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] gtable_0.3.3     xfun_0.39        bslib_0.4.2      processx_3.8.1  
 [5] callr_3.7.3      tzdb_0.3.0       vctrs_0.6.2      tools_4.3.0     
 [9] ps_1.7.5         generics_0.1.3   curl_5.0.0       fansi_1.0.4     
[13] pacman_0.5.1     pkgconfig_2.0.3  readxl_1.4.2     lifecycle_1.0.3 
[17] compiler_4.3.0   git2r_0.32.0     munsell_0.5.0    getPass_0.2-2   
[21] httpuv_1.6.9     htmltools_0.5.5  sass_0.4.6       yaml_2.3.7      
[25] later_1.3.1      pillar_1.9.0     jquerylib_0.1.4  whisker_0.4.1   
[29] cachem_1.0.8     tidyselect_1.2.0 zip_2.3.0        digest_0.6.31   
[33] stringi_1.7.12   rprojroot_2.0.3  fastmap_1.1.1    grid_4.3.0      
[37] colorspace_2.1-0 cli_3.6.1        magrittr_2.0.3   utf8_1.2.3      
[41] foreign_0.8-84   withr_2.5.0      scales_1.2.1     promises_1.2.0.1
[45] timechange_0.2.0 rmarkdown_2.21   httr_1.4.5       cellranger_1.1.0
[49] hms_1.1.3        openxlsx_4.2.5.2 evaluate_0.20    knitr_1.42      
[53] haven_2.5.2      rlang_1.1.1      Rcpp_1.0.10      glue_1.6.2      
[57] rstudioapi_0.14  jsonlite_1.8.4   R6_2.5.1         fs_1.6.2        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
