<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tea Heikkilä, Emilia Kaiser &amp; Jaakko Koskenniemi" />

<meta name="date" content="2022-05-03" />

<title>JAK1 eQTL</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">T1DSCREEN</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="Harmonize.html">Harmonize</a>
</li>
<li>
  <a href="IL2RA.html">IL2RA</a>
</li>
<li>
  <a href="IL2RB.html">IL2RB</a>
</li>
<li>
  <a href="IL2RG.html">IL2RG</a>
</li>
<li>
  <a href="IL6R.html">IL6R</a>
</li>
<li>
  <a href="CRP.html">CRP</a>
</li>
<li>
  <a href="IL12B.html">IL12B</a>
</li>
<li>
  <a href="JAK2.html">JAK2</a>
</li>
<li>
  <a href="TNF.html">TNF</a>
</li>
<li>
  <a href="TYK2.html">TYK2</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jkoskenniemi/T1DSCREEN">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">JAK1 eQTL</h1>
<h4 class="author">Tea Heikkilä, Emilia Kaiser &amp; Jaakko
Koskenniemi</h4>
<h4 class="date">2022-05-03</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-05-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>T1DSCREEN/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220313code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220313)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220313code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220313)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjkoskenniemiT1DSCREENtree4e1528f570aca3989296d5e39117043fa146fa69targetblank4e1528fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/tree/4e1528f570aca3989296d5e39117043fa146fa69" target="_blank">4e1528f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomjkoskenniemiT1DSCREENtree4e1528f570aca3989296d5e39117043fa146fa69targetblank4e1528fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/jkoskenniemi/T1DSCREEN/tree/4e1528f570aca3989296d5e39117043fa146fa69" target="_blank">4e1528f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/README_cis_eqtl.txt
    Ignored:    data/README_cis_eqtl_AF.txt
    Ignored:    data/cis-EQTL-AF.txt
    Ignored:    data/cis_sumstats/2018-07-18_SNP_AF_for_AlleleB_combined_allele_counts_and_MAF_pos_added.txt
    Ignored:    refs/

Untracked files:
    Untracked:  data/cis_eqtl_sumstats/
    Untracked:  data/cis_sumstats/JAK1_T1D.rds
    Untracked:  data/cis_sumstats/JAK3_T1D.rds
    Untracked:  data/export/IFNAR2_eqtl_TwoSampleMR.csv
    Untracked:  data/export/IL2RA_eqtl_TwoSampleMR.csv
    Untracked:  data/export/IL2RB_eqtl_TwoSampleMR.csv
    Untracked:  data/export/IL2RG_eqtl_TwoSampleMR.csv
    Untracked:  data/export/IL6R_eqtl_TwoSampleMR.csv
    Untracked:  data/export/JAK1_T1D_TwoSampleMR.csv
    Untracked:  data/export/JAK1_eqtl_TwoSampleMR.csv
    Untracked:  data/export/JAK2_eqtl_TwoSampleMR.csv
    Untracked:  data/export/JAK3_T1D_TwoSampleMR.csv
    Untracked:  data/export/JAK3_eqtl_TwoSampleMR.csv
    Untracked:  data/export/TYK2_eqtl_TwoSampleMR.csv

Unstaged changes:
    Modified:   .gitignore
    Modified:   analyses/CRP.Rmd
    Deleted:    analyses/CXCL10.Rmd
    Deleted:    analyses/TNF.Rmd
    Modified:   analyses/_site.yml
    Modified:   code/linux-shell-script-wget-and-decompress.txt
    Deleted:    data/IL6R_crp_LDmat-2023-04-25.RData

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analyses/JAK1.Rmd</code>) and HTML
(<code>docs/JAK1.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/4e1528f570aca3989296d5e39117043fa146fa69/analyses/JAK1.Rmd" target="_blank">4e1528f</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-05-03
</td>
<td>
First commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Load packages and import data</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>-- Attaching packages --------------------------------------- tidyverse 1.3.1 --</code></pre>
<pre><code>v ggplot2 3.4.1     v purrr   0.3.4
v tibble  3.1.7     v dplyr   1.0.9
v tidyr   1.2.0     v stringr 1.4.0
v readr   2.1.2     v forcats 0.5.1</code></pre>
<pre><code>Warning: package &#39;ggplot2&#39; was built under R version 4.2.2</code></pre>
<pre><code>-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(coloc)</code></pre>
<pre><code>This is a new update to coloc.</code></pre>
<pre class="r"><code>library(TwoSampleMR)</code></pre>
<pre><code>TwoSampleMR version 0.5.6 
[&gt;] New: Option to use non-European LD reference panels for clumping etc
[&gt;] Some studies temporarily quarantined to verify effect allele
[&gt;] See news(package=&#39;TwoSampleMR&#39;) and https://gwas.mrcieu.ac.uk for further details</code></pre>
<pre class="r"><code>library(ggpubr)
library(ieugwasr)</code></pre>
<pre><code>API: public: http://gwas-api.mrcieu.ac.uk/</code></pre>
<pre><code>
Attaching package: &#39;ieugwasr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:TwoSampleMR&#39;:

    ld_matrix</code></pre>
<pre class="r"><code>library(rio)</code></pre>
<pre><code>Warning: package &#39;rio&#39; was built under R version 4.2.2</code></pre>
<pre class="r"><code>library(here)</code></pre>
<pre><code>Warning: package &#39;here&#39; was built under R version 4.2.2</code></pre>
<pre><code>here() starts at C:/Users/jajoko/Documents/T1DSCREEN</code></pre>
<pre class="r"><code>here::i_am(&quot;analyses/JAK1.Rmd&quot;)</code></pre>
<pre><code>here() starts at C:/Users/jajoko/Documents/T1DSCREEN</code></pre>
<pre class="r"><code>JAK1_eqtl &lt;- read_exposure_data(&quot;data/export/JAK1_eqtl_TwoSampleMR.csv&quot;, sep=&quot;,&quot;)</code></pre>
<pre><code>Warning in format_data(as.data.frame(exposure_dat), type = &quot;exposure&quot;, snps =
NULL, : se column is not numeric. Coercing...</code></pre>
<pre><code>Warning in format_data(as.data.frame(exposure_dat), type = &quot;exposure&quot;, snps = NULL, : The following SNP(s) are missing required information for the MR tests and will be excluded
rs310233
rs310235
rs310244
rs2780891
rs167454
rs2780816
rs310227
rs310241
rs2780885
rs2780902
rs2780819
rs310213
rs2780822
rs310215
rs310225
rs310219
rs2780820
rs310216
rs2780895
rs2230588
rs2780818
rs831578
rs12063205
rs2132754
rs6672608
rs1041624
rs974019
rs1926310
rs872800
rs17127114
rs11208534
rs2780896
rs11208535
rs7550737
rs80309750
rs7525150
rs1553025
rs3915922
rs310197
rs310204
rs310256
rs310257
rs310255
rs3790545
rs2780833
rs310196
rs11208536
rs310252
rs310249
rs310250
rs310254
rs310208
rs36121854
rs2780831
rs928509
rs310231
rs310209
rs1048007
rs310230
rs310198
rs1556561
rs3790529
rs142102613
rs12027095
rs76286105
rs11208523
rs2254002
rs6588104
rs58510811
rs10493370
rs10493371
rs3790527
rs74570412
rs2780824
rs45598436
rs12022913
rs12353950
rs12029333
rs4915675
rs11208519
rs310245
rs11208518
rs12049564
rs6671355
rs12038159
rs7528403
rs17126970
rs10493368
rs12043554
rs10889498
rs12739954
rs12023256
rs6661723
rs1039125
rs12040026
rs56125796
rs10493367
rs17390055
rs4313429
rs310211
rs10889511
rs11208520
rs2780899
rs10493369
rs310229
rs12129819
rs998125
rs188698
rs310221
rs2780894
rs310214
rs310218
rs11208521
rs310217
rs1926312
rs310222
rs310207
rs310220
rs310224
rs2780827
rs310212
rs310228
rs10789166
rs11208577
rs2256443
rs12738806
rs310236
rs7549445
rs2780900
rs2780889
rs74942083
rs2780828
rs310223
rs10889517
rs76170379
rs3790538
rs2375474
rs12093269
rs6696692
rs4916005
rs182410506
rs7536540
rs2780817
rs12132206
rs12130991
rs6680481
rs12140511
rs12119466
rs12133255
rs12730549
rs4244165
rs4916014
rs10493372
rs12023040
rs12127284
rs67806955
rs79462686
rs478665
rs114269697
rs71647464
rs11208547
rs12569186
rs10493366
rs483317
rs56301536
rs11591096
rs11208550
rs561624
rs11208556
rs12127031
rs55947781
rs11208552
rs193114939
rs12046291
rs72675462
rs1039124
rs9645414
rs12752552
rs7546545
rs71645600
rs1497058
rs7546535
rs6663109
rs71647463
rs114008171
rs11585646
rs11576886
rs11208537
rs3737139
rs520179
rs17390852
rs10889512
rs6659146
rs6696345
rs2230586
rs72675448
rs17392542
rs11585932
rs12730896
rs11208530
rs11579283
rs147897519
rs74737950
rs72675467
rs55882186
rs72675451
rs72675457
rs2151572
rs2027053
rs12145200
rs971185
rs12563230
rs12562008
rs12563017
rs17127024
rs12563818
rs1826869
rs78825027
rs569581
rs11582696
rs12409178
rs72675478
rs185370982
rs1530887
rs12046636
rs7550068
rs72675438
rs2274945
rs189567887
rs578271
rs187299576
rs7539178
rs2256298
rs560773
rs576563
rs473780
rs77916841
rs11208532
rs7548666
rs114803872
rs80091449
rs529715
rs11579616
rs3014973
rs138138409
rs581239
rs542493
rs521188
rs185308582
rs528645
rs4916000
rs10493365
rs55989126
rs3790528
rs12028127
rs12043128
rs55850447
rs11808379
rs12029604
rs2935411
rs12044036
rs12038861
rs578481
rs186636583
rs12046407
rs11208584
rs542232
rs12047654
rs9633416
rs2274369
rs11208559
rs12037590
rs522815
rs11208531
rs11587486
rs11589788
rs10889502
rs17126987
rs12028020
rs571625
rs138374822
rs11588397
rs12402976
rs10493364
rs546595
rs78019881
rs12046987
rs60879425
rs489984
rs141116494
rs143400726
rs1390956
rs150027192
rs146453501
rs1766963
rs1887050
rs17389467
rs1497056
rs2780890
rs72675421
rs72675425
rs114615861
rs72675431
rs1887049
rs17389523
rs12409304
rs11584976
rs17389446
rs11208515
rs60545384
rs11208582
rs11581138
rs150147300
rs61032917
rs72679422
rs10789164
rs553002
rs2148323
rs1926304
rs3014974
rs181762394
rs6686274
rs12411195
rs12411196
rs72681327
rs6698882
rs61779779
rs11208516
rs149811799
rs150613722
rs150658359
rs181339352
rs72681330
rs141245109
rs11208506
rs6673673
rs4916011
rs499870
rs12568330
rs72675417
rs11208504
rs17391535
rs149037162
rs513857
rs41285408
rs17391500
rs75951234
rs145639273
rs10399745
rs6588109
rs151119857
rs58069373
rs12047470
rs579247
rs2935413
rs2935416
rs113789200
rs1983300
rs1926305
rs495588
rs17127169
rs72675415
rs12184322
rs11208581
rs1926309
rs78969683
rs67301710
rs61786113
rs1832442
rs4638162
rs550457
rs12407409
rs61784755
rs17127300
rs11208510
rs12730021
rs6588131
rs79593034
rs11208505
rs151042582
rs7549121
rs10889495
rs7516040
rs17127287
rs6696024
rs68112935
rs17127309
rs11579758
rs61784747
rs61786112
rs6588111
rs35741972
rs2935410
rs12027009
rs3844055
rs61786114
rs143038101
rs11208397
rs61784748
rs12130010
rs12408428
rs6696043
rs150123668
rs35623535
rs77762686
rs12408753
rs9436748
rs11208576
rs139708812
rs72675408
rs74841357
rs140798223
rs61784752
rs76454749
rs75359229
rs11208558
rs138579423
rs138945870
rs1983299
rs2780886
rs12408613
rs115470838
rs3014976
rs12402106
rs116125196
rs61784751
rs4916041
rs7534350
rs116894830
rs7520568
rs184777984
rs61784754
rs76055700
rs149301501
rs192055467
rs6673343
rs61786882
rs58870302
rs140984483
rs310247
rs184316561
rs12024635
rs11208554
rs6699671
rs1338516
rs55889764
rs3014975
rs78622333
rs7539994
rs61784743
rs3920617
rs1536171
rs61779780
rs6588129
rs78772779
rs2780901
rs35404763
rs3861946
rs147238114
rs7520742
rs534697
rs60803934
rs72673348
rs58367021
rs17397674
rs28599582
rs11208585
rs61784742
rs75686825
rs12030772
rs7520723
rs6663936
rs11208402
rs111313341
rs77223187
rs3014983
rs17393565
rs489500
rs184083406
rs11208401
rs6663598
rs149815342
rs72675414
rs74559196
rs116537847
rs75056797
rs142466008
rs12123592
rs12042059
rs12046741
rs525563
rs6698805
rs189151284
rs187323548
rs140879893
rs143719981
rs115706445
rs113926857
rs12404238
rs35112334
rs6704167
rs17127327
rs474568
rs143127987
rs554455
rs181350652
rs74838363
rs75967631
rs142487295
rs115833204
rs192480574
rs1327115
rs11208420
rs7529650
rs79305686
rs76635564
rs555146
rs139181902
rs6702545
rs142142785
rs1926308
rs10458572
rs1327120
rs140517619
rs12408851
rs10127731
rs4915938
rs2104503
rs11806971
rs1590640
rs11804481
rs17127680
rs17357400
rs111776826
rs141985781
rs61779782
rs11811946
rs11799558
rs71647497
rs12409877
rs115497103
rs113108966
rs7554485
rs61779781
rs189793921
rs150428831
rs825242
rs189601145
rs6660780
rs75380295
rs34439343
rs79728418
rs6660481
rs191380440
rs182285239
rs1887285
rs139223816
rs138926292
rs183657291
rs149201299
rs17127618
rs57484022
rs142603322
rs10789171
rs147540371
rs114327999
rs4636508
rs56353628
rs140512324
rs72677684
rs185990124
rs6682237
rs9436747
rs2935412
rs6657868
rs139874155
rs148720858
rs1570871
rs6686041
rs305547
rs825241
rs4915688
rs114683222
rs10889503
rs305549
rs12086738
rs11208538
rs72924268
rs184451770
rs182526754
rs114896931
rs112546671
rs17395035
rs74874110
rs473647
rs7542824
rs12745769
rs11208586
rs61781378
rs61779815
rs34103804
rs1566007
rs17412175
rs1329783
rs11208539
rs12406548
rs6664237
rs11208562
rs17128024
rs6668007
rs1436745
rs72679445
rs17127529
rs150805208
rs61779772
rs12758390
rs145728574
rs34476153
rs6662076
rs1874406
rs4916004
rs151329638
rs1536172
rs6661742
rs12129709
rs75081287
rs4469745
rs142854459
rs7528114
rs12402567
rs12025819
rs17359018
rs74860851
rs61765448
rs77917967
rs11208418
rs111392636
rs41312644
rs6674395
rs77492428
rs34981025
rs310201
rs12563191
rs140729586
rs17388633
rs310202
rs646443
rs10493361
rs12729349
rs1546187
rs75617507
rs4916007
rs7545743
rs10889515
rs12563451
rs34692920
rs1576388
rs189003683
rs145487715
rs12405010
rs79745672
rs149751078
rs11208409
rs75465244
rs11208408
rs55889533
rs72679419
rs2025803
rs6660166
rs11208410
rs17321902
rs71647473
rs72679433
rs4915946
rs1021566
rs7534739
rs10889474
rs11589721
rs7541665
rs4468199
rs12121669
rs151338194
rs79763246
rs10889475
rs6691140
rs17321951
rs1536467
rs1546186
rs11208411
rs310200
rs3762311
rs139321481
rs12727259
rs6673087
rs17397653
rs17384589
rs6690750
rs12091366
rs12403596
rs72920202
rs11208396
rs6691638
rs7523783
rs11208416
rs10889499
rs191026017
rs111891077
rs72673400
rs150852956
rs957576
rs141989857
rs147282315
rs17412368
rs147282262
rs1874405
rs34347295
rs1045895
rs77369072
rs72677698
rs11208509
rs2762833
rs17266284
rs10749749
rs72677699
rs189419726
rs11208407
rs500406
rs74860172
rs147171177
rs35211864
rs1496019
rs61779814
rs59266596
rs193026768
rs74911515
rs6588083
rs79508350
rs2375699
rs72679406
rs6698653
rs11578139
rs115561196
rs1408416
rs112516382
rs55655800
rs72673399
rs11208398
rs61784703
rs11576180
rs61784705
rs17126925
rs57712220
rs2025805
rs17750
rs114280901
rs146211259
rs17412403
rs7531110
rs147732540
rs17127141
rs75</code></pre>
<pre><code>Warning in format_data(as.data.frame(exposure_dat), type = &quot;exposure&quot;, snps =
NULL, : None of the provided SNPs can be used for MR analysis, they are missing
required information.</code></pre>
<pre class="r"><code>JAK1_T1D  &lt;- read_outcome_data(&quot;data/export/JAK1_T1D_TwoSampleMR.csv&quot;, sep=&quot;,&quot;)

JAK1 &lt;- harmonise_data(JAK1_eqtl, JAK1_T1D)</code></pre>
<pre><code>Harmonising Whole blood JAK1 mRNA (4vucPG) and Risk of type 1 diabetes (xN25Oe)</code></pre>
<pre><code>Removing the following SNPs for incompatible alleles:
rs76595887</code></pre>
<pre><code>Removing the following SNPs for being palindromic with intermediate allele frequencies:
rs1028099, rs10493356, rs10789183, rs10789185, rs11208700, rs112314233, rs11587159, rs12037619, rs12091826, rs12127472, rs12407651, rs12564738, rs12568532, rs12758714, rs1327118, rs1333739, rs17412175, rs1892346, rs2274437, rs2801578, rs2801580, rs2840677, rs310239, rs3790436, rs3828036, rs3828040, rs4332387, rs4492587, rs4512686, rs4592284, rs4593866, rs4655539, rs4655598, rs4915688, rs4916041, rs61779319, rs61779814, rs6588136, rs6694927, rs6700226, rs6701146, rs6704418, rs72669385, rs7516965, rs7527706, rs7534350, rs7535217, rs7536558, rs79653480, rs825203, rs955255</code></pre>
<div id="manhattan-plots" class="section level2">
<h2>Manhattan plots</h2>
<pre class="r"><code>#create Manhattan plots with vertical lines indicating the selected area (see below 2.4)
JAK1_eqtl.fig &lt;-
  JAK1 %&gt;%
  ggplot(aes(x = pos.exposure/1000, y=-log10(pval.exposure))) +
  geom_point()+
  geom_hline(yintercept=8, linetype=&quot;dashed&quot;)+
  ggtitle(&quot;GWAS of JAK1 gene expression in blood&quot;) +
  ylab(&quot;-log10(p)&quot;) +
  xlab(NULL)

JAK1_T1D.fig &lt;-
  JAK1 %&gt;%
  ggplot()+
  geom_point(mapping=aes(x = pos.exposure/1000, y=-log10(pval.outcome))) +
  geom_hline(yintercept=8, linetype=&quot;dashed&quot;)+
  # geom_vline(xintercept=36.9e6/1000, linetype=&quot;dashed&quot;)+
  # geom_vline(xintercept=37.3e6/1000, linetype=&quot;dashed&quot;)+
  ggtitle(&quot;GWAS of risk of T1D&quot;) +
  ylab(&quot;-log10(p)&quot;) +
  xlab(&quot;position(kbp)&quot;)

#Obtain range for figure of gene
layer_scales(JAK1_T1D.fig)$x$range$range</code></pre>
<pre><code>[1] 64764.66 66365.30</code></pre>
<pre class="r"><code>JAK1_gene.fig &lt;-
  ggplot(data = JAK1) +
  geom_blank() +
  geom_segment(x=37521.878, xend=37571.094, y=1, yend=1, linewidth = 2) +
  annotate(&quot;text&quot;, label = &quot;JAK1&quot;,
           x= (37521.878 + 37571.094) / 2, y=1, hjust = 0.5, vjust = -1) +
  xlim(36521.878, 38571.094) +
  ylim(0.75, 2) +
  ylab(NULL) + xlab(NULL) +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank())

fig_JAK1_manhattan &lt;- ggarrange(JAK1_gene.fig, JAK1_eqtl.fig, JAK1_T1D.fig,
                                heights = c(1, 3, 3), nrow = 3,
                                ncol = 1, align = &quot;hv&quot;)
# ggsave(&quot;figures/JAK1-manhattan-combined-fig.pdf&quot;,
#        height = 6, width = 8, units = &quot;in&quot;)</code></pre>
</div>
<div id="coloc-assuming-1-causal-variant" class="section level2">
<h2>Coloc assuming 1 causal variant</h2>
<pre class="r"><code>#recode MAF
JAK1 &lt;- JAK1 %&gt;%
  mutate(maf.exposure  = ifelse(eaf.exposure  &lt; 0.5, eaf.exposure,  1-eaf.exposure)) %&gt;%
  mutate(maf.outcome  = ifelse(eaf.outcome  &lt; 0.5, eaf.outcome,  1-eaf.outcome))

colnames(JAK1)</code></pre>
<pre><code> [1] &quot;SNP&quot;                    &quot;effect_allele.exposure&quot; &quot;other_allele.exposure&quot; 
 [4] &quot;effect_allele.outcome&quot;  &quot;other_allele.outcome&quot;   &quot;beta.exposure&quot;         
 [7] &quot;beta.outcome&quot;           &quot;eaf.exposure&quot;           &quot;eaf.outcome&quot;           
[10] &quot;remove&quot;                 &quot;palindromic&quot;            &quot;ambiguous&quot;             
[13] &quot;id.outcome&quot;             &quot;pos.outcome&quot;            &quot;pval.outcome&quot;          
[16] &quot;se.outcome&quot;             &quot;ncase.outcome&quot;          &quot;ncontrol.outcome&quot;      
[19] &quot;samplesize.outcome&quot;     &quot;outcome&quot;                &quot;mr_keep.outcome&quot;       
[22] &quot;pval_origin.outcome&quot;    &quot;data_source.outcome&quot;    &quot;pval.exposure&quot;         
[25] &quot;pos.exposure&quot;           &quot;samplesize.exposure&quot;    &quot;se.exposure&quot;           
[28] &quot;exposure&quot;               &quot;mr_keep.exposure&quot;       &quot;pval_origin.exposure&quot;  
[31] &quot;id.exposure&quot;            &quot;data_source.exposure&quot;   &quot;action&quot;                
[34] &quot;mr_keep&quot;                &quot;maf.exposure&quot;           &quot;maf.outcome&quot;           </code></pre>
<pre class="r"><code>D1 &lt;- list(
  type = &quot;quant&quot;, # quantitative trait
  pvalues = JAK1$pval.exposure,
  N = JAK1$samplesize.exposure,
  MAF = JAK1$maf.exposure,
  pos = JAK1$pos.exposure,
  snp = JAK1$SNP,
  sdY = 1)

D2 &lt;- list(
  type = &quot;cc&quot;, # case-control trait
  Beta = JAK1$beta.outcome,
  varBeta = JAK1$se.outcome,
  pvalues = JAK1$pval.outcome,
  N = 18942+501638, # Case-control study (Chiou et al. 2021 Nature)
  s = 18942/(18942+501638), # N_case/(N_case+ N_ctrl)
  MAF = JAK1$maf.exposure,  #eqtl used here in purpose
  pos = JAK1$pos.outcome,
  snp = JAK1$SNP)

check_dataset(D1)</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>check_dataset(D2)</code></pre>
<pre><code>Warning in check_dataset(D2): minimum p value is: 7.46e-06
If this is what you expected, this is not a problem.
If this is not as small as you expected, please check the 02_data vignette.</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>coloc_JAK1 &lt;- coloc.abf(D1, D2, p1 = 1e-4, p2 = 1e-4, p12 = 1e-5)</code></pre>
<pre><code>Warning in check_dataset(d = dataset2, 2): minimum p value is: 7.46e-06
If this is what you expected, this is not a problem.
If this is not as small as you expected, please check the 02_data vignette.</code></pre>
<pre><code>PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf 
 4.72e-07  4.48e-01  5.79e-07  5.49e-01  2.94e-03 
[1] &quot;PP abf for shared variant: 0.294%&quot;</code></pre>
<pre class="r"><code>sensitivity(coloc_JAK1, &quot;H4 &gt; 0.7&quot;)</code></pre>
<pre><code>Results fail decision rule H4 &gt; 0.7</code></pre>
<p><img src="figure/JAK1.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 14393)

Matrix products: default

locale:
[1] LC_COLLATE=Finnish_Finland.1252  LC_CTYPE=Finnish_Finland.1252   
[3] LC_MONETARY=Finnish_Finland.1252 LC_NUMERIC=C                    
[5] LC_TIME=Finnish_Finland.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] here_1.0.1        rio_0.5.29        ieugwasr_0.1.5    ggpubr_0.4.0     
 [5] TwoSampleMR_0.5.6 coloc_5.1.0.1     data.table_1.14.2 forcats_0.5.1    
 [9] stringr_1.4.0     dplyr_1.0.9       purrr_0.3.4       readr_2.1.2      
[13] tidyr_1.2.0       tibble_3.1.7      ggplot2_3.4.1     tidyverse_1.3.1  
[17] workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] colorspace_2.0-3   ggsignif_0.6.3     ellipsis_0.3.2     rprojroot_2.0.3   
 [5] fs_1.5.2           rstudioapi_0.13    farver_2.1.1       fansi_1.0.3       
 [9] lubridate_1.8.0    xml2_1.3.3         codetools_0.2-18   splines_4.2.1     
[13] cachem_1.0.6       knitr_1.39         jsonlite_1.8.0     broom_1.0.0       
[17] dbplyr_2.2.1       compiler_4.2.1     httr_1.4.3         backports_1.4.1   
[21] assertthat_0.2.1   Matrix_1.4-1       fastmap_1.1.0      cli_3.6.0         
[25] later_1.3.0        htmltools_0.5.2    tools_4.2.1        gtable_0.3.0      
[29] glue_1.6.2         Rcpp_1.0.8.3       carData_3.0-5      cellranger_1.1.0  
[33] jquerylib_0.1.4    vctrs_0.5.2        iterators_1.0.14   xfun_0.31         
[37] ps_1.7.1           openxlsx_4.2.5.1   rvest_1.0.2        lifecycle_1.0.3   
[41] irlba_2.3.5        rstatix_0.7.0      getPass_0.2-2      scales_1.2.0      
[45] hms_1.1.1          promises_1.2.0.1   susieR_0.12.16     yaml_2.3.5        
[49] curl_4.3.2         gridExtra_2.3      sass_0.4.2         reshape_0.8.9     
[53] stringi_1.7.6      highr_0.9          foreach_1.5.2      nortest_1.0-4     
[57] zip_2.2.0          shape_1.4.6        rlang_1.0.6        pkgconfig_2.0.3   
[61] matrixStats_0.62.0 evaluate_0.15      lattice_0.20-45    labeling_0.4.2    
[65] cowplot_1.1.1      processx_3.6.1     tidyselect_1.1.2   plyr_1.8.7        
[69] magrittr_2.0.3     R6_2.5.1           generics_0.1.3     DBI_1.1.3         
[73] pillar_1.8.0       haven_2.5.0        whisker_0.4        foreign_0.8-82    
[77] withr_2.5.0        survival_3.3-1     abind_1.4-5        mixsqp_0.3-43     
[81] modelr_0.1.8       crayon_1.5.1       car_3.1-0          utf8_1.2.2        
[85] tzdb_0.3.0         rmarkdown_2.14     viridis_0.6.2      grid_4.2.1        
[89] readxl_1.4.0       callr_3.7.1        git2r_0.30.1       mr.raps_0.2       
[93] reprex_2.0.1       digest_0.6.29      httpuv_1.6.5       munsell_0.5.0     
[97] glmnet_4.1-4       viridisLite_0.4.0  bslib_0.4.0       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
