<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2022-10-31" />

<title>JAK2</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Growth-T1D</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">JAK2</h1>
<h4 class="date">2022-10-31</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-05
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Rotaatio-2022/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220313code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220313)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220313code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220313)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstronge86595d">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> e86595d
</a>
</p>
</div>
<div id="strongRepositoryversionstronge86595d"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version e86595d.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analyses/.Rhistory

Untracked files:
    Untracked:  .Rprofile
    Untracked:  .gitattributes
    Untracked:  .gitignore
    Untracked:  CRP.Rmd
    Untracked:  Functional_variants.pdf
    Untracked:  IL2RB.Rmd
    Untracked:  IL6R.Rmd
    Untracked:  IL6ST.Rmd
    Untracked:  MONDO_0005147-known-drugs.tsv
    Untracked:  MR-articles-pubmed.csv
    Untracked:  Mimicking-randomized-controlled-trials-with-genome-wide-data-to-ritta.pptx
    Untracked:  Mimicking-randomized-controlled-trials-with-genome-wide-data.pptx
    Untracked:  README.md
    Untracked:  README.txt
    Untracked:  TNF.Rmd
    Untracked:  _workflowr.yml
    Untracked:  analyses/_site.yml
    Untracked:  analyses/about.Rmd
    Untracked:  analyses/index.Rmd
    Untracked:  analyses/license.Rmd
    Untracked:  articles/
    Untracked:  code/
    Untracked:  data/
    Untracked:  figures/
    Untracked:  guidelines-for-review-of-MR-studies-20177.pdf
    Untracked:  ld_finnge.zip
    Untracked:  mendelian randomization.Rproj
    Untracked:  output/

Unstaged changes:
    Deleted:    analyses/IL6R.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analyses/JAK2.Rmd</code>) and HTML
(<code>docs/JAK2.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
e86595d
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2022-11-05
</td>
<td>
wflow_publish(c("analyses/CRP.Rmd", "analyses/CXCL10.Rmd",
"analyses/IL2RA.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Load packages</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>-- Attaching packages --------------------------------------- tidyverse 1.3.1 --</code></pre>
<pre><code>v ggplot2 3.3.6     v purrr   0.3.4
v tibble  3.1.7     v dplyr   1.0.9
v tidyr   1.2.0     v stringr 1.4.0
v readr   2.1.2     v forcats 0.5.1</code></pre>
<pre><code>-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(coloc)</code></pre>
<pre><code>This is a new update to coloc.</code></pre>
<pre class="r"><code>library(TwoSampleMR)</code></pre>
<pre><code>TwoSampleMR version 0.5.6 
[&gt;] New: Option to use non-European LD reference panels for clumping etc
[&gt;] Some studies temporarily quarantined to verify effect allele
[&gt;] See news(package=&#39;TwoSampleMR&#39;) and https://gwas.mrcieu.ac.uk for further details</code></pre>
<pre class="r"><code>library(ggpubr)
library(ieugwasr)</code></pre>
<pre><code>API: public: http://gwas-api.mrcieu.ac.uk/</code></pre>
<pre><code>
Attaching package: &#39;ieugwasr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:TwoSampleMR&#39;:

    ld_matrix</code></pre>
<div id="jak2" class="section level1">
<h1>1 JAK2</h1>
<div id="obtain-data" class="section level2">
<h2>1.1 Obtain data</h2>
<p>GWAS sumstat of type 1 diabetes was obtained from the study by Chiou
et al. (Nature 2021, <a
href="https://doi.org/10.1038/s41586-021-03552-w"
class="uri">https://doi.org/10.1038/s41586-021-03552-w</a>). GWAS
sumstat for protein levels of JAK2 was obtained from the study by
Ferkingstad et al. (Nature 2021, <a
href="https://doi.org/10.1038/s41588-021-00978-w"
class="uri">https://doi.org/10.1038/s41588-021-00978-w</a>). Variants
within 1 megabase before the start site of JAK2 and 1 megabase after the
stop site are included.</p>
<pre class="r"><code>JAK2_gwas &lt;- readRDS(&quot;data/sumstats/cisdata/JAK2_gwas_data.rds&quot;) %&gt;% tibble()
JAK2_anno &lt;- readRDS(&quot;data/sumstats/cisdata/JAK2_anno_data.rds&quot;) %&gt;% tibble()
JAK2_T1D &lt;- readRDS(&quot;data/sumstats/cisdata/JAK2_T1D_data_harm.rds&quot;) %&gt;% tibble()</code></pre>
<p>Ferkingstad et al. state in their README file</p>
<blockquote>
<p>Note that in the summary statistics files, the effectAllele is not
always the minor allele, and therefore the ImpMAF is not always the
frequency of the effect allele (it may be the frequency of the other
allele). We now provide an Extra annotation file
(assocvariants.annotated.txt.gz) that also includes the effect allele
frequency – This file can be mapped to the other files using the “Name”
column. Also, the summary statistics files sometimes incorrectly have
effectAllele=otherAllele for multiallelic variants. In these cases the
effectAllele is correct, but the otherAllele should be ‘!’, meaning that
the effectAllele is tested against the other (two or more) alleles
(using the ‘!’ sign as shorthand for ‘not effectAllele’). This has been
corrected in the file Extra annotation file
(assocvariants.annotated.txt.gz).</p>
</blockquote>
<blockquote>
<p>Finally, a subset of the variants in the summary statistics files
should be excluded due to quality issues. These variants are listed in a
separate Excluded variants file (assocvariants.excluded.txt.gz). The
Extra annotation file (assocvariants.annotated.txt.gz) does not include
these variants.</p>
</blockquote>
<pre class="r"><code>#merge data (exclude those observations that are missing in anno)
JAK2_gwas &lt;- right_join(JAK2_gwas, JAK2_anno, by = &quot;Name&quot;)
rm(JAK2_anno)</code></pre>
<p>All cases where the variables in the two data files (annotation and
original sumstat of Ferkingstad) are either instances when - rsid has
not been assigned (and the original sumstat has NA and the annotation
“.”) - multiallelic allele, where the other allele is annotated with a
“!”</p>
<p>We decided to remove the multiallelic SNPs anyway, and after them,
the columns were identical (equal and not discordant in terms of missing
observations).</p>
<pre class="r"><code>#Remove multiallelic snps and recoed rsid &quot;.&quot; to NA
JAK2_gwas &lt;- JAK2_gwas %&gt;%
  mutate(rsids.y = replace(rsids.y, rsids.y == &quot;.&quot;, NA)) %&gt;%
  filter(otherAllele.y != &quot;!&quot;)

#Chrom, Pos, Rsids, effectAllele and otherAllele are the same in different files
JAK2_gwas %&gt;%
  filter(Chrom.x != Chrom.y | (!is.na(Chrom.x) &amp; is.na(Chrom.y)) |  (!is.na(Chrom.y) &amp; is.na(Chrom.x)) |
           Pos.x != Pos.y | (!is.na(Pos.x) &amp; is.na(Pos.y)) |  (!is.na(Pos.y) &amp; is.na(Pos.x)) |
           rsids.x != rsids.y | (!is.na(rsids.x) &amp; is.na(rsids.y)) |  (!is.na(rsids.y) &amp; is.na(rsids.x)) |
           effectAllele.x != effectAllele.y | (!is.na(effectAllele.x) &amp; is.na(effectAllele.y)) |  (!is.na(effectAllele.y) &amp; is.na(effectAllele.x))|
           otherAllele.x != otherAllele.y | (!is.na(otherAllele.x) &amp; is.na(otherAllele.y)) |  (!is.na(otherAllele.y) &amp; is.na(otherAllele.x)))</code></pre>
<pre><code># A tibble: 3 x 18
  Chrom.x Pos.x Name     rsids.x effec~1 other~2  Beta  Pval minus~3    SE     N
  &lt;chr&gt;   &lt;int&gt; &lt;chr&gt;    &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;int&gt;
1 &lt;NA&gt;       NA chr9:41~ &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;       NA    NA      NA    NA    NA
2 &lt;NA&gt;       NA chr9:48~ &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;       NA    NA      NA    NA    NA
3 &lt;NA&gt;       NA chr9:58~ &lt;NA&gt;    &lt;NA&gt;    &lt;NA&gt;       NA    NA      NA    NA    NA
# ... with 7 more variables: ImpMAF &lt;dbl&gt;, Chrom.y &lt;chr&gt;, Pos.y &lt;int&gt;,
#   rsids.y &lt;chr&gt;, effectAllele.y &lt;chr&gt;, otherAllele.y &lt;chr&gt;,
#   effectAlleleFreq &lt;dbl&gt;, and abbreviated variable names 1: effectAllele.x,
#   2: otherAllele.x, 3: minus_log10_pval
# i Use `colnames()` to see all variable names</code></pre>
<p>Thus, only those values variables with .x in their name are
preserved. Here, we also remove the remaining multiallelic SNPs (where
effect allele equals the other allele.)</p>
<pre class="r"><code>#Discard all variables that end with an &quot;.y&quot;
JAK2_gwas &lt;- JAK2_gwas %&gt;% 
  select(!ends_with(&quot;.y&quot;)) #remove column names *.y

#Remove &quot;.x&quot; from variable names
JAK2_gwas &lt;- JAK2_gwas %&gt;% 
  rename_with(~gsub(&quot;.x&quot;, &quot;&quot;, .x, fixed = TRUE)) 

#Remove the other version of the coding of multiallelic variants
JAK2_gwas &lt;- JAK2_gwas %&gt;%
  filter(effectAllele != otherAllele)</code></pre>
</div>
<div id="harmonize-and-check-for-alignment" class="section level2">
<h2>2.2 Harmonize and check for alignment</h2>
<p>Please note that the GWAS sumstat of type 1 diabetes has been
harmonized by GWAS catalog, but the GWAS sumstat for serum JAK2 levels
has not been harmonized.</p>
<pre class="r"><code>#Rename JAK2 variants
JAK2_gwas &lt;- JAK2_gwas %&gt;%
  rename(rsid = rsids,
         EA_prot = effectAllele, NEA_prot = otherAllele,
         EAF_prot=effectAlleleFreq, BETA_prot=Beta,
         p_prot = Pval, 
         minus_log10_p_prot = minus_log10_pval, 
         SE_prot = SE, N_prot = N, ImpMAF_prot = ImpMAF, variant_id_prot = Name)

#Rename columns
JAK2_T1D &lt;- JAK2_T1D %&gt;%
  rename(Chrom = chr, variant_id_T1D = id, EA_T1D = effectAllele, NEA_T1D = otherAllele, BETA_T1D = beta, SE_T1D = SE,
         p_T1D = pval, EAF_T1D = EAF)

#What is the difference between position and base_pair_location?
JAK2_T1D %&gt;%
  filter(pos != base_pair_location) %&gt;%
  select(Chrom, rsid, pos, base_pair_location) %&gt;% 
  print(n=200)  #none</code></pre>
<pre><code># A tibble: 0 x 4
# ... with 4 variables: Chrom &lt;chr&gt;, rsid &lt;chr&gt;, pos &lt;int&gt;,
#   base_pair_location &lt;int&gt;
# i Use `colnames()` to see all variable names</code></pre>
<pre class="r"><code>#What&#39;s the difference between Chrom and chromosome?
JAK2_T1D %&gt;%
  filter(Chrom != chromosome) %&gt;%
  select(Chrom, rsid, pos, base_pair_location) %&gt;% 
  print(n=200)  #none</code></pre>
<pre><code># A tibble: 0 x 4
# ... with 4 variables: Chrom &lt;chr&gt;, rsid &lt;chr&gt;, pos &lt;int&gt;,
#   base_pair_location &lt;int&gt;
# i Use `colnames()` to see all variable names</code></pre>
<pre class="r"><code>#Remove chromosome and base_pair_location
JAK2_T1D &lt;- JAK2_T1D %&gt;% select(-chromosome, -base_pair_location)

#Remove SNPs with multiple rsids
JAK2_gwas &lt;- JAK2_gwas %&gt;% filter(!str_detect(rsid, &quot;,&quot;))


#remove &quot;chr&quot; from the beginning of the JAK2_gwas
JAK2_gwas &lt;- JAK2_gwas %&gt;% 
  mutate(variant_id_prot = str_replace(variant_id_prot, &quot;chr&quot;, &quot;&quot;)) %&gt;%
  mutate(variant_id_prot = str_replace(variant_id_prot, &quot;:&quot;, &quot;_&quot;)) %&gt;%
  mutate(variant_id_prot = str_replace(variant_id_prot, &quot;:&quot;, &quot;_&quot;)) %&gt;%
  mutate(variant_id_prot = str_replace(variant_id_prot, &quot;:&quot;, &quot;_&quot;)) %&gt;%
  mutate(variant_id_prot = str_replace(variant_id_prot, &quot;:&quot;, &quot;_&quot;))

#merge the data
JAK2 &lt;- inner_join(JAK2_gwas, JAK2_T1D, by = &quot;rsid&quot;)

JAK2 %&gt;% 
  filter(Chrom.x != Chrom.y ) %&gt;%
  select(Chrom.x, Chrom.y) %&gt;% print(n=500) #all are the same</code></pre>
<pre><code># A tibble: 0 x 2
# ... with 2 variables: Chrom.x &lt;chr&gt;, Chrom.y &lt;chr&gt;
# i Use `colnames()` to see all variable names</code></pre>
<pre class="r"><code>JAK2 &lt;- JAK2 %&gt;% select(-Chrom.y) %&gt;% rename(Chrom = Chrom.x)


JAK2 %&gt;% colnames()</code></pre>
<pre><code> [1] &quot;Chrom&quot;              &quot;Pos&quot;                &quot;variant_id_prot&quot;   
 [4] &quot;rsid&quot;               &quot;EA_prot&quot;            &quot;NEA_prot&quot;          
 [7] &quot;BETA_prot&quot;          &quot;p_prot&quot;             &quot;minus_log10_p_prot&quot;
[10] &quot;SE_prot&quot;            &quot;N_prot&quot;             &quot;ImpMAF_prot&quot;       
[13] &quot;EAF_prot&quot;           &quot;pos&quot;                &quot;variant_id_T1D&quot;    
[16] &quot;EA_T1D&quot;             &quot;NEA_T1D&quot;            &quot;BETA_T1D&quot;          
[19] &quot;SE_T1D&quot;             &quot;p_T1D&quot;              &quot;EAF_T1D&quot;           </code></pre>
<pre class="r"><code>#Remove those without a rsid
JAK2 &lt;- JAK2 %&gt;% filter(!is.na(rsid))


#examine how many SNPs have disconcordant EAF &lt;40% OR &gt;60% between the two data sets
JAK2 %&gt;%
  filter(!(EAF_T1D &lt; 0.40 &amp; EAF_prot &lt; 0.40) | (EAF_T1D &gt; 0.60 &amp; EAF_prot &gt; 0.60))</code></pre>
<pre><code># A tibble: 1,838 x 21
   Chrom     Pos variant_~1 rsid  EA_prot NEA_p~2 BETA_~3 p_prot minus~4 SE_prot
   &lt;chr&gt;   &lt;int&gt; &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt;   &lt;chr&gt;     &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
 1 chr9  3987329 9_3987329~ rs78~ A       G        0.0104  0.203   0.693 0.00817
 2 chr9  3989685 9_3989685~ rs48~ G       C        0.0978  0.675   0.171 0.233  
 3 chr9  3989789 9_3989789~ rs48~ T       C        0.0044  0.629   0.201 0.00911
 4 chr9  3989993 9_3989993~ rs68~ G       A        0.0046  0.613   0.212 0.00910
 5 chr9  3990092 9_3990092~ rs57~ T       A        0.0045  0.616   0.211 0.00897
 6 chr9  3991729 9_3991729~ rs70~ T       C       -0.0123  0.155   0.811 0.00864
 7 chr9  3994090 9_3994090~ rs52~ A       T        0.0901  0.615   0.211 0.179  
 8 chr9  4002823 9_4002823~ rs50~ A       G       -0.0202  0.125   0.903 0.0132 
 9 chr9  4008954 9_4008954~ rs66~ T       C       -0.0171  0.200   0.699 0.0133 
10 chr9  4023987 9_4023987~ rs15~ T       A        0.0028  0.729   0.137 0.00810
# ... with 1,828 more rows, 11 more variables: N_prot &lt;int&gt;, ImpMAF_prot &lt;dbl&gt;,
#   EAF_prot &lt;dbl&gt;, pos &lt;int&gt;, variant_id_T1D &lt;chr&gt;, EA_T1D &lt;chr&gt;,
#   NEA_T1D &lt;chr&gt;, BETA_T1D &lt;dbl&gt;, SE_T1D &lt;dbl&gt;, p_T1D &lt;chr&gt;, EAF_T1D &lt;dbl&gt;,
#   and abbreviated variable names 1: variant_id_prot, 2: NEA_prot,
#   3: BETA_prot, 4: minus_log10_p_prot
# i Use `print(n = ...)` to see more rows, and `colnames()` to see all variable names</code></pre>
<pre class="r"><code>#JAK2 %&gt;% 
#filter(flipped==1) %&gt;% 
#filter((EAF_prot &gt; 0.4 &amp; EAF_prot &lt; 0.6)  | (EAF_T1D &gt; 0.4 &amp;  EAF_T1D &lt; 0.6)) #%&gt;% select (rsid, EA_prot, EA_T1D, NEA_prot, NEA_T1D, EAF_prot, EAF_T1D) %&gt;% #print(n=200)

# remove SNPs with EAF 40-60%
#JAK2 &lt;- JAK2 %&gt;% filter(!(flipped == 1 &amp; (EAF_prot &gt; 0.4 &amp; EAF_prot &lt; 0.6)  | #(EAF_T1D &gt; 0.4 &amp;  EAF_T1D &lt; 0.6)))</code></pre>
</div>
<div id="manhattan-plots" class="section level2">
<h2>2.3 Manhattan plots</h2>
<pre class="r"><code>#create Manhattan plots with vertical lines indicating the selected area (see below 2.4)
JAK2_prot.fig &lt;- 
  JAK2 %&gt;%
  ggplot(aes(x = Pos/1000, y=minus_log10_p_prot)) +
  geom_point()+
  geom_hline(yintercept=8, linetype=&quot;dashed&quot;)+
  geom_vline(xintercept=4.2e6/1000, linetype=&quot;dashed&quot;)+
  geom_vline(xintercept=4.4e6/1000, linetype=&quot;dashed&quot;)+
  ggtitle(&quot;GWAS of JAK2 in serum&quot;) +
  ylab(&quot;-log10(p)&quot;) +
  xlab(NULL)

JAK2_T1D.fig &lt;- 
  JAK2 %&gt;%
  mutate(minus_log10_p_T1D = -log10(as.numeric(p_T1D))) %&gt;%
  ggplot()+
  geom_point(mapping=aes(x = Pos/1000, y=minus_log10_p_T1D))+
  geom_hline(yintercept=8, linetype=&quot;dashed&quot;)+
  geom_vline(xintercept=4.2e6/1000, linetype=&quot;dashed&quot;)+
  geom_vline(xintercept=4.4e6/1000, linetype=&quot;dashed&quot;)+
  ggtitle(&quot;GWAS of risk of T1D&quot;) +
  ylab(&quot;-log10(p)&quot;) +
  xlab(&quot;position(kbp)&quot;)

layer_scales(JAK2_T1D.fig)$x$range$range</code></pre>
<pre><code>[1] 3985.359 6128.137</code></pre>
<pre class="r"><code>JAK2_gene.fig &lt;- 
  ggplot(data = JAK2) +
  geom_blank() +
  geom_segment(x=4985.033, xend=5128.183, y=1, yend=1, size = 2) +
  annotate(&quot;text&quot;, label = &quot;JAK2&quot;, x= (4985.033 + 5128.183) / 2, y=1, hjust = 0.5, vjust = -1) +
  xlim(4885.033, 5228.183) +
  ylim(0.75, 2) +
  ylab(NULL) + xlab(NULL) +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank(), 
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank())

ggarrange(JAK2_gene.fig, JAK2_prot.fig, JAK2_T1D.fig,
          heights = c(1, 3, 3), nrow = 3,
          ncol = 1, align = &quot;hv&quot;)</code></pre>
<p><img src="figure/JAK2.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(&quot;figures/JAK2-manhattan-combined-fig.pdf&quot;, height = 6, width = 8, units = &quot;in&quot;)</code></pre>
</div>
<div id="coloc-assuming-1-colocalizing-variant" class="section level2">
<h2>2.4 Coloc assuming 1 colocalizing variant</h2>
<pre class="r"><code>#select the area containing significant SNPs
JAK2_close &lt;- JAK2 %&gt;%
  filter(Pos &gt; 4.2e6 &amp; Pos &lt; 4.4e6)

#filter a few multiallelic snps away
duplicated_snps &lt;- JAK2_close %&gt;%
  filter(duplicated(rsid)) %&gt;% pull(rsid)

JAK2_close &lt;- JAK2_close %&gt;%
  filter(!rsid %in% duplicated_snps)

#remove the SNP with MAF=0
JAK2_close &lt;- JAK2_close %&gt;%
  filter(EAF_T1D != 0)

JAK2_close &lt;- JAK2_close %&gt;%
  mutate(MAF_T1D = ifelse(EAF_T1D &gt; 0.5, 1-EAF_T1D, EAF_T1D)) %&gt;%
  mutate(MAF_prot = ifelse(EAF_prot &gt; 0.5, 1-EAF_prot, EAF_prot))

JAK2_close &lt;- JAK2_close %&gt;%
  filter(MAF_T1D != 0 &amp; MAF_prot != 0)

D1 &lt;- list(
  type = &quot;quant&quot;, # quantitative trait
  beta = JAK2_close$BETA_prot,
  varbeta = JAK2_close$SE_prot^2, 
  pvalues = JAK2_close$p_prot,
  N = JAK2_close$N_prot,
  MAF = as.numeric(JAK2_close$MAF_prot),
  pos = JAK2_close$Pos,
  snp = JAK2_close$rsid,
  sdY = 1)

D2 &lt;- list(
  type = &quot;cc&quot;, # case-control trait
  beta = JAK2_close$BETA_T1D,
  varbeta = JAK2_close$SE_T1D^2,
  pvalues = JAK2_close$p_T1D,
  N = 18942+501638, # Case-control study (Chiou et al. 2021 Nature)
  s = 18942/(18942+501638), # N_case/(N_case+ N_ctrl)
  MAF = as.numeric(JAK2_close$MAF_T1D),
  pos = JAK2_close$Pos,
  snp = JAK2_close$rsid)

coloc_results_JAK2 &lt;- coloc.abf(D1, D2, p1 = 1e-4, p2 = 1e-4, p12 = 1e-5)</code></pre>
<pre><code>Warning in check_dataset(d = dataset1, 1): minimum p value is: 0.00076101
If this is what you expected, this is not a problem.
If this is not as small as you expected, please check the 02_data vignette.</code></pre>
<pre><code>PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf 
 2.09e-13  2.55e-14  8.85e-01  1.08e-01  6.91e-03 
[1] &quot;PP abf for shared variant: 0.691%&quot;</code></pre>
<pre class="r"><code>sensitivity(coloc_results_JAK2, &quot;H4 &gt; 0.1&quot;)</code></pre>
<pre><code>Results fail decision rule H4 &gt; 0.1</code></pre>
<p><img src="figure/JAK2.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="susie" class="section level2">
<h2>2.5 SusiE</h2>
<p>Älkää tehkö vielä tätä. Jätetään se kuitenkin tähän vielä jatkoa
varten.</p>
<pre class="r"><code># 
# ldmat_CXCL10 &lt;- ld_matrix_local(
#     IL2RA_T1D_merged$rsid,
#     plink_bin = genetics.binaRies::get_plink_binary(),
#     bfile = &quot;C:/Users/jajoko/Documents/MR Projects/plinkref/EUR&quot;)
# 
# #In LD-mat the rsid are followed by reference and effect alleles (e.g. rs10905492_G_A). The effect alleles have to be harmonized between the LD matrix and data. The data is changed to follow ld matrix, because that is easier to do than vice versa.
# 
# #Extract rsid, non-effect allele and effect allele from LD-matrix
# snps &lt;- data.frame(rsid_nea_ea = rownames(ldmat_IL2RA))
# snps &lt;- snps %&gt;% 
#   mutate(nea_ea = str_sub(rsid_nea_ea, -4,-1))
# 
# #Check that names in LD-matrix do not include indels (eg. G to CT)
# snps %&gt;% mutate(underscore = str_sub(nea_ea, 1, 1)) %&gt;% 
#   filter(underscore != &quot;_&quot;) #all have _ as the fourth last character, indicating that they are indeed single nucleotides
# 
# #Extract ea, nea, and rsid
# snps &lt;- snps %&gt;% 
#   mutate(nea = str_sub(nea_ea, 2,2), ea = str_sub(nea_ea, 4,4)) %&gt;%
#   mutate(rsid = str_sub(rsid_nea_ea, 1, -5))
# 
# 
# #prepare for merge 
# snps &lt;- snps %&gt;% 
#   select(-nea_ea) %&gt;%
#   rename(nea_ldmat = nea, ea_ldmat = ea, rsid_ldmat = rsid_nea_ea, rsid = rsid)
# 
# #Merge
# IL2RA_T1D_merged_ldsnps &lt;- 
#   left_join(snps, IL2RA_T1D_merged, by = &quot;rsid&quot;)
# 
# #check for alignment
# IL2RA_T1D_merged_ldsnps %&gt;% 
#   filter(ea_ldmat != NEA_T1D_aligned) %&gt;% nrow()
# 
# IL2RA_T1D_merged_ldsnps %&gt;% 
#   filter(ea_ldmat == NEA_T1D_aligned) %&gt;% nrow()
# 
# IL2RA_T1D_merged_ldsnps
# 
# 
# #ongelma: kuinka saada ld-matriisin snp:t samaan muotoon kuin IL2RA-datan??
# 
# 
# D1 &lt;- list(
#   type = &quot;quant&quot;, # quantitative trait
#   beta = IL2RA_T1D_merged_onlymat$BETA_prot_aligned,
#   varbeta = IL2RA_T1D_merged_onlymat$SE_prot^2, 
#   pvalues = IL2RA_T1D_merged_onlymat$p_prot,
#   N = IL2RA_T1D_merged_onlymat$N_prot,
#   MAF = as.numeric(IL2RA_T1D_merged_onlymat$MAF_prot),
#   snp = IL2RA_T1D_merged_onlymat$rsid,
#   LD = ldmat_IL2RA,
#   sdY = 1)
# 
# D2 &lt;- list(
#   type = &quot;cc&quot;, # case-control trait
#   beta = IL2RA_T1D_merged_onlymat$BETA_T1D_aligned,
#   varbeta = IL2RA_T1D_merged_onlymat$SE_T1D^2,
#   pvalues = IL2RA_T1D_merged_onlymat$p_T1D,
#   N = IL2RA_T1D_merged_onlymat$N_T1D, # Case-control study (Chiou et al. 2021 Nature)
#   s = 18942/(18942+501638), # N_case/(N_case+ N_ctrl)
#   MAF = as.numeric(IL2RA_T1D_merged_onlymat$MAF_T1D),
#   LD = ldmat_IL2RA,
#   snp = IL2RA_T1D_merged_onlymat$rsid)
# 
# check_alignment(D1)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 14393)

Matrix products: default

locale:
[1] LC_COLLATE=Finnish_Finland.1252  LC_CTYPE=Finnish_Finland.1252   
[3] LC_MONETARY=Finnish_Finland.1252 LC_NUMERIC=C                    
[5] LC_TIME=Finnish_Finland.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ieugwasr_0.1.5    ggpubr_0.4.0      TwoSampleMR_0.5.6 coloc_5.1.0.1    
 [5] data.table_1.14.2 forcats_0.5.1     stringr_1.4.0     dplyr_1.0.9      
 [9] purrr_0.3.4       readr_2.1.2       tidyr_1.2.0       tibble_3.1.7     
[13] ggplot2_3.3.6     tidyverse_1.3.1   workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] matrixStats_0.62.0 fs_1.5.2           lubridate_1.8.0    httr_1.4.3        
 [5] rprojroot_2.0.3    tools_4.2.1        backports_1.4.1    bslib_0.4.0       
 [9] utf8_1.2.2         R6_2.5.1           irlba_2.3.5        nortest_1.0-4     
[13] DBI_1.1.3          colorspace_2.0-3   withr_2.5.0        tidyselect_1.1.2  
[17] gridExtra_2.3      processx_3.6.1     compiler_4.2.1     git2r_0.30.1      
[21] glmnet_4.1-4       cli_3.3.0          rvest_1.0.2        xml2_1.3.3        
[25] labeling_0.4.2     sass_0.4.2         scales_1.2.0       callr_3.7.1       
[29] mixsqp_0.3-43      digest_0.6.29      rmarkdown_2.14     pkgconfig_2.0.3   
[33] htmltools_0.5.2    mr.raps_0.2        highr_0.9          dbplyr_2.2.1      
[37] fastmap_1.1.0      rlang_1.0.3        readxl_1.4.0       susieR_0.12.16    
[41] rstudioapi_0.13    farver_2.1.1       shape_1.4.6        jquerylib_0.1.4   
[45] generics_0.1.3     jsonlite_1.8.0     car_3.1-0          magrittr_2.0.3    
[49] Matrix_1.4-1       Rcpp_1.0.8.3       munsell_0.5.0      fansi_1.0.3       
[53] abind_1.4-5        viridis_0.6.2      lifecycle_1.0.1    stringi_1.7.6     
[57] whisker_0.4        yaml_2.3.5         carData_3.0-5      plyr_1.8.7        
[61] grid_4.2.1         promises_1.2.0.1   crayon_1.5.1       lattice_0.20-45   
[65] cowplot_1.1.1      splines_4.2.1      haven_2.5.0        hms_1.1.1         
[69] knitr_1.39         ps_1.7.1           pillar_1.8.0       ggsignif_0.6.3    
[73] codetools_0.2-18   reprex_2.0.1       glue_1.6.2         evaluate_0.15     
[77] getPass_0.2-2      modelr_0.1.8       vctrs_0.4.1        tzdb_0.3.0        
[81] httpuv_1.6.5       foreach_1.5.2      cellranger_1.1.0   gtable_0.3.0      
[85] reshape_0.8.9      assertthat_0.2.1   cachem_1.0.6       xfun_0.31         
[89] broom_1.0.0        rstatix_0.7.0      later_1.3.0        survival_3.3-1    
[93] viridisLite_0.4.0  iterators_1.0.14   ellipsis_0.3.2    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
