<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tea Heikkilä, Emilia Kaiser &amp; Jaakko Koskenniemi" />


<title>JAK2</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.0/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.0/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">T1DSCREEN</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="Harmonize.html">Harmonize</a>
</li>
<li>
  <a href="IL2RA.html">IL2RA</a>
</li>
<li>
  <a href="IL2RB.html">IL2RB</a>
</li>
<li>
  <a href="IL2RG.html">IL2RG</a>
</li>
<li>
  <a href="IL6R.html">IL6R</a>
</li>
<li>
  <a href="CRP.html">CRP</a>
</li>
<li>
  <a href="IL12B.html">IL12B</a>
</li>
<li>
  <a href="JAK1.html">JAK1</a>
</li>
<li>
  <a href="JAK2.html">JAK2</a>
</li>
<li>
  <a href="JAK3.html">JAK3</a>
</li>
<li>
  <a href="TYK2.html">TYK2</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jkoskenniemi/T1DSCREEN">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">JAK2</h1>
<h4 class="author">Tea Heikkilä, Emilia Kaiser &amp; Jaakko
Koskenniemi</h4>
<h4 class="date">2022-24-05</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-05-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>T1DSCREEN/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220313code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220313)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220313code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220313)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjkoskenniemiT1DSCREENtreefc32168e1730da0b68e2c69f77a74b354ae13749targetblankfc32168a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/tree/fc32168e1730da0b68e2c69f77a74b354ae13749" target="_blank">fc32168</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomjkoskenniemiT1DSCREENtreefc32168e1730da0b68e2c69f77a74b354ae13749targetblankfc32168a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/jkoskenniemi/T1DSCREEN/tree/fc32168e1730da0b68e2c69f77a74b354ae13749" target="_blank">fc32168</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/README_cis_eqtl.txt
    Ignored:    data/README_cis_eqtl_AF.txt
    Ignored:    data/cis-EQTL-AF.txt
    Ignored:    data/cis_eqtl_sumstats/2019-12-11-cis-eQTLsFDR-ProbeLevel-CohortInfoRemoved-BonferroniAdded.txt
    Ignored:    data/cis_eqtl_sumstats/2019-12-11-cis-eQTLsFDR0.05-ProbeLevel-CohortInfoRemoved-BonferroniAdded.txt/
    Ignored:    data/cis_eqtl_sumstats/cis-eQTL-SMR_20191212.tar.gz
    Ignored:    data/cis_sumstats/2018-07-18_SNP_AF_for_AlleleB_combined_allele_counts_and_MAF_pos_added.txt
    Ignored:    refs/

Untracked files:
    Untracked:  IL2RA_debug.Rmd

Unstaged changes:
    Modified:   analyses/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analyses/JAK2.Rmd</code>) and HTML
(<code>docs/JAK2.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/fc32168e1730da0b68e2c69f77a74b354ae13749/analyses/JAK2.Rmd" target="_blank">fc32168</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-05-04
</td>
<td>
wflow_publish(c("analyses/IL6R.Rmd", "analyses/JAK1.Rmd",
"analyses/JAK2.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/b49016a04b5830e2336cc7f1be6dd2ba7b0943b1/docs/JAK2.html" target="_blank">b49016a</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-05-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/c541480d23ea12e6e28d6b3a11d930db54659c36/docs/JAK2.html" target="_blank">c541480</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-05-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/1b4db48910a783cf4ed1dee7e818503a404e4152/docs/JAK2.html" target="_blank">1b4db48</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-05-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/d1d186cc27b1ed09009596fd246995ee4f5e708e/analyses/JAK2.Rmd" target="_blank">d1d186c</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-05-03
</td>
<td>
Added comment on eQTL data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/e0d3987fcbf69cf5b42d345572022434bb91990f/docs/JAK2.html" target="_blank">e0d3987</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-04-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/c8d146f5d34c2ded365b3a66c1596afc77c502b7/docs/JAK2.html" target="_blank">c8d146f</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/ef6d816af176f675b2895f7a15f6bcabed10716c/docs/JAK2.html" target="_blank">ef6d816</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/2db43172332aa29d52cebbf310b1c13a67840638/analyses/JAK2.Rmd" target="_blank">2db4317</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
First commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/24d77f36926c4897c8f7817b4eaa2bf63238fb73/docs/JAK2.html" target="_blank">24d77f3</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2022-11-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/e86595d567aec82d7280c4fdbbaed04cc0d85823/analyses/JAK2.Rmd" target="_blank">e86595d</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2022-11-05
</td>
<td>
wflow_publish(c("analyses/CRP.Rmd", "analyses/CXCL10.Rmd",
"analyses/IL2RA.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Load packages and import data</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>-- Attaching core tidyverse packages ------------------------ tidyverse 2.0.0 --
v dplyr     1.1.2     v readr     2.1.4
v forcats   1.0.0     v stringr   1.5.0
v ggplot2   3.4.2     v tibble    3.2.1
v lubridate 1.9.2     v tidyr     1.3.0
v purrr     1.0.1     
-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()
i Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;

The following objects are masked from &#39;package:lubridate&#39;:

    hour, isoweek, mday, minute, month, quarter, second, wday, week,
    yday, year

The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last

The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(coloc)</code></pre>
<pre><code>This is a new update to coloc.</code></pre>
<pre class="r"><code>library(TwoSampleMR)</code></pre>
<pre><code>TwoSampleMR version 0.5.6 
[&gt;] New: Option to use non-European LD reference panels for clumping etc
[&gt;] Some studies temporarily quarantined to verify effect allele
[&gt;] See news(package=&#39;TwoSampleMR&#39;) and https://gwas.mrcieu.ac.uk for further details</code></pre>
<pre class="r"><code>library(ggpubr)
library(ieugwasr)</code></pre>
<pre><code>API: public: http://gwas-api.mrcieu.ac.uk/

Attaching package: &#39;ieugwasr&#39;

The following object is masked from &#39;package:TwoSampleMR&#39;:

    ld_matrix</code></pre>
<pre class="r"><code>library(rio)
library(here)</code></pre>
<pre><code>here() starts at C:/Users/jajoko/Documents/T1DSCREEN</code></pre>
<pre class="r"><code>here::i_am(&quot;analyses/JAK2.Rmd&quot;)</code></pre>
<pre><code>here() starts at C:/Users/jajoko/Documents/T1DSCREEN</code></pre>
<div id="eqtl" class="section level1">
<h1>eQTL</h1>
<p>Load packages and import data</p>
<pre class="r"><code>JAK2_eqtl &lt;- read_exposure_data(&quot;data/export/JAK2_eqtl_TwoSampleMR.csv&quot;, sep=&quot;,&quot;)</code></pre>
<pre><code>Warning in format_data(as.data.frame(exposure_dat), type = &quot;exposure&quot;, snps =
NULL, : se column is not numeric. Coercing...</code></pre>
<pre><code>Warning in format_data(as.data.frame(exposure_dat), type = &quot;exposure&quot;, snps = NULL, : The following SNP(s) are missing required information for the MR tests and will be excluded
rs6476941
rs2149556
rs7036034
rs7858245
rs10974990
rs12340344
rs7040922
rs12115301
rs7029244
rs7874624
rs1328919
rs62541916
rs1581927
rs1590803
rs61131640
rs7038813
rs12344116
rs9657576
rs7030853
rs60460912
rs1590804
rs10815171
rs12343038
rs7022901
rs12340333
rs7036536
rs7040806
rs12340303
rs10758674
rs7029084
rs61461940
rs12338854
rs7861599
rs12340379
rs12339666
rs10116823
rs57299248
rs12001334
rs7871515
rs7032616
rs2094619
rs12341844
rs10974993
rs12684720
rs10974987
rs11791281
rs7047795
rs7036476
rs10115962
rs57698263
rs10815167
rs12337480
rs10815166
rs10974935
rs12337481
rs10115335
rs76960049
rs10815143
rs7850425
rs10121316
rs61284219
rs10974982
rs1571438
rs10115464
rs2094393
rs10974922
rs12336875
rs7470337
rs12352285
rs7040136
rs7039518
rs2149552
rs2094622
rs7046077
rs11787793
rs7038687
rs10815149
rs12337904
rs10815158
rs7863219
rs10974972
rs7046511
rs11999076
rs7849717
rs10815157
rs1360063
rs10815150
rs7035251
rs7045491
rs1328918
rs7847141
rs10123980
rs7857730
rs7038763
rs3780373
rs10118930
rs10815148
rs10974914
rs7850484
rs12353218
rs7030260
rs7851455
rs72699590
rs11788164
rs28378223
rs7034878
rs3780372
rs7852755
rs10117501
rs12335663
rs62541531
rs7025540
rs10974984
rs966871
rs3824433
rs7023146
rs10121077
rs2149555
rs59225790
rs10815153
rs10217652
rs12339774
rs11795305
rs1328917
rs10815154
rs10120763
rs2209773
rs7032785
rs10974960
rs7857081
rs7022333
rs10758673
rs12335546
rs10283563
rs62543572
rs7868130
rs62543560
rs9695933
rs7469563
rs2225125
rs62541959
rs62541556
rs7870694
rs12342421
rs10815165
rs7019912
rs62541957
rs10974910
rs57707737
rs149390964
rs9695934
rs62541958
rs7028112
rs1536798
rs62541543
rs57204002
rs7875265
rs72699568
rs12336577
rs10974912
rs1327494
rs4495487
rs11791350
rs59668095
rs12343374
rs58345215
rs3780365
rs12337134
rs10815145
rs72701605
rs1028912
rs10121491
rs1034072
rs7037207
rs11999928
rs7847294
rs12340617
rs111573904
rs7043371
rs12340895
rs7851556
rs62541534
rs12682970
rs7871070
rs2149553
rs12686652
rs10974938
rs7859390
rs62541541
rs1159782
rs12352022
rs11790841
rs60221565
rs10117459
rs12343065
rs10974939
rs3780367
rs62541944
rs12340866
rs10118789
rs3780371
rs10117565
rs3780377
rs62541899
rs62541903
rs10974980
rs10115970
rs117896567
rs10115518
rs10815164
rs12347727
rs2381195
rs62541538
rs11792531
rs62541915
rs10974983
rs11788834
rs2274649
rs7032706
rs12345127
rs1576271
rs62541945
rs11788949
rs11788790
rs10115172
rs3780374
rs62543877
rs10815144
rs72701654
rs12340827
rs10124627
rs10974979
rs62541539
rs12348771
rs3780369
rs3780368
rs62541540
rs59384377
rs10974916
rs12340894
rs11789744
rs10974964
rs7045000
rs3780370
rs2104594
rs10116560
rs10283730
rs150926910
rs150105389
rs10283564
rs4593605
rs7869668
rs11794778
rs6476939
rs10117591
rs7852988
rs2230724
rs12343867
rs67006981
rs149389628
rs2209774
rs10974915
rs62543863
rs2149560
rs10123976
rs12346093
rs7019858
rs10815147
rs12344629
rs4527935
rs12349918
rs3780366
rs12349785
rs17425819
rs17425637
rs10115312
rs10974969
rs2381213
rs2183137
rs16922772
rs10815159
rs10974963
rs11506661
rs10974921
rs11794708
rs7850294
rs62541532
rs12682672
rs3780378
rs7875908
rs1322222
rs7046736
rs12352214
rs10122037
rs11793886
rs6476945
rs884132
rs10974944
rs3780381
rs10815146
rs11788963
rs7865719
rs10123873
rs72699574
rs10114315
rs6476946
rs16922576
rs11790596
rs16922786
rs7869015
rs4560849
rs10974952
rs7043489
rs7874785
rs2224571
rs10491650
rs6476947
rs16922779
rs17492549
rs10491651
rs62543574
rs62543573
rs6476948
rs7856773
rs7033052
rs10975024
rs10283734
rs12349113
rs11506663
rs1120679
rs6476953
rs6476950
rs11792629
rs4358852
rs7027871
rs10975031
rs62543876
rs10120337
rs7025005
rs7874495
rs113683988
rs7035915
rs7035456
rs12342390
rs1853221
rs11790680
rs10283467
rs7868706
rs7850228
rs10758677
rs10815172
rs7858422
rs10118267
rs7847811
rs7859361
rs10283473
rs10975028
rs10975027
rs1590268
rs6476956
rs10123322
rs11506668
rs7019418
rs10118434
rs7848312
rs112603351
rs6476957
rs7859286
rs2146040
rs1830610
rs28872016
rs10975023
rs6476949
rs1575283
rs6476952
rs11506292
rs7870536
rs2038588
rs2208685
rs11506293
rs72703608
rs12350079
rs7023639
rs12351715
rs7045342
rs6476955
rs12349508
rs4500135
rs13285894
rs62541529
rs2381215
rs7856912
rs72701682
rs4372063
rs10975033
rs1322224
rs7043634
rs11793659
rs2146041
rs4587378
rs72701680
rs4342662
rs10974974
rs36051895
rs75354820
rs75032480
rs10481522
rs10815169
rs13440043
rs12351277
rs7034753
rs72699592
rs2181115
rs7862042
rs10974978
rs1575285
rs2031906
rs1555476
rs11999802
rs6476951
rs61657529
rs10975009
rs3780383
rs1575281
rs10975003
rs3780382
rs16922785
rs9987451
rs17426903
rs7036761
rs16922773
rs7848872
rs10975029
rs12343727
rs1536800
rs12347799
rs12720
rs4425810
rs147084794
rs4142064
rs10429491
rs1887428
rs10156475
rs7867137
rs7036833
rs60768043
rs10115208
rs10114531
rs11506660
rs7849191
rs10758669
rs7849067
rs7861755
rs7861763
rs10974900
rs2093446
rs2093447
rs2093448
rs2104166
rs10119004
rs4008433
rs10758670
rs10156618
rs2274472
rs10974913
rs12001578
rs2031752
rs3808850
rs7468147
rs7862851
rs7862852
rs1887426
rs113657238
rs10815141
rs10974892
rs1327497
rs1887425
rs7848416
rs1322223
rs10974967
rs10974897
rs78610467
rs7041547
rs62554837
rs75774657
rs7035015
rs1575284
rs148462161
rs11791395
rs58367852
rs62541511
rs116994058
rs10974975
rs12683801
rs10815140
rs11794946
rs10974976
rs11791507
rs12377744
rs141406245
rs10974905
rs2031754
rs12377160
rs7032169
rs10974865
rs10974977
rs1327500
rs59966455
rs72701653
rs112324541
rs2381216
rs77551840
rs16922908
rs10974973
rs2381194
rs113786699
rs2182726
rs62541542
rs62541558
rs113353229
rs75448287
rs62543884
rs75035022
rs62543904
rs9969735
rs10974837
rs78061489
rs10974835
rs9942887
rs62541503
rs7854773
rs10975043
rs148595697
rs77605397
rs6476959
rs7864244
rs2381180
rs7853475
rs10815112
rs10815130
rs7849277
rs139261028
rs10758678
rs1555477
rs12684204
rs75589511
rs146701017
rs16922344
rs144439143
rs10815119
rs10217321
rs715769
rs10115467
rs10815121
rs1410006
rs1571220
rs59774261
rs7862645
rs7867268
rs7020559
rs16922337
rs7023498
rs6476921
rs62543879
rs9942905
rs68158363
rs7025867
rs7025869
rs7874647
rs76521383
rs10815110
rs7853655
rs7856995
rs7856073
rs4291328
rs6476919
rs4541986
rs4495484
rs2025169
rs7870381
rs74332040
rs6476920
rs1327495
rs1327496
rs78745832
rs7864782
rs4428717
rs1327492
rs72701688
rs4574902
rs116866937
rs7863288
rs7862378
rs4742060
rs7872877
rs62557688
rs10481523
rs61399403
rs7854664
rs7868954
rs148091490
rs77533513
rs35774855
rs1475105
rs2297136
rs9969783
rs77845561
rs1407387
rs10815102
rs9721857
rs10119552
rs10974873
rs7868967
rs7853852
rs6476958
rs10974893
rs10974887
rs10975039
rs17803986
rs11999445
rs1407388
rs7021993
rs1971472
rs4742053
rs1971473
rs10974874
rs7033053
rs7032292
rs10815125
rs10975041
rs13288137
rs1327499
rs142940451
rs10974871
rs7868186
rs10815126
rs1041215
rs1327490
rs2224572
rs138578571
rs12684195
rs7022514
rs7046183
rs4742054
rs1571222
rs7040940
rs4742076
rs10815123
rs62541502
rs7047392
rs4742058
rs4742055
rs7868200
rs62541510
rs7869831
rs12235675
rs75574230
rs2208686
rs149460882
rs2146043
rs10815136
rs150260966
rs1410008
rs7862098
rs1575286
rs4742056
rs719673
rs144666602
rs6476929
rs7873020
rs10975045
rs7037047
rs9775690
rs10974877
rs62541491
rs41316003
rs183314067
rs7029647
rs4742057
rs72699540
rs10974858
rs148858951
rs146450005
rs17718680
rs7852291
rs7034721
rs4740805
rs10815124
rs3824439
rs117829760
rs148465266
rs117858295
rs7035298
rs76778936
rs148489539
rs1410004
rs10815107
rs12156457
rs10974857
rs6476931
rs17501156
rs4452849
rs10974879
rs10739071
rs7038475
rs10974859
rs10815111
rs7037557
rs10974847
rs2182725
rs12349020
rs13286824
rs149331299
rs7851572
rs10815228
rs139621882
rs10975129
rs4742100
rs7853905
rs7027180
rs12685848
rs118121949
rs4742078
rs148602745
rs10815232
rs10758668
rs58817806
rs143206068
rs116968409
rs4742099
rs7030147
rs7873651
rs10974880
rs10758665
rs144745077
rs4481663
rs4742102
rs140379169
rs62560216
rs140533307
rs3968
rs10121574
rs188285574
rs117275321
rs138648167
rs55774436
rs4742098
rs10815230
rs7875194
rs10815229
rs7043593
rs10125854
rs2381212
rs116945594
rs10974846
rs10125633
rs62557661
rs2890657
rs62559127
rs13297881
rs3780394
rs10974851
rs10975037</code></pre>
<pre><code>Warning in format_data(as.data.frame(exposure_dat), type = &quot;exposure&quot;, snps =
NULL, : None of the provided SNPs can be used for MR analysis, they are missing
required information.</code></pre>
<pre class="r"><code>JAK2_T1D  &lt;- read_outcome_data(&quot;data/export/JAK2_T1D_TwoSampleMR.csv&quot;, sep=&quot;,&quot;)

JAK2 &lt;- harmonise_data(JAK2_eqtl, JAK2_T1D)</code></pre>
<pre><code>Harmonising Whole blood JAK2 mRNA (4vucPG) and Risk of type 1 diabetes (xN25Oe)</code></pre>
<pre><code>Removing the following SNPs for being palindromic with intermediate allele frequencies:
rs10114548, rs10118317, rs10118468, rs10121225, rs10283872, rs1028912, rs1034531, rs10481522, rs10739030, rs10739058, rs10758591, rs10758645, rs10758700, rs10758702, rs10758704, rs10758706, rs10814847, rs10814894, rs10814951, rs10814972, rs10815009, rs10815146, rs10815258, rs10815273, rs10974345, rs10974495, rs10974520, rs10974560, rs10974566, rs10974897, rs10975056, rs10975263, rs10975266, rs1112114, rs12002985, rs12235675, rs12379291, rs12379318, rs1327496, rs13283292, rs13288897, rs184205, rs1971473, rs2104173, rs2760413, rs2760416, rs296842, rs296843, rs296849, rs301462, rs301485, rs367147, rs376208, rs378809, rs3858043, rs3858046, rs386616, rs390612, rs406455, rs409704, rs419083, rs4237150, rs4286995, rs4338160, rs4451360, rs453852, rs4740754, rs4740786, rs4741895, rs4741965, rs4741982, rs4741995, rs62543876, rs6415794, rs6476839, rs6476889, rs6476939, rs6476968, rs6915, rs7020394, rs7020673, rs7022369, rs7023770, rs7024686, rs7025900, rs7025968, rs7030195, rs7033052, rs7036458, rs7039093, rs7041581, rs7043371, rs7045178, rs7045196, rs751591, rs7851538, rs7853035, rs7853682, rs7853852, rs7854664, rs7857194, rs7865209, rs7869643, rs7871343, rs7872075, rs806053, rs806056, rs822336, rs841975, rs958480, rs9775228</code></pre>
<div id="manhattan-plots" class="section level2">
<h2>Manhattan plots</h2>
<pre class="r"><code>#create Manhattan plots with vertical lines indicating the selected area (see below 2.4)
JAK2_eqtl.fig &lt;-
  JAK2 %&gt;%
  ggplot(aes(x = pos.exposure/1000, y=-log10(pval.exposure))) +
  geom_point()+
  geom_hline(yintercept=8, linetype=&quot;dashed&quot;)+
  ggtitle(&quot;GWAS of JAK2 gene expression in blood&quot;) +
  ylab(&quot;-log10(p)&quot;) +
  xlab(NULL)

JAK2_T1D.fig &lt;-
  JAK2 %&gt;%
  ggplot()+
  geom_point(mapping=aes(x = pos.exposure/1000, y=-log10(pval.outcome))) +
  geom_hline(yintercept=8, linetype=&quot;dashed&quot;)+
  # geom_vline(xintercept=36.9e6/1000, linetype=&quot;dashed&quot;)+
  # geom_vline(xintercept=37.3e6/1000, linetype=&quot;dashed&quot;)+
  ggtitle(&quot;GWAS of risk of T1D&quot;) +
  ylab(&quot;-log10(p)&quot;) +
  xlab(&quot;position(kbp)&quot;)

#Obtain range for figure of gene
layer_scales(JAK2_T1D.fig)$x$range$range</code></pre>
<pre><code>[1] 4056.953 6056.468</code></pre>
<pre class="r"><code>JAK2_gene.fig &lt;-
  ggplot(data = JAK2) +
  geom_blank() +
  geom_segment(x=37521.878, xend=37571.094, y=1, yend=1, linewidth = 2) +
  annotate(&quot;text&quot;, label = &quot;JAK2&quot;,
           x= (37521.878 + 37571.094) / 2, y=1, hjust = 0.5, vjust = -1) +
  xlim(36521.878, 38571.094) +
  ylim(0.75, 2) +
  ylab(NULL) + xlab(NULL) +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank())

fig_JAK2_manhattan &lt;- ggarrange(JAK2_gene.fig, JAK2_eqtl.fig, JAK2_T1D.fig,
                                heights = c(1, 3, 3), nrow = 3,
                                ncol = 1, align = &quot;hv&quot;)
# ggsave(&quot;figures/JAK2-manhattan-combined-fig.pdf&quot;,
#        height = 6, width = 8, units = &quot;in&quot;)</code></pre>
</div>
<div id="coloc-assuming-1-causal-variant" class="section level2">
<h2>Coloc assuming 1 causal variant</h2>
<pre class="r"><code>#recode MAF
JAK2 &lt;- JAK2 %&gt;%
  mutate(maf.exposure  = ifelse(eaf.exposure  &lt; 0.5, eaf.exposure,  1-eaf.exposure)) %&gt;%
  mutate(maf.outcome  = ifelse(eaf.outcome  &lt; 0.5, eaf.outcome,  1-eaf.outcome))

colnames(JAK2)</code></pre>
<pre><code> [1] &quot;SNP&quot;                    &quot;effect_allele.exposure&quot; &quot;other_allele.exposure&quot; 
 [4] &quot;effect_allele.outcome&quot;  &quot;other_allele.outcome&quot;   &quot;beta.exposure&quot;         
 [7] &quot;beta.outcome&quot;           &quot;eaf.exposure&quot;           &quot;eaf.outcome&quot;           
[10] &quot;remove&quot;                 &quot;palindromic&quot;            &quot;ambiguous&quot;             
[13] &quot;id.outcome&quot;             &quot;pos.outcome&quot;            &quot;pval.outcome&quot;          
[16] &quot;se.outcome&quot;             &quot;ncase.outcome&quot;          &quot;ncontrol.outcome&quot;      
[19] &quot;samplesize.outcome&quot;     &quot;outcome&quot;                &quot;mr_keep.outcome&quot;       
[22] &quot;pval_origin.outcome&quot;    &quot;data_source.outcome&quot;    &quot;pval.exposure&quot;         
[25] &quot;pos.exposure&quot;           &quot;samplesize.exposure&quot;    &quot;se.exposure&quot;           
[28] &quot;exposure&quot;               &quot;mr_keep.exposure&quot;       &quot;pval_origin.exposure&quot;  
[31] &quot;id.exposure&quot;            &quot;data_source.exposure&quot;   &quot;action&quot;                
[34] &quot;mr_keep&quot;                &quot;maf.exposure&quot;           &quot;maf.outcome&quot;           </code></pre>
<pre class="r"><code>D1 &lt;- list(
  type = &quot;quant&quot;, # quantitative trait
  pvalues = JAK2$pval.exposure,
  N = JAK2$samplesize.exposure,
  MAF = JAK2$maf.exposure,
  pos = JAK2$pos.exposure,
  snp = JAK2$pos.exposure,
  sdY = 1)

D2 &lt;- list(
  type = &quot;cc&quot;, # case-control trait
  Beta = JAK2$beta.outcome,
  varBeta = JAK2$se.outcome,
  pvalues = JAK2$pval.outcome,
  N = 18942+501638, # Case-control study (Chiou et al. 2021 Nature)
  s = 18942/(18942+501638), # N_case/(N_case+ N_ctrl)
  MAF = JAK2$maf.exposure,  #eqtl used here in purpose
  pos = JAK2$pos.outcome,
  snp = JAK2$pos.exposure)

check_dataset(D1)</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>check_dataset(D2)</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>coloc_JAK2 &lt;- coloc.abf(D1, D2, p1 = 1e-4, p2 = 1e-4, p12 = 1e-5)</code></pre>
<pre><code>PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf 
2.88e-146  2.28e-13 1.26e-133  1.00e+00  2.00e-15 
[1] &quot;PP abf for shared variant: 2e-13%&quot;</code></pre>
<pre class="r"><code>sensitivity(coloc_JAK2, &quot;H4 &gt; 0.7&quot;)</code></pre>
<pre><code>Results fail decision rule H4 &gt; 0.7</code></pre>
<p><img src="figure/JAK2.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="pqtl" class="section level1">
<h1>pQTL</h1>
<pre class="r"><code>JAK2_prot &lt;- read_exposure_data(&quot;data/export/JAK2_prot_anno_TwoSampleMR.csv&quot;, sep=&quot;,&quot;)
JAK2_T1D  &lt;- read_outcome_data(&quot;data/export/JAK2_T1D_TwoSampleMR.csv&quot;, sep=&quot;,&quot;)

JAK2 &lt;- harmonise_data(JAK2_prot, JAK2_T1D)</code></pre>
<pre><code>Harmonising Serum JAK2 level (qs1g78) and Risk of type 1 diabetes (f66Ik5)</code></pre>
<pre><code>Removing the following SNPs for incompatible alleles:
rs1207517620, rs141508055, rs372207026, rs372744561, rs530444901, rs539806656, rs568382606, rs571687697, rs574803352, rs576911975, rs74661850, rs75295775, rs756581954, rs759860808, rs762632985, rs766925196, rs76885790, rs770929012, rs773275069, rs773889954, rs867874883</code></pre>
<pre><code>Removing the following SNPs for being palindromic with intermediate allele frequencies:
rs10114548, rs10118317, rs10118468, rs10118904, rs10121225, rs10283872, rs1028912, rs1034531, rs10481522, rs10739030, rs10739062, rs10758645, rs10758665, rs10758700, rs10758702, rs10758704, rs10758706, rs10814847, rs10814894, rs10814951, rs10814972, rs10815009, rs10815062, rs10815123, rs10815273, rs10815351, rs10974345, rs10974495, rs10974520, rs10974560, rs10974566, rs10974601, rs10974847, rs10974897, rs10975056, rs10975263, rs10975266, rs11515140, rs12002985, rs12235675, rs12379318, rs1327496, rs13283292, rs13288137, rs13288897, rs149556337, rs1556362, rs1556989, rs184205, rs1971473, rs2380947, rs2380948, rs2760413, rs2760416, rs296842, rs296843, rs296849, rs301462, rs301485, rs340891, rs367147, rs376208, rs378809, rs3858043, rs3858046, rs386616, rs3896197, rs3934283, rs406455, rs409704, rs418122, rs419083, rs422499, rs4286995, rs4291329, rs4352902, rs436300, rs4439181, rs444826, rs4481663, rs4481664, rs4484748, rs453852, rs4740754, rs4740775, rs4740786, rs4741878, rs4741880, rs4741893, rs4741895, rs4741965, rs4741982, rs4741995, rs62540578, rs62543876, rs637742, rs6415794, rs6476889, rs6476968, rs6477031, rs6477032, rs650816, rs6915, rs7020394, rs7020673, rs7022369, rs7023770, rs7024686, rs7025900, rs7025968, rs7033052, rs7036458, rs7039093, rs7043371, rs7045178, rs7045196, rs7046183, rs751591, rs7851538, rs7853035, rs7853682, rs7853852, rs7854664, rs7857194, rs7865209, rs7868186, rs7869643, rs7874642, rs7875108, rs806053, rs806056, rs822336, rs958480, rs9775228</code></pre>
<div id="manhattan-plots-1" class="section level2">
<h2>Manhattan plots</h2>
<p>There are no GWAS hits in GWAS of JAK2 in serum, so no coloc or MR is
done</p>
<pre class="r"><code>#create Manhattan plots with vertical lines indicating the selected area (see below 2.4)
JAK2_prot.fig &lt;- 
  JAK2 %&gt;%
  ggplot(aes(x = pos.exposure/1000, y=-log10(pval.exposure))) +
  geom_point()+
  geom_hline(yintercept=8, linetype=&quot;dashed&quot;)+
  # geom_vline(xintercept=36.9e6/1000, linetype=&quot;dashed&quot;)+
  # geom_vline(xintercept=37.3e6/1000, linetype=&quot;dashed&quot;)+
  ggtitle(&quot;GWAS of JAK2 in serum&quot;) +
  ylab(&quot;-log10(p)&quot;) +
  xlab(NULL)

JAK2_T1D.fig &lt;- 
  JAK2 %&gt;%
  ggplot()+
  geom_point(mapping=aes(x = pos.exposure/1000, y=-log10(pval.outcome))) +
  geom_hline(yintercept=8, linetype=&quot;dashed&quot;)+
  # geom_vline(xintercept=36.9e6/1000, linetype=&quot;dashed&quot;)+
  # geom_vline(xintercept=37.3e6/1000, linetype=&quot;dashed&quot;)+
  ggtitle(&quot;GWAS of risk of T1D&quot;) +
  ylab(&quot;-log10(p)&quot;) +
  xlab(&quot;position(kbp)&quot;)

#Obtain range for figure of gene
layer_scales(JAK2_T1D.fig)$x$range$range</code></pre>
<pre><code>[1] 3985.359 6128.137</code></pre>
<pre class="r"><code>JAK2_gene.fig &lt;- 
  ggplot(data = JAK2) +
  geom_blank() +
  geom_segment(x=37521.878, xend=37571.094, y=1, yend=1, linewidth = 2) +
  annotate(&quot;text&quot;, label = &quot;JAK2&quot;, 
           x= (37521.878 + 37571.094) / 2, y=1, hjust = 0.5, vjust = -1) +
  xlim(36521.878, 38571.094) +
  ylim(0.75, 2) +
  ylab(NULL) + xlab(NULL) +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank(), 
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank())

fig_JAK2_manhattan &lt;- ggarrange(JAK2_gene.fig, JAK2_prot.fig, JAK2_T1D.fig,
          heights = c(1, 3, 3), nrow = 3,
          ncol = 1, align = &quot;hv&quot;)

fig_JAK2_manhattan</code></pre>
<p><img src="figure/JAK2.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggsave(&quot;figures/JAK2-manhattan-combined-fig.pdf&quot;, 
#        height = 6, width = 8, units = &quot;in&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.0 (2023-04-21 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 14393)

Matrix products: default


locale:
[1] LC_COLLATE=Finnish_Finland.1252  LC_CTYPE=Finnish_Finland.1252   
[3] LC_MONETARY=Finnish_Finland.1252 LC_NUMERIC=C                    
[5] LC_TIME=Finnish_Finland.1252    

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] here_1.0.1        rio_0.5.29        ieugwasr_0.1.5    ggpubr_0.6.0     
 [5] TwoSampleMR_0.5.6 coloc_5.1.0.1     data.table_1.14.8 lubridate_1.9.2  
 [9] forcats_1.0.0     stringr_1.5.0     dplyr_1.1.2       purrr_1.0.1      
[13] readr_2.1.4       tidyr_1.3.0       tibble_3.2.1      ggplot2_3.4.2    
[17] tidyverse_2.0.0   workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0   viridisLite_0.4.2  farver_2.1.1       viridis_0.6.3     
 [5] fastmap_1.1.1      reshape_0.8.9      promises_1.2.0.1   digest_0.6.31     
 [9] timechange_0.2.0   lifecycle_1.0.3    processx_3.8.1     magrittr_2.0.3    
[13] compiler_4.3.0     rlang_1.1.1        sass_0.4.6         tools_4.3.0       
[17] utf8_1.2.3         yaml_2.3.7         knitr_1.42         ggsignif_0.6.4    
[21] labeling_0.4.2     curl_5.0.0         plyr_1.8.8         abind_1.4-5       
[25] withr_2.5.0        foreign_0.8-84     grid_4.3.0         fansi_1.0.4       
[29] susieR_0.12.35     git2r_0.32.0       colorspace_2.1-0   scales_1.2.1      
[33] cli_3.6.1          rmarkdown_2.21     crayon_1.5.2       generics_0.1.3    
[37] rstudioapi_0.14    httr_1.4.5         tzdb_0.3.0         readxl_1.4.2      
[41] cachem_1.0.8       cellranger_1.1.0   matrixStats_0.63.0 vctrs_0.6.2       
[45] Matrix_1.5-4       jsonlite_1.8.4     carData_3.0-5      car_3.1-2         
[49] callr_3.7.3        hms_1.1.3          mixsqp_0.3-48      rstatix_0.7.2     
[53] irlba_2.3.5.1      jquerylib_0.1.4    glue_1.6.2         ps_1.7.5          
[57] cowplot_1.1.1      stringi_1.7.12     gtable_0.3.3       later_1.3.1       
[61] munsell_0.5.0      pillar_1.9.0       htmltools_0.5.5    R6_2.5.1          
[65] rprojroot_2.0.3    evaluate_0.20      lattice_0.21-8     highr_0.10        
[69] haven_2.5.2        backports_1.4.1    openxlsx_4.2.5.2   broom_1.0.4       
[73] httpuv_1.6.9       bslib_0.4.2        zip_2.3.0          Rcpp_1.0.10       
[77] gridExtra_2.3      whisker_0.4.1      xfun_0.39          fs_1.6.2          
[81] getPass_0.2-2      pkgconfig_2.0.3   </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
