<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tea Heikkilä, Emilia Kaiser &amp; Jaakko Koskenniemi" />


<title>TYK2</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">T1DSCREEN</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="Harmonize.html">Harmonize</a>
</li>
<li>
  <a href="IL2RA.html">IL2RA</a>
</li>
<li>
  <a href="IL2RB.html">IL2RB</a>
</li>
<li>
  <a href="IL2RG.html">IL2RG</a>
</li>
<li>
  <a href="IL6R.html">IL6R</a>
</li>
<li>
  <a href="CRP.html">CRP</a>
</li>
<li>
  <a href="IL12B.html">IL12B</a>
</li>
<li>
  <a href="JAK1.html">JAK1</a>
</li>
<li>
  <a href="JAK2.html">JAK2</a>
</li>
<li>
  <a href="JAK3.html">JAK3</a>
</li>
<li>
  <a href="TYK2.html">TYK2</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jkoskenniemi/T1DSCREEN">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">TYK2</h1>
<h4 class="author">Tea Heikkilä, Emilia Kaiser &amp; Jaakko
Koskenniemi</h4>
<h4 class="date">2022-26-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-05-04
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>T1DSCREEN/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220313code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220313)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220313code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220313)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjkoskenniemiT1DSCREENtreef6788b70b86715c68422e46d1f790902ba5ad8f1targetblankf6788b7a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/tree/f6788b70b86715c68422e46d1f790902ba5ad8f1" target="_blank">f6788b7</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomjkoskenniemiT1DSCREENtreef6788b70b86715c68422e46d1f790902ba5ad8f1targetblankf6788b7a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/jkoskenniemi/T1DSCREEN/tree/f6788b70b86715c68422e46d1f790902ba5ad8f1" target="_blank">f6788b7</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/README_cis_eqtl.txt
    Ignored:    data/README_cis_eqtl_AF.txt
    Ignored:    data/cis-EQTL-AF.txt
    Ignored:    data/cis_sumstats/2018-07-18_SNP_AF_for_AlleleB_combined_allele_counts_and_MAF_pos_added.txt
    Ignored:    refs/

Untracked files:
    Untracked:  data/cis_eqtl_sumstats/
    Untracked:  data/cis_sumstats/JAK1_T1D.rds
    Untracked:  data/cis_sumstats/JAK3_T1D.rds
    Untracked:  data/export/IFNAR2_eqtl_TwoSampleMR.csv
    Untracked:  data/export/IL2RA_eqtl_TwoSampleMR.csv
    Untracked:  data/export/IL2RB_eqtl_TwoSampleMR.csv
    Untracked:  data/export/IL2RG_eqtl_TwoSampleMR.csv
    Untracked:  data/export/IL6R_eqtl_TwoSampleMR.csv
    Untracked:  data/export/JAK1_T1D_TwoSampleMR.csv
    Untracked:  data/export/JAK1_eqtl_TwoSampleMR.csv
    Untracked:  data/export/JAK2_eqtl_TwoSampleMR.csv
    Untracked:  data/export/JAK3_T1D_TwoSampleMR.csv
    Untracked:  data/export/JAK3_eqtl_TwoSampleMR.csv
    Untracked:  data/export/TYK2_eqtl_TwoSampleMR.csv

Unstaged changes:
    Modified:   .gitignore
    Deleted:    analyses/CRP.Rmd
    Deleted:    analyses/CXCL10.Rmd
    Modified:   analyses/IL2RB.Rmd
    Deleted:    analyses/TNF.Rmd
    Modified:   analyses/TYK2.Rmd
    Modified:   code/Gather.R
    Modified:   code/linux-shell-script-wget-and-decompress.txt
    Deleted:    data/IL6R_crp_LDmat-2023-04-25.RData

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analyses/TYK2.Rmd</code>) and HTML
(<code>docs/TYK2.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/50537efb85e3ebeaf5e3740903500eb4747c204b/docs/TYK2.html" target="_blank">50537ef</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-05-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/cc02856e43d0d35ca2e8494f2768faaab1f3fbe4/analyses/TYK2.Rmd" target="_blank">cc02856</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-05-03
</td>
<td>
Added eQTL data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/e0d3987fcbf69cf5b42d345572022434bb91990f/docs/TYK2.html" target="_blank">e0d3987</a>
</td>
<td>
jkoskenniemi
</td>
<td>
2023-04-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/cfd36db7f48c142d3da502efdb244319c068660c/docs/TYK2.html" target="_blank">cfd36db</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/b91e2516e9492e66c55da9d43b75156b2a15873a/docs/TYK2.html" target="_blank">b91e251</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/c132eaa6f8c552f37374aff1c63b23ee2eb3bc5e/docs/TYK2.html" target="_blank">c132eaa</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/956f02071a5cbde649644a2ab3d8ce250117dd8c/analyses/TYK2.Rmd" target="_blank">956f020</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Fixed issue affecting the X-axis of gene figures in Manhattan plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/c8d146f5d34c2ded365b3a66c1596afc77c502b7/docs/TYK2.html" target="_blank">c8d146f</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/3eb9b1559cece93b86b725ebe76d2f7a97832cbd/docs/TYK2.html" target="_blank">3eb9b15</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/0e13d7ad89b41e1c401aeea4e17257ab823e5124/analyses/TYK2.Rmd" target="_blank">0e13d7a</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2023-04-26
</td>
<td>
First commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/jkoskenniemi/T1DSCREEN/24d77f36926c4897c8f7817b4eaa2bf63238fb73/docs/TYK2.html" target="_blank">24d77f3</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2022-11-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/jkoskenniemi/T1DSCREEN/blob/e86595d567aec82d7280c4fdbbaed04cc0d85823/analyses/TYK2.Rmd" target="_blank">e86595d</a>
</td>
<td>
Jaakko Koskenniemi
</td>
<td>
2022-11-05
</td>
<td>
wflow_publish(c("analyses/CRP.Rmd", "analyses/CXCL10.Rmd",
"analyses/IL2RA.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Load packages and import data</p>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>-- Attaching packages --------------------------------------- tidyverse 1.3.1 --</code></pre>
<pre><code>v ggplot2 3.4.1     v purrr   0.3.4
v tibble  3.1.7     v dplyr   1.0.9
v tidyr   1.2.0     v stringr 1.4.0
v readr   2.1.2     v forcats 0.5.1</code></pre>
<pre><code>Warning: package &#39;ggplot2&#39; was built under R version 4.2.2</code></pre>
<pre><code>-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(data.table)</code></pre>
<pre><code>
Attaching package: &#39;data.table&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    between, first, last</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    transpose</code></pre>
<pre class="r"><code>library(coloc)</code></pre>
<pre><code>This is a new update to coloc.</code></pre>
<pre class="r"><code>library(TwoSampleMR)</code></pre>
<pre><code>TwoSampleMR version 0.5.6 
[&gt;] New: Option to use non-European LD reference panels for clumping etc
[&gt;] Some studies temporarily quarantined to verify effect allele
[&gt;] See news(package=&#39;TwoSampleMR&#39;) and https://gwas.mrcieu.ac.uk for further details</code></pre>
<pre class="r"><code>library(ggpubr)
library(ieugwasr)</code></pre>
<pre><code>API: public: http://gwas-api.mrcieu.ac.uk/</code></pre>
<pre><code>
Attaching package: &#39;ieugwasr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:TwoSampleMR&#39;:

    ld_matrix</code></pre>
<pre class="r"><code>library(rio)</code></pre>
<pre><code>Warning: package &#39;rio&#39; was built under R version 4.2.2</code></pre>
<pre class="r"><code>library(here)</code></pre>
<pre><code>Warning: package &#39;here&#39; was built under R version 4.2.2</code></pre>
<pre><code>here() starts at C:/Users/jajoko/Documents/T1DSCREEN</code></pre>
<pre class="r"><code>here::i_am(&quot;analyses/TYK2.Rmd&quot;)</code></pre>
<pre><code>here() starts at C:/Users/jajoko/Documents/T1DSCREEN</code></pre>
<p>#eQTL</p>
<p>Load packages and import data</p>
<pre class="r"><code>TYK2_eqtl &lt;- read_exposure_data(&quot;data/export/TYK2_eqtl_TwoSampleMR.csv&quot;, sep=&quot;,&quot;)</code></pre>
<pre><code>Warning in format_data(as.data.frame(exposure_dat), type = &quot;exposure&quot;, snps =
NULL, : se column is not numeric. Coercing...</code></pre>
<pre><code>Warning in format_data(as.data.frame(exposure_dat), type = &quot;exposure&quot;, snps = NULL, : The following SNP(s) are missing required information for the MR tests and will be excluded
rs34725611
rs2304256
rs11085727
rs35251378
rs11085725
rs11085730
rs11879191
rs34953890
rs35164067
rs11669299
rs12720270
rs11670704
rs12983921
rs10416143
rs11672021
rs60167376
rs62130736
rs74179926
rs35452057
rs58172634
rs74179925
rs11666164
rs7259385
rs1051738
rs10415088
rs7259088
rs8101195
rs280497
rs35084366
rs11670508
rs12974442
rs430092
rs281423
rs281422
rs7247481
rs280519
rs281427
rs91755
rs7259116
rs7343087
rs8112449
rs280526
rs280525
rs62131889
rs11880990
rs35615354
rs12982179
rs8101473
rs11085732
rs7249459
rs8112675
rs12983665
rs11878853
rs17697965
rs11085731
rs12978984
rs12978105
rs11881782
rs6417247
rs10854116
rs12977594
rs35916055
rs11878501
rs34266232
rs34159543
rs12611227
rs7253917
rs75688257
rs34681319
rs12610298
rs3760650
rs2043330
rs7260594
rs7253185
rs113540846
rs112350143
rs753859
rs57653312
rs7246953
rs60956147
rs11668623
rs59351473
rs11671552
rs11666754
rs11085734
rs34246431
rs11670504
rs4804133
rs12720250
rs12720248
rs12720233
rs12720235
rs12720222
rs12720231
rs74179927
rs28382805
rs61451658
rs12459219
rs28378712
rs12985305
rs10408642
rs280523
rs12720260
rs12720253
rs73923215
rs4804135
rs11668429
rs12720214
rs77477628
rs12981942
rs9676881
rs1048290
rs28834106
rs11882133
rs60542850
rs62131887
rs2633973
rs1387025
rs35951601
rs1352426
rs7245388
rs35221071
rs12720352
rs6511701
rs12720234
rs2304237
rs10408916
rs55765999
rs7250656
rs10403396
rs3176769
rs7254041
rs7254330
rs7253912
rs7257871
rs3176766
rs3176767
rs3176768
rs6511698
rs281434
rs7258015
rs2633971
rs12720328
rs9807913
rs12971603
rs34585310
rs34723380
rs513663
rs1058154
rs477174
rs4804134
rs7250300
rs7249914
rs11669094
rs2358681
rs7248408
rs7256672
rs117845533
rs115947017
rs74644412
rs144309607
rs2569703
rs78527617
rs7247667
rs61198164
rs35359804
rs117039536
rs117196614
rs11666263
rs7247893
rs2230399
rs710845
rs280498
rs73510898
rs281440
rs203999
rs281439
rs12720356
rs74179931
rs35929247
rs12720346
rs5030347
rs10411056
rs35373372
rs118171875
rs12720279
rs280521
rs8111359
rs281417
rs12462944
rs378395
rs395782
rs12720299
rs901886
rs11575071
rs12720307
rs11575072
rs7409490
rs79709391
rs3745261
rs56381248
rs281416
rs79487480
rs77489129
rs58852827
rs59125425
rs2358581
rs62640397
rs117569384
rs117734369
rs2304258
rs55732741
rs17000730
rs76525253
rs2231486
rs150096100
rs78083518
rs75617577
rs79442975
rs75850379
rs3745263
rs78630786
rs3745262
rs3181049
rs56286467
rs149383028
rs144457626
rs118036551
rs144816341
rs77291073
rs28382774
rs281420
rs56003768
rs79618866
rs55821462
rs144977144
rs13306429
rs144584313
rs79627557
rs55651720
rs78763691
rs77027367
rs78344052
rs185824625
rs17000708
rs147198376
rs78899861
rs76255695
rs190630833
rs3181046
rs80120234
rs78370725
rs77358503
rs117495488
rs79406755
rs117083751
rs17000699
rs182844296
rs76109397
rs77837238
rs76829093
rs75681283
rs77070081
rs147406468
rs141342588
rs55779981
rs75420883
rs142284963
rs79947303
rs77783303
rs76812489
rs74944457
rs79889104
rs80348279
rs1045384
rs12984834
rs34031043
rs78195795
rs4804501
rs79758628
rs112599736
rs73019227
rs3176764
rs3181047
rs77000928
rs11673632
rs80262216
rs78404141
rs35193259
rs149658244
rs77072129
rs77952821
rs45524632
rs2304240
rs185380354
rs149410973
rs55925728
rs140428507
rs144581076
rs144161253
rs144303718
rs138208481
rs28372821
rs142413827
rs78372513
rs79817709
rs141749460
rs144659401
rs150547837
rs142663484
rs142770866
rs139561821
rs146039497
rs2304238
rs2304239
rs17697947
rs148534836
rs5030377
rs141000043
rs75231016
rs147664744
rs12462703
rs7408233
rs12983510
rs150778385
rs4804500
rs12459144
rs147965647
rs8100564
rs34157869
rs12460981
rs12462855
rs12975591
rs191954384
rs7245877
rs150461309
rs7258615
rs12720302
rs8108236
rs12720320
rs28382801
rs114758136
rs113456243
rs10404367
rs146020300
rs5498
rs186255753
rs34197572
rs75206631
rs76415590
rs76669781
rs140735577
rs76884064
rs78902064
rs55898302
rs78503130
rs34924007
rs34346015
rs34256230
rs183464907
rs150549387
rs923366
rs117679371
rs148055837
rs35597394
rs148841287
rs35295222
rs28382800
rs151009655
rs192560669
rs117281937
rs115347026
rs34777454
rs78623539
rs74457164
rs2163857
rs2129944
rs34177255
rs3093029
rs35817519
rs117576526
rs12461980
rs77782871
rs117591554
rs147239058
rs74204086
rs62130696
rs2569702
rs11878494
rs141567219
rs76134468
rs144563376
rs34403566
rs118167193
rs11879158
rs12463093
rs74367067
rs12460655
rs28382780
rs59451338
rs150445689
rs141059827
rs79918623
rs12720219
rs2278442
rs892188
rs28382788
rs6511694
rs190740393
rs11881227
rs11879176
rs146067009
rs2569693
rs77376335
rs74806945
rs76456798
rs75375271
rs114170067
rs117320342
rs78255613
rs78415953
rs138119840
rs11575074
rs10416653
rs117761027
rs12977237
rs510506
rs9749588
rs117539637
rs77995435
rs138523949
rs140752376
rs2075742
rs62130688
rs7247198
rs1056538
rs12610026
rs80161551
rs2228615
rs62130686
rs78064630
rs12972990
rs114268653
rs147262407
rs144649149
rs62130687
rs118115488
rs62130685
rs55939834
rs11085735
rs78854124
rs146564035
rs3093030
rs5030397
rs117454376
rs7256518
rs885743
rs79343819
rs12720326
rs34016352
rs140283599
rs147622113
rs1124765
rs116947105
rs5030390
rs77587461
rs12608948
rs78021472
rs4804556
rs8106630
rs62131895
rs147255904
rs8104304
rs11878588
rs2075741
rs10416073
rs34312839
rs3177696
rs146795181
rs4804528
rs76560514
rs139311191
rs74795234
rs114202680
rs144427215
rs12720280
rs75523935
rs61223991
rs150109720
rs12150978
rs8104480
rs11668544
rs12720294
rs4371271
rs55677033
rs75095909
rs78853431
rs79362074
rs111870775
rs144295214
rs11666993
rs74630689
rs11880594
rs7259486
rs12984270
rs8108977
rs7258189
rs8100910
rs60314748
rs11490554
rs280499
rs10411252
rs189762945
rs6511695
rs117679913
rs12720281
rs150434441
rs4804512
rs12720283
rs8099996
rs6511696
rs4804503
rs140789286
rs62131898
rs75055018
rs80066865
rs146329737
rs76619635
rs10418435
rs8103309
rs75508516
rs281414
rs12720218
rs12720215
rs112387158
rs117093042
rs191571564
rs113337445
rs9797720
rs59919460
rs117784065
rs2011172
rs7253253
rs12986385
rs1560712
rs4804138
rs281415
rs280524
rs2231482
rs115771260
rs1968445
rs117798643
rs892080
rs141848087
rs8113091
rs143978290
rs7253965
rs10426276
rs145427182
rs11878887
rs4024546
rs12720293
rs11878725
rs1465702
rs2043306
rs12720216
rs8100756
rs11085749
rs2017213
rs2163856
rs7258215
rs75407602
rs8107703
rs10854114
rs146531345
rs6511697
rs17000728
rs4804496
rs77793651
rs7251693
rs11671350
rs6511702
rs62130684
rs2033493
rs7259597
rs7250265
rs8111338
rs2116941
rs113207017
rs78200042
rs79185120
rs10854115
rs8104426
rs142238940
rs76123728
rs111645853
rs193000890
rs4804142
rs112739873
rs4804526
rs4804505
rs8102623
rs76215286
rs111799733
rs56335280
rs8099957
rs76925188
rs9973219
rs11666992
rs34370923
rs11672431
rs11879293
rs61282597
rs11670097
rs61692159
rs4804547
rs139154013
rs139437835
rs7245722
rs12980685
rs8102273
rs73003512
rs17616105
rs74179956
rs150908276
rs79927583
rs35991287
rs76499861
rs8103559
rs57434220
rs34949626
rs4804557
rs111380365
rs74428614
rs8103948
rs74178201
rs7254474
rs6511718
rs3786723
rs11668706
rs148313380
rs12985734
rs75239980
rs77482332
rs148215750
rs77389625
rs34837972
rs57579358
rs11671888
rs60412558
rs7257769
rs10404517
rs7275
rs10417578
rs8101254
rs73018660
rs12971616
rs35479557
rs10409578
rs11669339
rs34677258
rs60448955
rs35175941
rs8108935
rs35568987
rs11665795
rs12610374
rs73011374
rs76638997
rs79982275
rs74179958
rs281419
rs3786725
rs7259156
rs35734575
rs117212815
rs11085757
rs75184451
rs1975160
rs10418111
rs113878361
rs79011483
rs141611067
rs12983312
rs3786727
rs61244955
rs75323740
rs11881260
rs12609501
rs11879571
rs11669103
rs11879026
rs74179939
rs75192074
rs117707627
rs11669125
rs11671107
rs62129985
rs8106381
rs11666925
rs34318261
rs143928955
rs12976693
rs10409001
rs62129964
rs35781735
rs74827019
rs4804553
rs113379022
rs35093578
rs11668224
rs7258256
rs12979266
rs12609502
rs35887384
rs75864054
rs11670205
rs7252007
rs150538767
rs10415811
rs12983573
rs876794
rs113680403
rs60956778
rs11666990
rs56385993
rs11673439
rs17265047
rs112508627
rs12977506
rs62128754
rs11668364
rs12610607
rs36045835
rs5030348
rs139742916
rs4804554
rs141625758
rs12975916
rs12976824
rs68010235
rs12979495
rs35519966
rs4804529
rs12611017
rs75407647
rs14052</code></pre>
<pre><code>Warning in format_data(as.data.frame(exposure_dat), type = &quot;exposure&quot;, snps =
NULL, : None of the provided SNPs can be used for MR analysis, they are missing
required information.</code></pre>
<pre class="r"><code>TYK2_T1D  &lt;- read_outcome_data(&quot;data/export/TYK2_T1D_TwoSampleMR.csv&quot;, sep=&quot;,&quot;)

TYK2 &lt;- harmonise_data(TYK2_eqtl, TYK2_T1D)</code></pre>
<pre><code>Harmonising Whole blood TYK2 mRNA (4vucPG) and Risk of type 1 diabetes (xN25Oe)</code></pre>
<pre><code>Removing the following SNPs for being palindromic with intermediate allele frequencies:
rs1037686, rs10409562, rs10413630, rs10416414, rs10417443, rs10421649, rs11085587, rs113098768, rs113623676, rs11668706, rs11673302, rs12460740, rs12710260, rs12979606, rs12981155, rs12981841, rs12982396, rs12982962, rs12985305, rs13344456, rs1368459, rs1644726, rs1644730, rs1862471, rs2016304, rs2075741, rs2287813, rs2305792, rs2361222, rs2445885, rs2569703, rs2738446, rs2738448, rs281432, rs322144, rs322145, rs3786698, rs3826709, rs3826803, rs4611572, rs4804507, rs4804526, rs55686442, rs61032644, rs62128757, rs6511685, rs67475684, rs6887, rs710845, rs7245579, rs7252965, rs7254559, rs7256356, rs7256625, rs7258340, rs73017023, rs749261, rs753859, rs8112953, rs8113000, rs8182463, rs885743, rs896412, rs918539, rs9973219</code></pre>
<div id="manhattan-plots" class="section level2">
<h2>Manhattan plots</h2>
<pre class="r"><code>#create Manhattan plots with vertical lines indicating the selected area (see below 2.4)
TYK2_eqtl.fig &lt;-
  TYK2 %&gt;%
  ggplot(aes(x = pos.exposure/1000, y=-log10(pval.exposure))) +
  geom_point()+
  geom_hline(yintercept=8, linetype=&quot;dashed&quot;)+
  ggtitle(&quot;GWAS of TYK2 gene expression in blood&quot;) +
  ylab(&quot;-log10(p)&quot;) +
  xlab(NULL)

TYK2_T1D.fig &lt;-
  TYK2 %&gt;%
  ggplot()+
  geom_point(mapping=aes(x = pos.exposure/1000, y=-log10(pval.outcome))) +
  geom_hline(yintercept=8, linetype=&quot;dashed&quot;)+
  # geom_vline(xintercept=36.9e6/1000, linetype=&quot;dashed&quot;)+
  # geom_vline(xintercept=37.3e6/1000, linetype=&quot;dashed&quot;)+
  ggtitle(&quot;GWAS of risk of T1D&quot;) +
  ylab(&quot;-log10(p)&quot;) +
  xlab(&quot;position(kbp)&quot;)

#Obtain range for figure of gene
layer_scales(TYK2_T1D.fig)$x$range$range</code></pre>
<pre><code>[1]  9572.19 11474.33</code></pre>
<pre class="r"><code>TYK2_gene.fig &lt;-
  ggplot(data = TYK2) +
  geom_blank() +
  geom_segment(x=37521.878, xend=37571.094, y=1, yend=1, linewidth = 2) +
  annotate(&quot;text&quot;, label = &quot;TYK2&quot;,
           x= (37521.878 + 37571.094) / 2, y=1, hjust = 0.5, vjust = -1) +
  xlim(36521.878, 38571.094) +
  ylim(0.75, 2) +
  ylab(NULL) + xlab(NULL) +
  theme(axis.text.y = element_blank(), axis.ticks.y = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank())

fig_TYK2_manhattan &lt;- ggarrange(TYK2_gene.fig, TYK2_eqtl.fig, TYK2_T1D.fig,
                                heights = c(1, 3, 3), nrow = 3,
                                ncol = 1, align = &quot;hv&quot;)
# ggsave(&quot;figures/TYK2-manhattan-combined-fig.pdf&quot;,
#        height = 6, width = 8, units = &quot;in&quot;)</code></pre>
</div>
<div id="coloc-assuming-1-causal-variant" class="section level2">
<h2>Coloc assuming 1 causal variant</h2>
<pre class="r"><code>#recode MAF
TYK2 &lt;- TYK2 %&gt;%
  mutate(maf.exposure  = ifelse(eaf.exposure  &lt; 0.5, eaf.exposure,  1-eaf.exposure)) %&gt;%
  mutate(maf.outcome  = ifelse(eaf.outcome  &lt; 0.5, eaf.outcome,  1-eaf.outcome))

colnames(TYK2)</code></pre>
<pre><code> [1] &quot;SNP&quot;                    &quot;effect_allele.exposure&quot; &quot;other_allele.exposure&quot; 
 [4] &quot;effect_allele.outcome&quot;  &quot;other_allele.outcome&quot;   &quot;beta.exposure&quot;         
 [7] &quot;beta.outcome&quot;           &quot;eaf.exposure&quot;           &quot;eaf.outcome&quot;           
[10] &quot;remove&quot;                 &quot;palindromic&quot;            &quot;ambiguous&quot;             
[13] &quot;id.outcome&quot;             &quot;pos.outcome&quot;            &quot;pval.outcome&quot;          
[16] &quot;se.outcome&quot;             &quot;ncase.outcome&quot;          &quot;ncontrol.outcome&quot;      
[19] &quot;samplesize.outcome&quot;     &quot;outcome&quot;                &quot;mr_keep.outcome&quot;       
[22] &quot;pval_origin.outcome&quot;    &quot;data_source.outcome&quot;    &quot;pval.exposure&quot;         
[25] &quot;pos.exposure&quot;           &quot;samplesize.exposure&quot;    &quot;se.exposure&quot;           
[28] &quot;exposure&quot;               &quot;mr_keep.exposure&quot;       &quot;pval_origin.exposure&quot;  
[31] &quot;id.exposure&quot;            &quot;data_source.exposure&quot;   &quot;action&quot;                
[34] &quot;mr_keep&quot;                &quot;maf.exposure&quot;           &quot;maf.outcome&quot;           </code></pre>
<pre class="r"><code>D1 &lt;- list(
  type = &quot;quant&quot;, # quantitative trait
  pvalues = TYK2$pval.exposure,
  N = TYK2$samplesize.exposure,
  MAF = TYK2$maf.exposure,
  pos = TYK2$pos.exposure,
  snp = TYK2$pos.exposure,
  sdY = 1)

D2 &lt;- list(
  type = &quot;cc&quot;, # case-control trait
  Beta = TYK2$beta.outcome,
  varBeta = TYK2$se.outcome,
  pvalues = TYK2$pval.outcome,
  N = 18942+501638, # Case-control study (Chiou et al. 2021 Nature)
  s = 18942/(18942+501638), # N_case/(N_case+ N_ctrl)
  MAF = TYK2$maf.exposure,  #eqtl used here in purpose
  pos = TYK2$pos.outcome,
  snp = TYK2$pos.outcome)

check_dataset(D1)</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>check_dataset(D2)</code></pre>
<pre><code>NULL</code></pre>
<pre class="r"><code>coloc_TYK2 &lt;- coloc.abf(D1, D2, p1 = 1e-4, p2 = 1e-4, p12 = 1e-5)</code></pre>
<pre><code>PP.H0.abf PP.H1.abf PP.H2.abf PP.H3.abf PP.H4.abf 
 2.62e-03  9.90e-01  1.53e-06  5.75e-04  6.37e-03 
[1] &quot;PP abf for shared variant: 0.637%&quot;</code></pre>
<pre class="r"><code>sensitivity(coloc_TYK2, &quot;H4 &gt; 0.7&quot;)</code></pre>
<pre><code>Results fail decision rule H4 &gt; 0.7</code></pre>
<p><img src="figure/TYK2.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="pqtl" class="section level1">
<h1>pQTL</h1>
<pre class="r"><code>TYK2_prot &lt;- read_exposure_data(&quot;data/export/TYK2_prot_anno_TwoSampleMR.csv&quot;, sep=&quot;,&quot;)
TYK2 &lt;- harmonise_data(TYK2_prot, TYK2_T1D)</code></pre>
<pre><code>Harmonising Serum TYK2 level (qs1g78) and Risk of type 1 diabetes (xN25Oe)</code></pre>
<pre><code>Removing the following SNPs for incompatible alleles:
rs151165116, rs183479123, rs185264289, rs373524333, rs374964295, rs534737045, rs537688931, rs544285808, rs546195204, rs555277654, rs558187292, rs558830521, rs571297393, rs571727951, rs572069029, rs747556448, rs757688587, rs767437573, rs77444625, rs775432199, rs779038836</code></pre>
<pre><code>Removing the following SNPs for being palindromic with intermediate allele frequencies:
rs1037686, rs10409562, rs10416414, rs10417443, rs11085587, rs11668537, rs11668706, rs11673302, rs12462944, rs12975236, rs12979606, rs12981841, rs12982396, rs12982962, rs12985305, rs13344456, rs1368459, rs1644726, rs1644730, rs1862471, rs1895371, rs1968236, rs2075741, rs2161468, rs2287813, rs2305792, rs2445885, rs2569703, rs2738446, rs2738448, rs281432, rs322144, rs322145, rs3786698, rs3826709, rs3826803, rs4804161, rs4804526, rs55686442, rs61032644, rs62104327, rs62128757, rs6511685, rs67475684, rs6887, rs710845, rs7245579, rs7252965, rs7254559, rs7256625, rs7258340, rs73017023, rs749261, rs753859, rs8112953, rs8113000, rs8182463, rs885743, rs896412, rs918539, rs9973219</code></pre>
<div id="manhattan-plots-1" class="section level2">
<h2>Manhattan plots</h2>
<p>There are no GWAS hits in TYK2 levels in serum. Thus, coloc and MR is
not done.</p>
<pre><code>[1]  9461.514 11491.155</code></pre>
<p><img src="figure/TYK2.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 14393)

Matrix products: default

locale:
[1] LC_COLLATE=Finnish_Finland.1252  LC_CTYPE=Finnish_Finland.1252   
[3] LC_MONETARY=Finnish_Finland.1252 LC_NUMERIC=C                    
[5] LC_TIME=Finnish_Finland.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] here_1.0.1        rio_0.5.29        ieugwasr_0.1.5    ggpubr_0.4.0     
 [5] TwoSampleMR_0.5.6 coloc_5.1.0.1     data.table_1.14.2 forcats_0.5.1    
 [9] stringr_1.4.0     dplyr_1.0.9       purrr_0.3.4       readr_2.1.2      
[13] tidyr_1.2.0       tibble_3.1.7      ggplot2_3.4.1     tidyverse_1.3.1  

loaded via a namespace (and not attached):
 [1] matrixStats_0.62.0 fs_1.5.2           lubridate_1.8.0    httr_1.4.3        
 [5] rprojroot_2.0.3    tools_4.2.1        backports_1.4.1    bslib_0.4.0       
 [9] utf8_1.2.2         R6_2.5.1           irlba_2.3.5        nortest_1.0-4     
[13] DBI_1.1.3          colorspace_2.0-3   withr_2.5.0        tidyselect_1.1.2  
[17] gridExtra_2.3      curl_4.3.2         compiler_4.2.1     git2r_0.30.1      
[21] glmnet_4.1-4       cli_3.6.0          rvest_1.0.2        xml2_1.3.3        
[25] labeling_0.4.2     sass_0.4.2         scales_1.2.0       mixsqp_0.3-43     
[29] digest_0.6.29      foreign_0.8-82     rmarkdown_2.14     pkgconfig_2.0.3   
[33] htmltools_0.5.2    mr.raps_0.2        highr_0.9          dbplyr_2.2.1      
[37] fastmap_1.1.0      rlang_1.0.6        readxl_1.4.0       susieR_0.12.16    
[41] rstudioapi_0.13    farver_2.1.1       shape_1.4.6        jquerylib_0.1.4   
[45] generics_0.1.3     jsonlite_1.8.0     zip_2.2.0          car_3.1-0         
[49] magrittr_2.0.3     Matrix_1.4-1       Rcpp_1.0.8.3       munsell_0.5.0     
[53] fansi_1.0.3        abind_1.4-5        viridis_0.6.2      lifecycle_1.0.3   
[57] stringi_1.7.6      whisker_0.4        yaml_2.3.5         carData_3.0-5     
[61] plyr_1.8.7         grid_4.2.1         promises_1.2.0.1   crayon_1.5.1      
[65] lattice_0.20-45    cowplot_1.1.1      splines_4.2.1      haven_2.5.0       
[69] hms_1.1.1          knitr_1.39         pillar_1.8.0       ggsignif_0.6.3    
[73] codetools_0.2-18   reprex_2.0.1       glue_1.6.2         evaluate_0.15     
[77] modelr_0.1.8       vctrs_0.5.2        tzdb_0.3.0         httpuv_1.6.5      
[81] foreach_1.5.2      cellranger_1.1.0   gtable_0.3.0       reshape_0.8.9     
[85] assertthat_0.2.1   cachem_1.0.6       openxlsx_4.2.5.1   xfun_0.31         
[89] broom_1.0.0        rstatix_0.7.0      later_1.3.0        survival_3.3-1    
[93] viridisLite_0.4.0  iterators_1.0.14   workflowr_1.7.0    ellipsis_0.3.2    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
